<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta property="og:title" content="Chapter 7 Reproducible research | Introduction to bioinformatics" />
<meta property="og:type" content="book" />


<meta property="og:description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2019-01-07" />


<meta name="description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain.">

<title>Chapter 7 Reproducible research | Introduction to bioinformatics</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/jquery-1.12.4/jquery.min.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.5/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>



</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Introduction to bioinformatics<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html">Preamble</a>
<a href="dataorg.html"><span class="toc-section-number">1</span> Data organisation with Spreadsheets</a>
<a href="starting-with-r.html"><span class="toc-section-number">2</span> Starting with R</a>
<a href="starting-with-data.html"><span class="toc-section-number">3</span> Starting with data</a>
<a href="manipulating-and-analyzing-data-with-dplyr.html"><span class="toc-section-number">4</span> Manipulating and analyzing data with dplyr</a>
<a href="data-visualization.html"><span class="toc-section-number">5</span> Data visualization</a>
<a href="joining-tables.html"><span class="toc-section-number">6</span> Joining tables</a>
<a id="active-page" href="reproducible-research.html"><span class="toc-section-number">7</span> Reproducible research</a><ul class="toc-sections">
<li class="toc"><a href="#working-reproducibly"> Working reproducibly</a></li>
<li class="toc"><a href="#knitr-and-rmarkdown"> <code>knitr</code> and <code>rmarkdown</code></a></li>
<li class="toc"><a href="#additional-features"> Additional features</a></li>
<li class="toc"><a href="#docker-containers"> Docker containers</a></li>
</ul>
<a href="sec-bioinfo.html"><span class="toc-section-number">8</span> Bioinformatics</a>
<a href="conclusions.html"><span class="toc-section-number">9</span> Conclusions</a>
<a href="session-information.html">Session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="reproducible-research" class="section level1">
<h1><span class="header-section-number">Chapter 7</span> Reproducible research</h1>
<p>For a general introduction on the topic in French, see <span class="citation">Pouzat, Davison, and Hinsen (<label for="tufte-mn-12" class="margin-toggle">&#8853;</label><input type="checkbox" id="tufte-mn-12" class="margin-toggle">2015<span class="marginnote">Pouzat, Christophe, Andrew Davison, and Konrad Hinsen. 2015. “La Recherche Reproductible : Une Communication Scientifique Explicite.” <em>Statistique et Société</em> 3 (1). <a href="http://www.publications-sfds.fr/index.php/stat_soc/article/view/448">http://www.publications-sfds.fr/index.php/stat_soc/article/view/448</a>.</span>)</span>.</p>
<div id="working-reproducibly" class="section level2">
<h2><span class="header-section-number">7.1</span> Working reproducibly</h2>
<p>Possibly explain briefly the different types of preproducibility.</p>
<p>Useful reference: <span class="citation">Markowetz (<label for="tufte-mn-13" class="margin-toggle">&#8853;</label><input type="checkbox" id="tufte-mn-13" class="margin-toggle">2015<span class="marginnote">Markowetz, Florian. 2015. “Five Selfish Reasons to Work Reproducibly.” <em>Genome Biology</em> 16 (1): 274. <a href="https://doi.org/10.1186/s13059-015-0850-7">https://doi.org/10.1186/s13059-015-0850-7</a>.</span>)</span>.</p>
</div>
<div id="knitr-and-rmarkdown" class="section level2">
<h2><span class="header-section-number">7.2</span> <code>knitr</code> and <code>rmarkdown</code></h2>
<p>Reproducible research is an essential part of any data analysis. With
the tools that are available, one can argue that it has become more
difficult not to produce reproducible reports than to producing then.</p>
<p>Reproducible documents have been a part of R since the very
beginning. See for example <span class="citation">Gentleman and Temple Lang (<label for="tufte-mn-14" class="margin-toggle">&#8853;</label><input type="checkbox" id="tufte-mn-14" class="margin-toggle">2004<span class="marginnote">Gentleman, Robert, and Duncan Temple Lang. 2004. “Statistical Analyses and Reproducible Research.” <em>Bioconductor Project Working Papers. Working Paper 2</em>. <a href="https://biostats.bepress.com/bioconductor/paper2">https://biostats.bepress.com/bioconductor/paper2</a>.</span>)</span>, to see how such <em>compendia</em> play a
central role within the <a href="https://bioconductor.org/">Bioconductor</a>
project (more about Bioconductor in it’s dedicated
chapter). Originally, these were written in LaTeX, interleaved with R
code chunks, forming so called Sweave documents (with extension
<code>.Rnw</code>).</p>
<p>More recently, it has become to use the
<a href="https://daringfireball.net/projects/markdown/">markdown</a> syntax
markup language, rather than LaTeX. Once interleaved with R code
chunks, these documents become Rmarkdown files (<code>.Rmd</code>). The can be
converted into markdown using <code>knitr::knit</code>, that executes the code
chunk and incorporates their output in the resulting markdown
documents, which itself is converted to one of many output formats,
typically pdf of html using <a href="http://pandoc.org/">pandoc</a>. I R, this
final conversion is done using <code>rmarkdown::render</code> (that relies on
pandoc). The <a href="http://rmarkdown.rstudio.com/">rmarkdown</a> package is
developed and maintained by RStudio and benefits from excellent
documentation, support and integrates into the RStudio editor.</p>
<p><img src="figs/rmd.png" /></p>
<p>RStudio now also support <em>R Notebook</em> documents<label for="tufte-sn-5" class="margin-toggle sidenote-number">5</label><input type="checkbox" id="tufte-sn-5" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">5</span> See also <a href="https://jupyter.org/">Jupyter notebooks</a>
initially developed for Python, but that can run R code as well.</span> that execute
individual code chunks that are displayed directly in the source
document.</p>
<p><img src="figs/rnb.png" /></p>
</div>
<div id="additional-features" class="section level2">
<h2><span class="header-section-number">7.3</span> Additional features</h2>
<ul>
<li><p>Among the most options that can set for code chunks is
<code>cache</code>. Setting <code>cache=TRUE</code> will avoid that specific code chunk to
be cached and not recomputed every time the documented in <em>knitted</em>,
unless the code chunk was modified. This is an important feature
when long computations are necessary.</p></li>
<li><p>The <code>DT::datatable</code> function allows to create dynamic tables
directly from R, as show below.</p></li>
</ul>
<pre class="sourceCode r"><code class="sourceCode r">ip &lt;-<span class="st"> </span><span class="kw">installed.packages</span>()
DT<span class="op">::</span><span class="kw">datatable</span>(ip)</code></pre>
<div id="htmlwidget-602b47c1ae4e0aa1655d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-602b47c1ae4e0aa1655d">{"x":{"filter":"none","data":[["abind","acepack","affy","affyio","ALL","annotate","AnnotationDbi","AnnotationHub","apeglm","assertthat","backports","base64enc","bbmle","beachmat","beeswarm","BH","bibtex","bigmemory","bigmemory.sri","bindr","bindrcpp","Biobase","biobroom","BiocGenerics","BiocInstaller","BiocManager","BiocParallel","BiocStyle","BiocVersion","biocViews","bioDist","biomaRt","Biostrings","bit","bit64","bitops","blob","bookdown","BRAIN","brew","broom","callr","caret","caTools","cellranger","checkmate","chron","circlize","cleaver","cli","clipr","clisymbols","coda","colorspace","commonmark","ComplexHeatmap","crayon","crosstalk","curl","CVST","data.table","DBI","dbplyr","ddalpha","DelayedArray","DelayedMatrixStats","dendextend","DEoptimR","desc","DESeq","DESeq2","devtools","digest","dimRed","diptest","DO.db","doParallel","DOSE","dplyr","DRR","DT","e1071","edgeR","emdbook","evaluate","ExperimentHub","faahKO","fansi","fastmatch","fgsea","flexmix","FNN","forcats","foreach","formatR","Formula","fpc","fs","futile.logger","futile.options","gbm","gdata","genefilter","geneplotter","generics","GenomeInfoDb","GenomeInfoDbData","GenomicRanges","geometry","GetoptLong","GGally","ggbeeswarm","ggplot2","ggrepel","ggvis","gh","git2r","Glimma","GlobalOptions","glue","gmm","GO.db","GOSemSim","gower","gplots","graph","gridBase","gridExtra","grImport2","gsubfn","gtable","gtools","haven","HDF5Array","hexbin","highlight","highr","Hmisc","hms","hpar","htmlTable","htmltools","htmlwidgets","httpuv","httr","hwriter","igraph","impute","imputeLCMD","ini","interactiveDisplayBase","ipred","IRanges","iterators","jpeg","jsonlite","kernlab","KinSwingR","knitr","labeling","lambda.r","LaplacesDemon","later","latticeExtra","lava","lazyeval","limma","locfit","lpSolve","lubridate","magic","magrittr","MALDIquant","manipulateWidget","markdown","MassSpecWavelet","matrixStats","mcclust","mclust","memoise","metRology","microbenchmark","mime","miniUI","mixtools","mlbench","MLInterfaces","mnormt","ModelMetrics","modelr","modeltools","msdata","msmbstyle","msmsEDA","msmsTests","MSnbase","MSnID","MSstatsBioData","multtest","munsell","mvtnorm","mzID","mzR","ncdf4","network","nipals","NMF","norm","numDeriv","officer","openssl","org.Hs.eg.db","org.Mm.eg.db","packrat","patchwork","pcaMethods","pillar","pkgbuild","pkgconfig","pkgdown","pkgload","pkgmaker","PKI","plogr","plotly","pls","plyr","png","PolynomF","prabclus","praise","preprocessCore","prettyunits","processx","prodlim","progress","pRoloc","pRolocdata","pRolocGUI","promises","ProtGenerics","proto","protViz","proxy","pryr","ps","psy","purrr","qvalue","R.cache","R.methodsS3","R.oo","R.utils","R6","randomForest","RANN","RBGL","rcmdcheck","RColorBrewer","Rcpp","RcppArmadillo","RcppClassic","RcppEigen","RcppNumerical","RcppRoll","RCurl","rda","Rdisop","readr","readxl","recipes","redoc","registry","rematch","rematch2","remotes","reprex","reshape","reshape2","RforProteomics","rgl","rhdf5","Rhdf5lib","RIdeogram","rjson","RJSONIO","rlang","rmarkdown","rngtools","robustbase","rols","roxygen2","rprojroot","rpx","rsconnect","RSQLite","rstudioapi","rsvg","Rtsne","RUnit","rvest","rWikiPathways","S4Vectors","sampling","sandwich","scales","scater","segmented","selectr","servr","sessioninfo","sfsmisc","shape","shiny","shinycssloaders","shinydashboard","SingleCellExperiment","snow","sourcetools","SparseM","sqldf","SQUAREM","stringi","stringr","sugsvarsel","SummarizedExperiment","synapter","synapterdata","testthat","threejs","tibble","tidies","tidyr","tidyselect","tidyverse","timeDate","tinytex","tmvtnorm","topGO","trimcluster","tufte","UpSetR","usethis","utf8","uuid","Vennerable","vipor","viridis","viridisLite","visNetwork","vsn","waveslim","webshot","whisker","whoami","withr","xaringan","xcms","xfun","XML","xml2","xopen","xtable","XVector","yaml","zeallot","zip","zlibbioc","zoo","base","boot","class","cluster","codetools","compiler","datasets","foreign","graphics","grDevices","grid","KernSmooth","lattice","MASS","Matrix","methods","mgcv","nlme","nnet","parallel","rJava","rpart","spatial","splines","stats","stats4","survival","tcltk","tools","utils"],["abind","acepack","affy","affyio","ALL","annotate","AnnotationDbi","AnnotationHub","apeglm","assertthat","backports","base64enc","bbmle","beachmat","beeswarm","BH","bibtex","bigmemory","bigmemory.sri","bindr","bindrcpp","Biobase","biobroom","BiocGenerics","BiocInstaller","BiocManager","BiocParallel","BiocStyle","BiocVersion","biocViews","bioDist","biomaRt","Biostrings","bit","bit64","bitops","blob","bookdown","BRAIN","brew","broom","callr","caret","caTools","cellranger","checkmate","chron","circlize","cleaver","cli","clipr","clisymbols","coda","colorspace","commonmark","ComplexHeatmap","crayon","crosstalk","curl","CVST","data.table","DBI","dbplyr","ddalpha","DelayedArray","DelayedMatrixStats","dendextend","DEoptimR","desc","DESeq","DESeq2","devtools","digest","dimRed","diptest","DO.db","doParallel","DOSE","dplyr","DRR","DT","e1071","edgeR","emdbook","evaluate","ExperimentHub","faahKO","fansi","fastmatch","fgsea","flexmix","FNN","forcats","foreach","formatR","Formula","fpc","fs","futile.logger","futile.options","gbm","gdata","genefilter","geneplotter","generics","GenomeInfoDb","GenomeInfoDbData","GenomicRanges","geometry","GetoptLong","GGally","ggbeeswarm","ggplot2","ggrepel","ggvis","gh","git2r","Glimma","GlobalOptions","glue","gmm","GO.db","GOSemSim","gower","gplots","graph","gridBase","gridExtra","grImport2","gsubfn","gtable","gtools","haven","HDF5Array","hexbin","highlight","highr","Hmisc","hms","hpar","htmlTable","htmltools","htmlwidgets","httpuv","httr","hwriter","igraph","impute","imputeLCMD","ini","interactiveDisplayBase","ipred","IRanges","iterators","jpeg","jsonlite","kernlab","KinSwingR","knitr","labeling","lambda.r","LaplacesDemon","later","latticeExtra","lava","lazyeval","limma","locfit","lpSolve","lubridate","magic","magrittr","MALDIquant","manipulateWidget","markdown","MassSpecWavelet","matrixStats","mcclust","mclust","memoise","metRology","microbenchmark","mime","miniUI","mixtools","mlbench","MLInterfaces","mnormt","ModelMetrics","modelr","modeltools","msdata","msmbstyle","msmsEDA","msmsTests","MSnbase","MSnID","MSstatsBioData","multtest","munsell","mvtnorm","mzID","mzR","ncdf4","network","nipals","NMF","norm","numDeriv","officer","openssl","org.Hs.eg.db","org.Mm.eg.db","packrat","patchwork","pcaMethods","pillar","pkgbuild","pkgconfig","pkgdown","pkgload","pkgmaker","PKI","plogr","plotly","pls","plyr","png","PolynomF","prabclus","praise","preprocessCore","prettyunits","processx","prodlim","progress","pRoloc","pRolocdata","pRolocGUI","promises","ProtGenerics","proto","protViz","proxy","pryr","ps","psy","purrr","qvalue","R.cache","R.methodsS3","R.oo","R.utils","R6","randomForest","RANN","RBGL","rcmdcheck","RColorBrewer","Rcpp","RcppArmadillo","RcppClassic","RcppEigen","RcppNumerical","RcppRoll","RCurl","rda","Rdisop","readr","readxl","recipes","redoc","registry","rematch","rematch2","remotes","reprex","reshape","reshape2","RforProteomics","rgl","rhdf5","Rhdf5lib","RIdeogram","rjson","RJSONIO","rlang","rmarkdown","rngtools","robustbase","rols","roxygen2","rprojroot","rpx","rsconnect","RSQLite","rstudioapi","rsvg","Rtsne","RUnit","rvest","rWikiPathways","S4Vectors","sampling","sandwich","scales","scater","segmented","selectr","servr","sessioninfo","sfsmisc","shape","shiny","shinycssloaders","shinydashboard","SingleCellExperiment","snow","sourcetools","SparseM","sqldf","SQUAREM","stringi","stringr","sugsvarsel","SummarizedExperiment","synapter","synapterdata","testthat","threejs","tibble","tidies","tidyr","tidyselect","tidyverse","timeDate","tinytex","tmvtnorm","topGO","trimcluster","tufte","UpSetR","usethis","utf8","uuid","Vennerable","vipor","viridis","viridisLite","visNetwork","vsn","waveslim","webshot","whisker","whoami","withr","xaringan","xcms","xfun","XML","xml2","xopen","xtable","XVector","yaml","zeallot","zip","zlibbioc","zoo","base","boot","class","cluster","codetools","compiler","datasets","foreign","graphics","grDevices","grid","KernSmooth","lattice","MASS","Matrix","methods","mgcv","nlme","nnet","parallel","rJava","rpart","spatial","splines","stats","stats4","survival","tcltk","tools","utils"],["/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library","/opt/Rpatched/lib64/R/library"],["1.4-5","1.4.1","1.60.0","1.52.0","1.24.0","1.60.0","1.44.0","2.14.2","1.4.1","0.2.0","1.1.3","0.1-3","1.0.20","1.4.0","0.2.3","1.66.0-1","0.4.2","4.5.33","0.1.3","0.1.1","0.2.2","2.42.0","1.14.0","0.28.0","1.32.1","1.30.4","1.16.2","2.10.0","3.8.0","1.50.9","1.54.0","2.38.0","2.50.1","1.1-14","0.9-7","1.0-6","1.1.1","0.9","1.28.0","1.0-6","0.5.1","3.1.0","6.0-81","1.17.1.1","1.1.0","1.8.5","2.3-53","0.4.5","1.20.0","1.0.1","0.4.1","1.2.0","0.19-2","1.3-2","1.7","1.20.0","1.3.4","1.0.0","3.2","0.2-2","1.11.8","1.0.0","1.2.2","1.3.7","0.8.0","1.4.0","1.9.0","1.0-8","1.2.0","1.34.0","1.22.1","2.0.1","0.6.18","0.2.2","0.75-7","2.9","1.0.14","3.8.0","0.7.8","0.0.3","0.5","1.7-0","3.24.1","1.3.10","0.12","1.8.0","1.22.0","0.4.0","1.1-0","1.8.0","2.3-14","1.1.2.2","0.3.0","1.4.4","1.5","1.2-3","2.1-11.1","1.2.6","1.4.3","1.0.1","2.1.4","2.18.0","1.64.0","1.60.0","0.0.2","1.18.1","1.2.0","1.34.0","0.3-6","0.1.7","1.4.0","0.6.0","3.1.0","0.8.0","0.4.4","1.0.1","0.23.0","1.10.0","0.1.0","1.3.0","1.6-2","3.7.0","2.8.0","0.1.2","3.0.1","1.60.0","0.4-7","2.3","0.1-4","0.7","0.2.0","3.8.1","2.0.0","1.10.1","1.27.2","0.4.7.2","0.7","4.1-1","0.4.2","1.24.0","1.12","0.3.6","1.3","1.4.5","1.4.0","1.3.2","1.2.2","1.56.0","2.0","0.3.1","1.20.0","0.9-8","2.16.0","1.0.10","0.1-8","1.6","0.9-27","1.0.3","1.21","0.3","1.2.3","16.1.1","0.7.5","0.6-28","1.6.4","0.2.1","3.38.3","1.5-9.1","5.6.13","1.7.4","1.5-9","1.5","1.18","0.10.0","0.9","1.48.0","0.54.0","1.0","5.4.2","1.1.0","0.9-28-1","1.4-6","0.6","0.1.1.1","1.1.0","2.1-1","1.62.0","1.5-5","1.2.2","0.1.2","0.2-22","0.22.0","0.0.3","1.20.0","1.20.0","2.9.3","1.16.1","1.4.0","2.38.0","0.5.0","1.0-8","1.20.0","2.16.0","1.16","1.13.0.1","0.5","0.21.0","1.0-9.5","2016.8-1","0.3.2","1.1","3.7.0","3.7.0","0.5.0","0.0.1","1.74.0","1.3.1","1.0.2","2.0.2","1.3.0","1.0.2","0.27","0.1-5.1","0.2.0","4.8.0","2.7-0","1.8.4","0.1-7","1.0-2","2.2-6","1.0.0","1.44.0","1.0.2","3.2.1","2018.04.18","1.2.0","1.23.1","1.20.0","1.16.0","1.0.1","1.14.0","1.0.0","0.3.1","0.4-22","0.1.4","1.2.1","1.1","0.2.5","2.14.0","0.13.0","1.7.1","1.22.0","2.7.0","2.3.0","4.6-14","2.6","1.58.1","1.3.2","1.1-2","1.0.0","0.9.200.5.0","0.9.11","0.3.3.5.0","0.3-2","0.3.0","1.95-4.11","1.0.2-2.1","1.42.0","1.3.0","1.1.0","0.1.4","0.0.0.9000","0.5","1.0.1","2.0.1","2.0.2","0.2.1","0.8.8","1.4.3","1.20.0","0.99.16","2.26.0","1.4.2","0.1.0","0.2.20","1.3-1.1","0.3.0.1","1.11","1.3.1","0.93-3","2.11.0","6.1.1","1.3-2","1.18.0","0.8.12","2.1.1","0.8","1.3","0.15","0.4.32","0.3.2","1.2.0","0.20.1","2.8","2.5-0","1.0.0","1.10.0","0.5-3.0","0.4-1","0.11","1.1.1","1.1-3","1.4.4","1.2.0","0.2.0","0.7.1","1.4.0","0.4-3","0.1.7","1.77","0.4-11","2017.10-1","1.2.4","1.3.1","0.1.0","1.12.0","2.7.0","1.20.0","2.0.1","0.3.1","1.4.2","0.0.3","0.8.2","0.2.5","1.2.1","3043.102","0.9","1.4-10","2.34.0","0.1-2.1","0.4","1.3.3","1.4.0","1.1.4","0.1-2","3.1.0.9000","0.4.5","0.5.1","0.3.0","2.0.5","3.50.0","1.7.5","0.5.1","0.3-2","1.2.0","2.1.2","0.8","3.4.1","0.4","3.98-1.16","1.2.0","1.0.0","1.8-3","0.22.0","2.2.0","0.1.0","1.0.0","1.28.0","1.8-4","3.5.1","1.3-20","7.3-14","2.0.7-1","0.2-15","3.5.1","3.5.1","0.8-71","3.5.1","3.5.1","3.5.1","2.23-15","0.20-35","7.3-51.1","1.2-14","3.5.1","1.8-25","3.1-137","7.3-12","3.5.1","0.9-10","4.1-13","7.3-11","3.5.1","3.5.1","3.5.1","2.43-1","3.5.1","3.5.1","3.5.1"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"base","recommended","recommended","recommended","recommended","base","base","recommended","base","base","base","recommended","recommended","recommended","recommended","base","recommended","recommended","recommended","base",null,"recommended","recommended","base","base","base","recommended","base","base","base"],["R (&gt;= 1.5.0)",null,"R (&gt;= 2.8.0), BiocGenerics (&gt;= 0.1.12), Biobase (&gt;= 2.5.5)","R (&gt;= 2.6.0)","R (&gt;= 2.10), Biobase (&gt;= 2.5.5)","R (&gt;= 2.10), AnnotationDbi (&gt;= 1.27.5), XML","R (&gt;= 2.7.0), methods, utils, stats4, BiocGenerics (&gt;=\n0.23.1), Biobase (&gt;= 1.17.0), IRanges","BiocGenerics (&gt;= 0.15.10)",null,null,"R (&gt;= 3.0.0)","R (&gt;= 2.9.0)","R (&gt;= 3.0.0), stats4","R (&gt;= 3.5)",null,null,"R (&gt;= 3.0.2)","R (&gt;= 3.2.0),",null,null,null,"R (&gt;= 2.10), BiocGenerics (&gt;= 0.27.1), utils","R (&gt;= 3.0.0), broom","methods, utils, graphics, stats, parallel","R (&gt;= 3.5.0), R (&lt; 3.6.0)","R (&gt;= 3.5.0)","methods",null,"R (&gt;= 3.5.0), R (&lt; 3.6.0)","R (&gt;= 3.5.0)","R (&gt;= 2.0), methods, Biobase,KernSmooth","methods","R (&gt;= 3.5.0), methods, BiocGenerics, S4Vectors, IRanges,\nXVector","R (&gt;= 2.9.2)","R (&gt;= 3.0.1), bit (&gt;= 1.1-12), utils, methods, stats",null,null,null,"R (&gt;= 2.8.1), PolynomF, Biostrings, lattice",null,"R (&gt;= 3.1)",null,"R (&gt;= 2.10), lattice (&gt;= 0.20), ggplot2","R (&gt;= 2.2.0)","R (&gt;= 3.0.0)","R (&gt;= 3.0.0)","R (&gt;= 2.12.0)","R (&gt;= 3.0.0), graphics","R (&gt;= 3.0.0), methods, Biostrings (&gt;= 1.29.8)","R (&gt;= 2.10)",null,null,"R (&gt;= 2.14.0)","R (&gt;= 2.13.0), methods",null,"R (&gt;= 3.1.2), methods, grid, graphics, stats, grDevices",null,null,"R (&gt;= 3.0.0)","kernlab,Matrix","R (&gt;= 3.1.0)","R (&gt;= 3.0.0), methods","R (&gt;= 3.2)","R (&gt;= 2.10), stats, utils, graphics, grDevices, MASS, class,\nrobustbase, sfsmisc, geometry","R (&gt;= 3.4), methods, stats4, matrixStats, BiocGenerics (&gt;=\n0.27.1), S4Vectors (&gt;= 0.19.15), IRanges (&gt;= 2.11.17),\nBiocParallel","DelayedArray (&gt;= 0.7.37)","R (&gt;= 3.0.0)",null,"R (&gt;= 3.1.0)","BiocGenerics (&gt;= 0.7.5), Biobase (&gt;= 2.21.7), locfit, lattice","S4Vectors (&gt;= 0.9.25), IRanges, GenomicRanges,\nSummarizedExperiment (&gt;= 1.1.6)","R (&gt;= 3.0.2)","R (&gt;= 3.1.0)","R (&gt;= 3.0.0), DRR",null,"R (&gt;= 2.7.0), methods, AnnotationDbi (&gt;= 1.9.7)","R (&gt;= 2.14.0), foreach(&gt;= 1.2.0), iterators(&gt;= 1.0.0),\nparallel, utils","R (&gt;= 3.4.0)","R (&gt;= 3.1.2)","kernlab, CVST, Matrix",null,null,"R (&gt;= 2.15.0), limma (&gt;= 3.34.5)",null,"R (&gt;= 3.0.2)","methods, BiocGenerics (&gt;= 0.15.10), AnnotationHub (&gt;= 2.13.4)","xcms","R (&gt;= 3.1.0)",null,"R (&gt;= 3.3), Rcpp","R (&gt;= 2.15.0), lattice","R (&gt;= 3.0.0)","R (&gt;= 3.1)","R (&gt;= 2.5.0)","R (&gt;= 3.0.2)","R (&gt;= 2.0.0), stats","R (&gt;= 2.0)","R (&gt;= 3.1)","R (&gt;= 3.0.0)","R (&gt;= 2.8.0)","R (&gt;= 2.9.0)","R (&gt;= 2.3.0)",null,"R (&gt;= 2.10), methods, Biobase, BiocGenerics, lattice, annotate","R (&gt;= 3.1)","R (&gt;= 3.1), methods, BiocGenerics (&gt;= 0.13.8), S4Vectors (&gt;=\n0.17.25), IRanges (&gt;= 2.13.12)","R (&gt;= 3.3)","R (&gt;= 2.10), methods, stats4, BiocGenerics (&gt;= 0.25.3),\nS4Vectors (&gt;= 0.19.11), IRanges (&gt;= 2.15.12), GenomeInfoDb (&gt;=\n1.15.2)","R (&gt;= 2.5.0), magic","R (&gt;= 3.0.0)","R (&gt;= 3.1), ggplot2 (&gt; 2.2.0)","R (&gt;= 3.0.0), ggplot2 (&gt;= 2.0)","R (&gt;= 3.1)","R (&gt;= 3.0.0), ggplot2 (&gt;= 2.2.0)","R (&gt;= 3.0)",null,"R (&gt;= 3.1)","R (&gt;= 3.4.0)","R (&gt;= 3.3.0), methods","R (&gt;= 3.1)","R (&gt;= 2.10.0), sandwich","R (&gt;= 2.7.0), methods, AnnotationDbi (&gt;= 1.43.1)","R (&gt;= 3.4.0)",null,"R (&gt;= 3.0)","R (&gt;= 2.10), methods, BiocGenerics (&gt;= 0.13.11)","R (&gt;= 2.3.0)",null,"R (&gt;= 2.12.0)","proto","R (&gt;= 2.14)","methods, stats, utils","R (&gt;= 3.1)","R (&gt;= 3.4), methods, DelayedArray (&gt;= 0.7.41), rhdf5 (&gt;=\n2.25.6)","R (&gt;= 2.0.1), methods","R (&gt;= 3.1.3)","R (&gt;= 3.0.2)","lattice, survival (&gt;= 2.40-1), Formula, ggplot2 (&gt;= 2.2)",null,"R (&gt;= 2.15)",null,"R (&gt;= 2.14.1)",null,"R (&gt;= 2.15.1), methods","R (&gt;= 3.1)","R (&gt;= 2.6.0)","methods","R (&gt;= 2.10)","R (&gt;= 2.10), tmvtnorm, norm, pcaMethods, impute",null,"R (&gt;= 2.10), methods, BiocGenerics","R (&gt;= 2.10)","R (&gt;= 3.1.0), methods, utils, stats, BiocGenerics (&gt;= 0.25.3),\nS4Vectors (&gt;= 0.19.11)","R (&gt;= 2.5.0), utils","R (&gt;= 2.9.0)","methods","R (&gt;= 2.10)","R (&gt;= 3.5)","R (&gt;= 3.1.0)",null,"R (&gt;= 3.0.0)","R (&gt;= 3.0.0)",null,"R (&gt;= 2.10.0), lattice, RColorBrewer","R (&gt;= 3.0)","R (&gt;= 3.1.0)","R (&gt;= 2.3.0)","R (&gt;= 2.0.1)",null,"methods, R (&gt;= 3.0.0)","R (&gt;= 2.10), abind",null,"R (&gt;= 3.2.0), methods",null,"R (&gt;= 2.11.1)","waveslim","R (&gt;= 2.12.0)","R (&gt;= 2.10), lpSolve","R (&gt;= 3.0)",null,"R (&gt;= 2.14.0), base, stats",null,null,null,"R (&gt;= 3.2.0)","R (&gt;= 2.10)","R (&gt;= 3.5), methods, BiocGenerics (&gt;= 0.13.11), Biobase,\nannotate, cluster","R (&gt;= 2.2.0)","R (&gt;= 3.2.2)","R (&gt;= 3.1)","stats, stats4","R (&gt;= 2.10)","bookdown","R (&gt;= 3.0.1), MSnbase","R (&gt;= 3.0.1), MSnbase, msmsEDA","R (&gt;= 3.1), methods, BiocGenerics (&gt;= 0.7.1), Biobase (&gt;=\n2.15.2), mzR (&gt;= 2.15.1), S4Vectors, ProtGenerics (&gt;= 1.9.1)","R (&gt;= 2.10), Rcpp","R (&gt;= 3.4)","R (&gt;= 2.10), methods, BiocGenerics, Biobase",null,"R(&gt;= 1.9.0)","methods","Rcpp (&gt;= 0.10.1), methods, utils",null,"R (&gt;= 2.10), utils","R (&gt;= 3.4.0)","R (&gt;= 3.0.0), methods, utils, pkgmaker (&gt;= 0.20), registry,\nrngtools (&gt;= 1.2.3), cluster",null,"R (&gt;= 2.11.1)",null,null,"R (&gt;= 2.7.0), methods, AnnotationDbi (&gt;= 1.43.1)","R (&gt;= 2.7.0), methods, AnnotationDbi (&gt;= 1.43.1)","R (&gt;= 3.0.0)",null,"Biobase, methods",null,"R (&gt;= 3.1)",null,"R (&gt;= 3.1.0)",null,"R (&gt;= 3.0.0), methods, stats, registry","R (&gt;= 2.9.0), base64enc",null,"R (&gt;= 3.2.0), ggplot2 (&gt;= 3.0.0)","R (&gt;= 2.10)","R (&gt;= 3.1.0)","R (&gt;= 2.9.0)","R (&gt;= 3.0.0), graphics","R (&gt;= 2.10), MASS, mclust",null,null,null,null,"R (&gt;= 2.9.0)",null,"R (&gt;= 2.15), MSnbase (&gt;= 1.19.20), MLInterfaces (&gt;= 1.37.1),\nmethods, Rcpp (&gt;= 0.10.3), BiocParallel","R (&gt;= 2.15), MSnbase","methods, R (&gt;= 3.1.0), pRoloc (&gt;= 1.11.1), Biobase, MSnbase\n(&gt;= 2.1.11)",null,"methods",null,"R (&gt;= 3.3), methods","R (&gt;= 3.4.0)","R (&gt;= 3.1.0)","R (&gt;= 3.1)",null,"R (&gt;= 3.1)","R(&gt;= 2.10)","R (&gt;= 2.14.0)","R (&gt;= 2.13.0)","R (&gt;= 2.13.0), R.methodsS3 (&gt;= 1.7.1)","R (&gt;= 2.14.0), R.oo (&gt;= 1.21.0)","R (&gt;= 3.0)","R (&gt;= 3.2.2), stats",null,"graph, methods",null,"R (&gt;= 2.0.0)","R (&gt;= 3.0.0)","R (&gt;= 3.3.0)","R (&gt;= 2.12.0)","R (&gt;= 2.15.1)",null,"R (&gt;= 2.15.1)","R (&gt;= 3.0.0), methods, bitops","R (&gt;= 2.10)","R (&gt;= 2.0.0), RcppClassic","R (&gt;= 3.1)",null,"R (&gt;= 3.1), dplyr",null,"R (&gt;= 2.6.0)",null,null,"R (&gt;= 3.0.0)","R (&gt;= 3.1)","R (&gt;= 2.6.1)","R (&gt;= 3.1)","MSnbase (&gt;= 2.5.3)","R (&gt;= 3.2.0)","R (&gt;= 3.5.0), methods",null,"R (&gt;= 3.4.0)","R (&gt;= 3.1.0)",null,"R (&gt;= 3.1.0)","R (&gt;= 3.0)","R (&gt;= 3.0.0), methods, pkgmaker (&gt;= 0.20)","R (&gt;= 3.1.0)","methods","R (&gt;= 3.1)","R (&gt;= 3.0.0)","methods","R (&gt;= 3.0.0)","R (&gt;= 3.1.0)",null,null,null,"R (&gt;= 2.5.0), utils (&gt;= 2.5.0), methods (&gt;= 2.5.0), graphics\n(&gt;= 2.5.0)","R (&gt;= 3.0.1), xml2","R (&gt;= 3.5)","R (&gt;= 3.3.0), methods, utils, stats, stats4, BiocGenerics (&gt;=\n0.23.3)",null,"R (&gt;= 2.10.0)","R (&gt;= 3.1)","R (&gt;= 3.5), SingleCellExperiment, ggplot2, BiocParallel",null,"R (&gt;= 3.0)","R (&gt;= 3.0.0)",null,"R (&gt;= 3.2.0)","R (&gt;= 2.01)","R (&gt;= 3.0.2), methods",null,"R (&gt;= 3.0)","R (&gt;= 3.5), SummarizedExperiment","R (&gt;= 2.13.1), utils","R (&gt;= 3.0.2)","R (&gt;= 2.15), methods","R (&gt;= 3.1.0), gsubfn (&gt;= 0.6), proto, RSQLite","R (&gt;= 3.0)","R (&gt;= 2.14)","R (&gt;= 3.1)","metRology, Matrix, BiocParallel, Rcpp, mnormt, mcclust","R (&gt;= 3.2), methods, GenomicRanges (&gt;= 1.33.6), Biobase,\nDelayedArray (&gt;= 0.3.20)","R (&gt;= 3.1.0), methods, MSnbase (&gt;= 2.1.2)","R (&gt;= 2.10), synapter (&gt;= 1.99.2)","R (&gt;= 3.1)","R (&gt;= 3.0.0), igraph (&gt;= 1.0.0)","R (&gt;= 3.1.0)","R (&gt;= 3.4.0), Biobase, MSnbase, dplyr, methods","R (&gt;= 3.1)","R (&gt;= 3.1)",null,"R (&gt;= 2.15.1), graphics, utils, stats, methods",null,"R (&gt;= 1.9.0), mvtnorm, utils, Matrix, stats4, gmm","R (&gt;= 2.10.0), methods, BiocGenerics (&gt;= 0.13.6), graph (&gt;=\n1.14.0), Biobase (&gt;= 2.0.0), GO.db (&gt;= 2.3.0), AnnotationDbi\n(&gt;= 1.7.19), SparseM (&gt;= 0.73)","R (&gt;= 1.9.0)",null,"R (&gt;= 3.0)","R (&gt;= 3.1)","R (&gt;= 2.10)","R (&gt;= 2.9.0)","methods, R (&gt;= 2.10)","R (&gt;= 3.0.0)","R (&gt;= 2.10), viridisLite (&gt;= 0.3.0)","R (&gt;= 2.10)","R (&gt;= 3.0)","R (&gt;= 3.4.0), Biobase","R (&gt;= 2.11.0), graphics, grDevices, stats, utils","R (&gt;= 3.0)",null,null,"R (&gt;= 3.0.2)",null,"R (&gt;= 2.14.0), methods, Biobase, BiocParallel (&gt;= 1.8.0),\nMSnbase (&gt;= 2.7.8)",null,"R (&gt;= 2.13.0), methods, utils","R (&gt;= 3.1.0)","R (&gt;= 3.1)","R (&gt;= 2.10.0)","R (&gt;= 2.8.0), methods, BiocGenerics (&gt;= 0.19.2), S4Vectors (&gt;=\n0.19.15), IRanges (&gt;= 2.15.12)",null,null,null,null,"R (&gt;= 3.1.0), stats",null,"R (&gt;= 3.0.0), graphics, stats","R (&gt;= 3.0.0), stats, utils","R (&gt;= 3.2.0)","R (&gt;= 2.1)",null,null,"R (&gt;= 3.0.0)",null,null,null,"R (&gt;= 2.5.0), stats","R (&gt;= 3.0.0)","R (&gt;= 3.1.0), grDevices, graphics, stats, utils","R (&gt;= 3.2.0)",null,"R (&gt;= 2.14.0), nlme (&gt;= 3.1-64)","R (&gt;= 3.4.0)","R (&gt;= 2.14.0), stats, utils",null,"R (&gt;= 2.5.0), methods","R (&gt;= 2.15.0), graphics, stats, grDevices","R (&gt;= 3.0.0), graphics, stats, utils",null,null,null,"R (&gt;= 2.13.0)",null,null,null],["methods, utils",null,"affyio (&gt;= 1.13.3), BiocManager, graphics, grDevices, methods,\npreprocessCore, stats, utils, zlibbioc","zlibbioc, methods",null,"Biobase, DBI, xtable, graphics, utils, stats, methods,\nBiocGenerics (&gt;= 0.13.8), RCurl","DBI, RSQLite, S4Vectors (&gt;= 0.9.25)","utils, methods, grDevices, RSQLite, BiocManager, curl,\nAnnotationDbi (&gt;= 1.31.19), S4Vectors, interactiveDisplayBase,\nhttr, yaml","emdbook, SummarizedExperiment, GenomicRanges, methods, stats,\nutils, Rcpp","tools","utils",null,"stats, numDeriv, lattice, MASS, methods","utils, methods, Rhdf5lib (&gt;= 1.1.4), rhdf5, HDF5Array (&gt;=\n1.9.5), DelayedArray (&gt;= 0.7.38), Rcpp (&gt;= 0.12.14),\nBiocGenerics","stats, graphics, grDevices, utils",null,"stringr, utils","methods, Rcpp, utils, bigmemory.sri","methods",null,"bindr (&gt;= 0.1.1), Rcpp (&gt;= 0.12.16)","methods","dplyr, tidyr, Biobase","methods, utils, graphics, stats, parallel",null,"utils","stats, utils, futile.logger, parallel, snow","bookdown, knitr (&gt;= 1.12), rmarkdown (&gt;= 1.2), stats, utils,\nyaml, BiocManager",null,"Biobase, graph (&gt;= 1.9.26), methods, RBGL (&gt;= 1.13.5), tools,\nutils, XML, RCurl, RUnit",null,"utils, XML, RCurl, AnnotationDbi, progress, stringr, httr","graphics, methods, stats, utils",null,null,null,"methods, prettyunits","htmltools (&gt;= 0.3.6), knitr (&gt;= 1.16), rmarkdown (&gt;= 1.9),\nxfun (&gt;= 0.2), tinytex, yaml (&gt;= 2.1.14)",null,null,"backports, dplyr, generics (&gt;= 0.0.2), methods, nlme, purrr,\nreshape2, stringr, tibble, tidyr","processx (&gt;= 3.2.1), R6, utils","foreach, methods, plyr, ModelMetrics (&gt;= 1.1.0), nlme,\nreshape2, stats, stats4, utils, grDevices, recipes (&gt;= 0.1.4),\nwithr (&gt;= 2.0.0)","bitops","rematch, tibble","backports (&gt;= 1.1.0), utils","graphics, stats","GlobalOptions (&gt;= 0.1.0), shape, grDevices, utils, stats,\ncolorspace, methods, grid","S4Vectors, IRanges","assertthat, crayon (&gt;= 1.3.4), methods, utils","utils",null,"lattice","graphics, grDevices",null,"circlize (&gt;= 0.4.1), GetoptLong, colorspace, RColorBrewer,\nGlobalOptions (&gt;= 0.1.0)","grDevices, methods, utils","htmltools (&gt;= 0.3.5), jsonlite, lazyeval, R6, shiny (&gt;= 0.11),\nggplot2",null,null,"methods",null,"assertthat (&gt;= 0.2.0), DBI (&gt;= 0.7), dplyr (&gt;= 0.7.4), glue\n(&gt;= 1.2.0), methods, purrr (&gt;= 0.2.4), R6 (&gt;= 2.2.2), rlang (&gt;=\n0.1.6), tibble (&gt;= 1.4.1), tidyselect (&gt;= 0.2.2), utils","Rcpp (&gt;= 0.11.0)","stats","methods, matrixStats (&gt;= 0.53.1), Matrix, S4Vectors (&gt;=\n0.17.5), IRanges, HDF5Array (&gt;= 1.7.10), BiocParallel","utils, stats, datasets, magrittr (&gt;= 1.0.1), ggplot2, fpc,\nwhisker, viridis","stats","assertthat, utils, R6, crayon, rprojroot","genefilter, geneplotter, methods, MASS, RColorBrewer","BiocGenerics (&gt;= 0.7.5), Biobase, BiocParallel, genefilter,\nmethods, locfit, geneplotter, ggplot2, Hmisc, Rcpp (&gt;= 0.11.0)","callr, cli, digest, git2r (&gt;= 0.23.0), httr (&gt;= 0.4),\njsonlite, memoise (&gt;= 1.0.0), pkgbuild (&gt;= 1.0.2), pkgload (&gt;=\n1.0.1), rcmdcheck (&gt;= 1.3.0), remotes (&gt;= 2.0.0), rstudioapi\n(&gt;= 0.6.0.9000), sessioninfo, stats, tools, usethis (&gt;= 1.4.0),\nutils, withr",null,"magrittr, methods",null,"methods, AnnotationDbi",null,"AnnotationDbi, BiocParallel, DO.db, fgsea, ggplot2, GOSemSim\n(&gt;= 2.0.0), methods, qvalue, reshape2, S4Vectors, stats, utils","assertthat (&gt;= 0.2.0), bindrcpp (&gt;= 0.2.0.9000), glue (&gt;=\n1.1.1), magrittr (&gt;= 1.5), methods, pkgconfig (&gt;= 2.0.1), R6\n(&gt;= 2.2.2), Rcpp (&gt;= 0.12.19), rlang (&gt;= 0.3.0), tibble (&gt;=\n1.4.2), tidyselect (&gt;= 0.2.3), utils","stats, methods","htmltools (&gt;= 0.3.6), htmlwidgets (&gt;= 1.3), magrittr,\ncrosstalk, promises","graphics, grDevices, class, stats, methods, utils","graphics, stats, utils, methods, locfit, Rcpp","MASS, lattice, plyr, coda, bbmle","methods","utils, S4Vectors, BiocManager, curl",null,null,null,"data.table, BiocParallel, stats, ggplot2 (&gt;= 2.2.0),\ngridExtra, grid, fastmatch, Matrix, utils","graphics, grid, grDevices, methods, modeltools (&gt;= 0.2-16),\nnnet, stats, stats4",null,"magrittr, rlang, tibble","codetools, utils, iterators",null,null,"MASS, cluster, mclust, flexmix, prabclus, class, diptest,\nmvtnorm, robustbase, kernlab, trimcluster, grDevices, graphics,\nmethods, stats, utils","methods, Rcpp","utils, lambda.r (&gt;= 1.1.0), futile.options",null,"gridExtra, lattice, parallel, survival","gtools, stats, methods, utils","S4Vectors (&gt;= 0.9.42), AnnotationDbi, annotate, Biobase,\ngraphics, methods, stats, survival","AnnotationDbi, graphics, grDevices, grid, RColorBrewer, stats,\nutils","methods","stats, stats4, utils, RCurl, GenomeInfoDbData",null,"utils, stats, XVector (&gt;= 0.19.8)",null,"rjson, GlobalOptions (&gt;= 0.1.0), methods","grDevices, grid, gtable (&gt;= 0.2.0), plyr (&gt;= 1.8.3), progress,\nRColorBrewer, reshape (&gt;= 0.8.5), utils, rlang","beeswarm, vipor","digest, grid, gtable (&gt;= 0.1.1), lazyeval, MASS, mgcv, plyr\n(&gt;= 1.7.1), reshape2, rlang (&gt;= 0.2.1), scales (&gt;= 0.5.0),\nstats, tibble, viridisLite, withr (&gt;= 2.0.0)","grid, Rcpp, scales (&gt;= 0.3.0)","assertthat, jsonlite (&gt;= 0.9.11), shiny (&gt;= 0.11.1), magrittr,\ndplyr (&gt;= 0.4.0), lazyeval, htmltools (&gt;= 0.2.4), methods","ini, jsonlite, httr","graphics, utils","edgeR, grDevices, jsonlite, methods, stats, S4Vectors, utils","utils","methods","stats, methods, grDevices, graphics","methods, AnnotationDbi","AnnotationDbi, GO.db, methods, utils",null,"gtools, gdata, stats, caTools, KernSmooth","stats, stats4, utils","graphics, grid","gtable, grid, grDevices, graphics, utils","methods, grDevices, grid, XML, png, jpeg, base64enc",null,"grid",null,"forcats (&gt;= 0.2.0), hms, Rcpp (&gt;= 0.11.4), readr (&gt;= 0.1.0),\ntibble","utils, tools, BiocGenerics (&gt;= 0.25.1), S4Vectors, IRanges","lattice, grid, graphics, grDevices, stats, utils","grDevices, tools",null,"methods, latticeExtra, cluster, rpart, nnet, acepack, foreign,\ngtable, grid, gridExtra, data.table, htmlTable (&gt;= 1.11.0),\nviridis, htmltools, base64enc","methods, pkgconfig, rlang","utils","stringr, knitr (&gt;= 1.6), magrittr (&gt;= 1.5), methods,\ncheckmate, htmlwidgets, htmltools, rstudioapi (&gt;= 0.6)","utils, digest, Rcpp","grDevices, htmltools (&gt;= 0.3), jsonlite (&gt;= 0.9.16), yaml","Rcpp (&gt;= 0.11.0), utils, promises, later (&gt;= 0.7.3)","curl (&gt;= 0.9.1), jsonlite, mime, openssl (&gt;= 0.8), R6",null,"graphics, grDevices, magrittr, Matrix, pkgconfig (&gt;= 2.0.0),\nstats, utils",null,null,null,"shiny","rpart (&gt;= 3.1-8), MASS, survival, nnet, class, prodlim","stats4",null,null,null,"methods, stats, grDevices, graphics","data.table, BiocParallel, sqldf, stats, grid","evaluate (&gt;= 0.10), highr, markdown, stringr (&gt;= 0.6), yaml\n(&gt;= 2.1.19), methods, xfun, tools",null,"formatR","parallel, grDevices, graphics, stats, utils","Rcpp (&gt;= 0.12.9), rlang","grid, stats, utils, grDevices","grDevices, graphics, methods, numDeriv, stats, survival,\nSQUAREM, utils",null,"grDevices, graphics, stats, utils, methods","lattice",null,"stringr, Rcpp (&gt;= 0.12.13),",null,null,"parallel","shiny (&gt;= 1.0.3), miniUI, htmltools, htmlwidgets, knitr,\nmethods, tools, base64enc, grDevices, codetools, webshot","utils, mime (&gt;= 0.3)",null,null,null,"stats, utils, graphics, grDevices","digest (&gt;= 0.6.3)","graphics, MASS, numDeriv, robustbase","graphics, stats","tools","shiny (&gt;= 0.13), htmltools (&gt;= 0.3), utils","MASS, segmented, stats, survival",null,"gdata, pls, sfsmisc, MASS, rpart, rda, genefilter, fpc, ggvis,\nshiny, gbm, RColorBrewer, hwriter, threejs (&gt;= 0.2.2), mlbench,\nstats4, tools, grDevices, graphics, stats",null,"Rcpp, data.table","broom, dplyr, magrittr, purrr (&gt;= 0.2.2), tibble, tidyr (&gt;=\n0.8.0), rlang (&gt;= 0.2.0)","methods",null,"tufte, htmltools, xfun, knitr (&gt;= 1.18), rmarkdown, stringr,\nmagrittr","MASS, gplots, RColorBrewer","edgeR, qvalue","BiocParallel, IRanges (&gt;= 2.13.28), plyr, preprocessCore, vsn,\ngrid, stats4, affy, impute, pcaMethods, MALDIquant (&gt;= 1.16),\nmzID (&gt;= 1.5.2), digest, lattice, ggplot2, XML, scales, MASS,\nRcpp","MSnbase (&gt;= 1.12.1), mzID (&gt;= 1.3.5), R.cache, foreach,\ndoParallel, parallel, methods, iterators, data.table, Biobase,\nProtGenerics, reshape2, dplyr, mzR",null,"survival, MASS, stats4","colorspace, methods","stats, methods","XML, plyr, parallel, doParallel, foreach, iterators,\nProtGenerics","Biobase, BiocGenerics (&gt;= 0.13.6), ProtGenerics (&gt;= 1.9.1),\nncdf4",null,null,null,"graphics, stats, stringr (&gt;= 1.0.0), digest, grid, grDevices,\ngridBase, colorspace, RColorBrewer, foreach, doParallel,\nggplot2, reshape2",null,null,"Rcpp (&gt;= 0.12.12), R6, grDevices, base64enc, zip,\ndigest,uuid,utils,stats, magrittr,htmltools, xml2 (&gt;= 1.1.0)",null,"methods, AnnotationDbi","methods, AnnotationDbi","tools, utils","ggplot2 (&gt;= 3.0.0), gtable, grid, stats, grDevices, utils","BiocGenerics, Rcpp (&gt;= 0.11.3), MASS","cli (&gt;= 1.0.0), crayon (&gt;= 1.3.4), fansi (&gt;= 0.4.0), methods,\nrlang (&gt;= 0.3.0.1), utf8 (&gt;= 1.1.3)","callr (&gt;= 2.0.0), cli, crayon, desc, prettyunits, R6,\nrprojroot, withr (&gt;= 2.1.2)","utils","callr (&gt;= 2.0.2), cli, crayon, desc, digest, evaluate, fs (&gt;=\n1.2.0), fansi, highlight, httr, magrittr, MASS, memoise,\nopenssl, pkgload (&gt;= 1.0.2), purrr, processx, R6, rematch2,\nrlang (&gt;= 0.2.0), rmarkdown (&gt;= 1.1.9007), roxygen2, rsconnect,\nrstudioapi, tibble, tools, whisker, withr, xml2 (&gt;= 1.1.1),\nyaml","desc, methods, pkgbuild, rlang, rprojroot, rstudioapi, utils,\nwithr","tools, grDevices, utils, graphics, codetools, digest, stringi,\nmagrittr, stringr, xtable, withr, bibtex (&gt;= 0.4)",null,null,"tools, scales, httr, jsonlite, magrittr, digest, viridisLite,\nbase64enc, htmltools, htmlwidgets (&gt;= 1.2), tidyr, hexbin,\nRColorBrewer, dplyr, tibble, lazyeval (&gt;= 0.2.0), rlang,\ncrosstalk, purrr, data.table, promises","grDevices, graphics, methods, stats","Rcpp (&gt;= 0.11.0)",null,"stats",null,null,"stats","magrittr, assertthat, methods","ps, R6, utils","Rcpp (&gt;= 0.11.5), stats, graphics, survival, KernSmooth, lava","hms, prettyunits, R6, crayon","Biobase, mclust (&gt;= 4.3), caret, e1071, sampling, class,\nkernlab, lattice, nnet, randomForest, proxy, FNN, hexbin,\nBiocGenerics, stats, dendextend, RColorBrewer, scales, MASS,\nknitr, mvtnorm, LaplacesDemon, coda, mixtools, gtools, plyr,\nggplot2, biomaRt, utils, grDevices, graphics","Biobase, utils","shiny (&gt;= 0.9.1), scales, dplyr, DT (&gt;= 0.1.40), graphics,\nutils, ggplot2","R6, Rcpp, later, rlang, stats, magrittr",null,null,"Rcpp (&gt;= 0.12.8)","stats, utils","stringr, codetools, methods, Rcpp (&gt;= 0.11.0)","utils",null,"magrittr (&gt;= 1.5), rlang (&gt;= 0.1), tibble","splines, ggplot2, grid, reshape2","utils, R.methodsS3 (&gt;= 1.7.1), R.oo (&gt;= 1.21.0), R.utils (&gt;=\n2.6.0), digest (&gt;= 0.6.13)","utils","methods, utils","methods, utils, tools, R.methodsS3 (&gt;= 1.7.1)",null,null,null,"methods","callr (&gt;= 2.0.0), cli, crayon, desc (&gt;= 1.2.0), digest,\npkgbuild, prettyunits, R6, rprojroot, sessioninfo (&gt;= 1.1.1),\nutils, withr, xopen",null,"methods, utils","Rcpp (&gt;= 0.11.0), stats, utils, methods","Rcpp (&gt;= 0.10.2), methods","Matrix (&gt;= 1.1-0), Rcpp (&gt;= 0.11.0), stats, utils","Rcpp","Rcpp",null,null,null,"Rcpp (&gt;= 0.12.0.5), tibble, hms (&gt;= 0.4.1), R6, clipr, methods","cellranger, Rcpp (&gt;= 0.12.12), tibble (&gt;= 1.1)","generics, gower, ipred, lubridate, magrittr, Matrix, purrr (&gt;=\n0.2.3), RcppRoll, rlang (&gt;= 0.3.0.1), stats, tibble, tidyr,\ntidyselect (&gt;= 0.1.1), timeDate, utils, withr","rmarkdown, officer, tools, utils, knitr, xfun, mime, stringi,\nxml2, backports, whoami","utils",null,"tibble","methods, utils","callr (&gt;= 2.0.0), clipr (&gt;= 0.4.0), fs, rlang, rmarkdown,\ntools, utils, whisker, withr","plyr","plyr (&gt;= 1.8.1), Rcpp, stringr","R.utils, shiny, biocViews, BiocManager","graphics, grDevices, stats, utils, htmlwidgets, htmltools,\nknitr, jsonlite (&gt;= 0.9.20), shiny, magrittr, crosstalk,\nmanipulateWidget (&gt;= 0.9.0)","Rhdf5lib (&gt;= 1.3.2)",null,"ggplot2,\ngrDevices,\ngrImport2,\nrsvg,\nscales,\ntools",null,"methods",null,"tools, utils, knitr (&gt;= 1.18), yaml (&gt;= 2.1.19), htmltools (&gt;=\n0.3.5), evaluate (&gt;= 0.8), base64enc, jsonlite, mime, tinytex\n(&gt;= 0.3), methods, stringr (&gt;= 1.2.0)","stringr, digest, utils, stats, parallel","stats, graphics, utils, methods, DEoptimR","httr, progress, jsonlite, utils, Biobase, BiocGenerics (&gt;=\n0.23.1)","brew, commonmark, desc (&gt;= 1.2.0), digest, methods, pkgload\n(&gt;= 1.0.2), purrr, R6 (&gt;= 2.1.2), Rcpp (&gt;= 0.11.0), stringi,\nstringr (&gt;= 1.0.0), utils, xml2","backports","xml2, RCurl, utils","openssl, RCurl, jsonlite, packrat (&gt;= 0.4.8-1), yaml (&gt;=\n2.1.5), rstudioapi (&gt;= 0.5)","bit64, blob (&gt;= 1.1.1), DBI (&gt;= 1.0.0), memoise, methods,\npkgconfig, Rcpp (&gt;= 0.12.7)",null,null,"Rcpp (&gt;= 0.11.0), stats",null,"httr (&gt;= 0.5), selectr, magrittr","caTools, httr, RJSONIO, utils",null,"MASS, lpSolve","stats, utils, zoo","labeling, munsell (&gt;= 0.5), R6, RColorBrewer, Rcpp,\nviridisLite","BiocGenerics, SummarizedExperiment, Matrix, dplyr, ggbeeswarm,\ngrid, DelayedMatrixStats, methods, plyr, reshape2, S4Vectors,\nstats, utils, viridis, Rcpp (&gt;= 0.12.14), DelayedArray",null,"methods, stringr, R6","mime (&gt;= 0.2), httpuv (&gt;= 1.4.0), xfun, jsonlite","cli, tools, utils, withr","grDevices, methods, utils, stats","stats, graphics, grDevices","utils, grDevices, httpuv (&gt;= 1.4.4), mime (&gt;= 0.3), jsonlite\n(&gt;= 0.9.16), xtable, digest, htmltools (&gt;= 0.3.5), R6 (&gt;= 2.0),\nsourcetools, later (&gt;= 0.7.2), promises (&gt;= 1.0.1), tools,\ncrayon, rlang","digest, glue, grDevices, shiny","utils, shiny (&gt;= 1.0.0), htmltools (&gt;= 0.2.6), promises","S4Vectors, methods, BiocGenerics, utils, stats",null,null,"graphics, stats, utils","DBI, chron",null,"tools, utils, stats","glue (&gt;= 1.2.0), magrittr, stringi (&gt;= 1.1.7)","Rcpp","utils, stats, tools, Matrix, BiocGenerics (&gt;= 0.15.3),\nS4Vectors (&gt;= 0.17.25), IRanges (&gt;= 2.13.16), GenomeInfoDb (&gt;=\n1.13.1)","RColorBrewer, lattice, qvalue, multtest, utils, tools,\nBiobase, knitr, Biostrings, cleaver (&gt;= 1.3.3), readr (&gt;= 0.2),\nrmarkdown (&gt;= 1.0)","utils","cli, crayon, digest, magrittr, methods, praise, R6 (&gt;= 2.2.0),\nrlang, withr (&gt;= 2.0.0)","htmlwidgets (&gt;= 0.3.2), base64enc, crosstalk, methods, stats","cli, crayon, methods, pillar (&gt;= 1.1.0), rlang, utils","tidyr, tibble","dplyr (&gt;= 0.7.0), glue, magrittr, purrr, Rcpp, rlang, stringi,\ntibble, tidyselect (&gt;= 0.2.5)","glue (&gt;= 1.3.0), purrr, rlang (&gt;= 0.2.2), Rcpp (&gt;= 0.12.0)","broom (&gt;= 0.4.2), cli (&gt;= 1.0.0), crayon (&gt;= 1.3.4), dplyr (&gt;=\n0.7.4), dbplyr (&gt;= 1.1.0), forcats (&gt;= 0.2.0), ggplot2 (&gt;=\n2.2.1), haven (&gt;= 1.1.0), hms (&gt;= 0.3), httr (&gt;= 1.3.1),\njsonlite (&gt;= 1.5), lubridate (&gt;= 1.7.1), magrittr (&gt;= 1.5),\nmodelr (&gt;= 0.1.1), purrr (&gt;= 0.2.4), readr (&gt;= 1.1.1), readxl\n(&gt;= 1.0.0), reprex (&gt;= 0.1.1), rlang (&gt;= 0.1.4), rstudioapi (&gt;=\n0.7), rvest (&gt;= 0.3.2), stringr (&gt;= 1.2.0), tibble (&gt;= 1.3.4),\ntidyr (&gt;= 0.7.2), xml2 (&gt;= 1.1.1)",null,"xfun (&gt;= 0.3)","stats, methods","lattice, matrixStats, DBI",null,"htmltools, xfun, knitr (&gt;= 1.18), rmarkdown","ggplot2, gridExtra, plyr, utils, stats, methods, grDevices,\nscales","clipr (&gt;= 0.3.0), clisymbols, crayon, curl (&gt;= 2.7), desc, fs\n(&gt;= 1.2.0), gh, git2r (&gt;= 0.23), glue, rlang, rprojroot (&gt;=\n1.2), rstudioapi, utils, whisker",null,null,"graph, RBGL, grid, lattice, RColorBrewer, reshape2, gtools","stats, graphics","stats, ggplot2 (&gt;= 1.0.1), gridExtra",null,"htmlwidgets, htmltools, jsonlite, magrittr, utils, methods,\ngrDevices, stats","methods, affy, limma, lattice, ggplot2",null,"magrittr, jsonlite, callr",null,"httr, jsonlite, utils","stats, graphics, grDevices","htmltools, knitr (&gt;= 1.16), servr (&gt;= 0.5), xfun (&gt;= 0.2),\nrmarkdown","mzR (&gt;= 2.13.1), BiocGenerics, ProtGenerics, lattice,\nRColorBrewer, plyr, RANN, multtest, MassSpecWavelet (&gt;= 1.5.2),\nS4Vectors, robustbase","tools",null,"Rcpp","processx","stats, utils","methods, utils, zlibbioc, BiocGenerics, S4Vectors, IRanges",null,null,null,null,"utils, graphics, grDevices, lattice (&gt;= 0.20-27)",null,null,"MASS","graphics, grDevices, stats, utils",null,null,null,"methods, utils, stats","grDevices",null,"grDevices, utils",null,"grid, grDevices, graphics, stats, utils","methods","methods, graphics, grid, stats, utils, lattice","utils, stats","methods, stats, graphics, Matrix","graphics, stats, utils, lattice",null,"tools, compiler",null,null,null,"graphics, stats","utils, grDevices, graphics","graphics, methods, stats","graphics, Matrix, methods, splines, stats, utils","utils",null,null],[null,null,"preprocessCore",null,null,null,null,null,"Rcpp, RcppEigen, RcppNumerical",null,null,null,null,"Rcpp, Rhdf5lib",null,null,null,"BH, Rcpp",null,null,"plogr, Rcpp",null,null,null,null,null,"BH",null,null,null,null,null,"S4Vectors, IRanges, XVector (&gt;= 0.21.4)",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"BH, Rcpp","S4Vectors",null,null,null,null,null,"Rcpp, RcppArmadillo",null,null,null,null,null,null,null,"BH (&gt;= 1.58.0-1), bindrcpp (&gt;= 0.2.0.9000), plogr (&gt;=\n0.1.10), Rcpp (&gt;= 0.12.15)",null,null,null,"Rcpp",null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,null,"S4Vectors, IRanges",null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,"Rcpp",null,"Rcpp, BH, later",null,null,null,null,null,null,null,null,"S4Vectors",null,null,null,null,null,null,null,null,null,"Rcpp, BH",null,null,null,null,null,null,"Rcpp,",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,"Rcpp, zlibbioc, Rhdf5lib (&gt;= 1.1.4)",null,null,null,null,null,null,"Rcpp",null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,"Rcpp",null,"Rcpp, RcppArmadillo",null,null,"later, Rcpp",null,null,"Rcpp",null,"Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp","Rcpp","Rcpp","Rcpp, RcppEigen","Rcpp",null,null,"RcppClassic, Rcpp","Rcpp, BH","Rcpp",null,null,null,null,null,null,null,null,"Rcpp",null,null,"Rhdf5lib",null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,"BH, plogr (&gt;= 0.2.0), Rcpp",null,null,"Rcpp",null,null,null,null,null,null,"Rcpp","Rhdf5lib, Rcpp, beachmat",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp, RcppArmadillo",null,null,null,null,null,null,null,"Rcpp","Rcpp (&gt;= 0.12.0),",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp (&gt;= 0.12.12)",null,null,"S4Vectors, IRanges",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,"testthat","tkWidgets (&gt;= 1.19.0), affydata, widgetTools",null,"rpart","hgu95av2.db, genefilter, Biostrings (&gt;= 2.25.10), IRanges,\nrae230a.db, rae230aprobe, tkWidgets, GO.db, org.Hs.eg.db,\norg.Mm.eg.db, hom.Hs.inp.db, humanCHRLOC, Rgraphviz, RUnit,","hgu95av2.db, GO.db, org.Sc.sgd.db, org.At.tair.db, KEGG.db,\nRUnit, TxDb.Hsapiens.UCSC.hg19.knownGene, hom.Hs.inp.db,\norg.Hs.eg.db, reactome.db, AnnotationForge, graph,\nEnsDb.Hsapiens.v75, BiocStyle, knitr","IRanges, GenomicRanges, GenomeInfoDb, VariantAnnotation,\nRsamtools, rtracklayer, BiocStyle, knitr, AnnotationForge,\nrBiopaxParser, RUnit, GenomicFeatures, MSnbase, mzR,\nBiostrings, SummarizedExperiment, ExperimentHub, gdsfmt","DESeq2, airway, knitr, rmarkdown, testthat","testthat",null,null,"emdbook, rms, ggplot2, RUnit, MuMIn, AICcmodavg, Hmisc,\noptimx (&gt;= 2013.8.6), knitr, testthat","testthat, BiocStyle, knitr, rmarkdown, Matrix, devtools",null,null,"testthat","knitr, testthat",null,"testthat","testthat","tools, tkWidgets, ALL, RUnit, golubEsets","limma, DESeq2, airway, ggplot2, plyr, GenomicRanges,\ntestthat, magrittr, edgeR, qvalue, knitr, data.table, MSnbase,\nSummarizedExperiment","Biobase, S4Vectors, IRanges, GenomicRanges, Rsamtools,\nAnnotationDbi, oligoClasses, oligo, affyPLM, flowClust, affy,\nDESeq2, MSnbase, annotate, RUnit","remotes, RUnit, BiocGenerics","BiocStyle, BiocVersion, remotes, testthat, knitr, withr","BiocGenerics, tools, foreach, BatchJobs, BBmisc, doParallel,\nRmpi, GenomicRanges, RNAseqData.HNRNPC.bam.chr14,\nTxDb.Hsapiens.UCSC.hg19.knownGene, VariantAnnotation,\nRsamtools, GenomicAlignments, ShortRead, codetools, RUnit,\nBiocStyle, knitr, batchtools, data.table","BiocGenerics, RUnit, htmltools",null,"BiocGenerics, knitr","locfit","annotate, BiocStyle, knitr, rmarkdown, testthat","BSgenome (&gt;= 1.13.14), BSgenome.Celegans.UCSC.ce2 (&gt;=\n1.3.11), BSgenome.Dmelanogaster.UCSC.dm3 (&gt;= 1.3.11),\nBSgenome.Hsapiens.UCSC.hg18, drosophila2probe, hgu95av2probe,\nhgu133aprobe, GenomicFeatures (&gt;= 1.3.14), hgu95av2cdf, affy\n(&gt;= 1.41.3), affydata (&gt;= 1.11.5), RUnit",null,null,null,"covr, pillar (&gt;= 1.2.1), testthat","htmlwidgets, rstudioapi, miniUI, rsconnect (&gt;= 0.4.3), servr\n(&gt;= 0.3), shiny, testit (&gt;= 0.5), tufte, webshot",null,null,"AER, akima, AUC, bbmle, betareg, biglm, binGroup, boot, brms,\nbtergm, car, caret, coda, covr, e1071, emmeans, ergm, gam (&gt;=\n1.15), gamlss, gamlss.data, gamlss.dist, geepack, ggplot2,\nglmnet, gmm, Hmisc, irlba, joineRML, Kendall, knitr, ks,\nLahman, lavaan, lfe, lme4, lmodel2, lmtest, lsmeans, maps,\nmaptools, MASS, Matrix, mclust, mgcv, muhaz, multcomp, network,\nnnet, orcutt (&gt;= 2.2), ordinal, plm, plyr, poLCA, psych,\nquantreg, rgeos, rmarkdown, robust, rsample, rstan, rstanarm,\nsp, speedglm, statnet.common, survey, survival, testthat,\ntseries, xergm, zoo","covr, crayon, pingr, ps, testthat, withr","BradleyTerry2, e1071, earth (&gt;= 2.2-3), fastICA, gam (&gt;=\n1.15), ipred, kernlab, knitr, klaR, MASS, ellipse, mda, mgcv,\nmlbench, MLmetrics, nnet, party (&gt;= 0.9-99992), pls, pROC,\nproxy, randomForest, RANN, spls, subselect, pamr, superpc,\nCubist, testthat (&gt;= 0.9.1), rpart, dplyr","MASS, rpart","covr, testthat (&gt;= 1.0.0), knitr, rmarkdown","R6, bit, fastmatch, data.table (&gt;= 1.9.8), devtools, ggplot2,\nknitr, magrittr, microbenchmark, rmarkdown, testthat (&gt;=\n0.11.0), tibble","scales, ggplot2","knitr, dendextend (&gt;= 1.0.1), ComplexHeatmap (&gt;= 1.13.2),\ngridBase, png","testthat (&gt;= 0.8), knitr, BiocStyle (&gt;= 0.0.14), BRAIN,\nUniProt.ws (&gt;= 2.1.4)","covr, fansi, mockery, testthat, webshot, withr","rstudioapi (&gt;= 0.5), testthat, covr","testthat",null,"datasets, stats, utils, KernSmooth, MASS, kernlab, mvtnorm,\nvcd, dichromat, tcltk, shiny, shinyjs","curl, testthat, xml2","testthat (&gt;= 0.3), knitr, markdown, cluster, MASS, pvclust,\ndendsort, HilbertCurve, Cairo, png, jpeg, tiff, fastcluster,\ndendextend (&gt;= 1.0.1)","mockery, rstudioapi, testthat, withr",null,"spelling, testthat (&gt;= 1.0.0), knitr, jsonlite, rmarkdown,\nmagrittr, httpuv, webutils",null,"bit64, curl, R.utils, knitr, xts, nanotime, zoo","blob, covr, hms, knitr, magrittr, rprojroot, rmarkdown,\nRSQLite (&gt;= 1.1-2), testthat, xml2","bit64, covr, knitr, Lahman (&gt;= 5.0.0), nycflights13 (&gt;=\n0.2.2), rmarkdown, RMariaDB (&gt;= 1.0.2), RMySQL (&gt;= 0.10.11),\nRPostgreSQL (&gt;= 0.4.1), RSQLite (&gt;= 2.0), testthat (&gt;= 2.0.0),\nwithr",null,"Matrix, HDF5Array, genefilter, SummarizedExperiment, airway,\npryr, DelayedMatrixStats, knitr, BiocStyle, RUnit","testthat, knitr, rmarkdown, covr, BiocStyle, microbenchmark,\nprofmem","knitr, rmarkdown, testthat, seriation, colorspace, plyr, ape,\nprofdpm, microbenchmark, gplots, heatmaply, d3heatmap,\ndynamicTreeCut, pvclust, corrplot, DendSer, MASS, cluster,\ncirclize (&gt;= 0.2.5), covr",null,"covr, testthat, whoami, withr","pasilla (&gt;= 0.2.10), vsn, gplots","testthat, knitr, rmarkdown, vsn, pheatmap, RColorBrewer, IHW,\napeglm, ashr, tximport, tximportData, readr, pbapply, airway,\npasilla (&gt;= 0.2.10)","BiocInstaller, BiocManager, bitops, covr (&gt;= 3.2.0), crayon,\ncurl (&gt;= 0.9), evaluate, foghorn (&gt;= 1.0.1), gmailr (&gt; 0.7.0),\nknitr, lintr (&gt;= 0.2.1), mockery, pingr, MASS, pkgdown, Rcpp\n(&gt;= 0.10.0), rhub, rmarkdown, roxygen2 (&gt;= 6.1.0), rversions,\nspelling (&gt;= 1.1), testthat (&gt;= 1.0.2), whisker","knitr, rmarkdown","NMF, MASS, Matrix, RANN, RSpectra, Rtsne, cccd, coRanking,\ndiffusionMap, energy, fastICA, ggplot2, graphics, igraph,\nkeras, kernlab, knitr, lle, loe, optimx, pcaL1, pcaPP,\nreticulate, rgl, scales, scatterplot3d, stats, tensorflow,\ntestthat, tidyr, tinytex, umap, vegan",null,null,"caret, mlbench, rpart, RUnit","BiocStyle, clusterProfiler, knitr, org.Hs.eg.db, testthat","bit64 (&gt;= 0.9.7), callr, covr (&gt;= 3.0.1), DBI (&gt;= 0.7.14),\ndbplyr (&gt;= 1.2.0), dtplyr (&gt;= 0.0.2), ggplot2 (&gt;= 2.2.1), hms\n(&gt;= 0.4.1), knitr (&gt;= 1.19), Lahman (&gt;= 3.0-1), lubridate,\nMASS, mgcv (&gt;= 1.8.23), microbenchmark (&gt;= 1.4.4), nycflights13\n(&gt;= 0.2.2), rmarkdown (&gt;= 1.8), RMySQL (&gt;= 0.10.13),\nRPostgreSQL (&gt;= 0.6.2), RSQLite (&gt;= 2.0), testthat (&gt;= 2.0.0),\nwithr (&gt;= 2.1.1)","knitr, rmarkdown","jsonlite (&gt;= 0.9.16), knitr (&gt;= 1.8), rmarkdown, shiny (&gt;=\n0.12.1)","cluster, mlbench, nnet, randomForest, rpart, SparseM, xtable,\nMatrix, MASS","AnnotationDbi, jsonlite, org.Hs.eg.db, readr, rhdf5, splines","R2WinBUGS, ellipse, SuppDists, numDeriv, testthat, rgl","testthat, lattice, ggplot2","knitr, BiocStyle",null,"unitizer, knitr, rmarkdown",null,"testthat, knitr, rmarkdown, reactome.db, AnnotationDbi,\nparallel, org.Mm.eg.db, limma, GEOquery","actuar, codetools, diptest, ellipse, gclus, glmnet, lme4 (&gt;=\n1.0), MASS, mgcv (&gt;= 1.8-0), mlbench, mlogit, multcomp,\nmvtnorm, SuppDists, survival","chemometrics, mvtnorm","covr, ggplot2, testthat","randomForest","codetools, shiny, testit, rmarkdown, knitr",null,"tclust, pdfCluster","testthat, covr, pillar (&gt;= 1.0.0), crayon, withr","testthat, jsonlite",null,"knitr, pdp, RUnit, splines, viridis","RUnit","class, hgu95av2.db, tkWidgets, ALL, ROC, DESeq, pasilla,\nRColorBrewer, BiocStyle, knitr","Rgraphviz, fibroEset, hgu95av2.db, hu6800.db, hgu133a.db","covr, pkgload, testthat, tibble","GenomicRanges, Rsamtools, GenomicAlignments, BSgenome,\nGenomicFeatures, BSgenome.Scerevisiae.UCSC.sacCer2,\nBSgenome.Celegans.UCSC.ce2, BSgenome.Hsapiens.NCBI.GRCh38,\nTxDb.Dmelanogaster.UCSC.dm3.ensGene, RUnit, BiocStyle, knitr",null,"Matrix, Biobase, AnnotationDbi, annotate, Biostrings (&gt;=\n2.25.3), SummarizedExperiment (&gt;= 0.1.5), Rsamtools (&gt;=\n1.13.53), GenomicAlignments, rtracklayer, BSgenome,\nGenomicFeatures, Gviz, VariantAnnotation, AnnotationHub,\nDESeq2, DEXSeq, edgeR, KEGGgraph, RNAseqData.HNRNPC.bam.chr14,\npasillaBamSubset, KEGG.db, hgu95av2.db, hgu95av2probe,\nBSgenome.Scerevisiae.UCSC.sacCer2, BSgenome.Hsapiens.UCSC.hg19,\nBSgenome.Mmusculus.UCSC.mm10,\nTxDb.Athaliana.BioMart.plantsmart22,\nTxDb.Dmelanogaster.UCSC.dm3.ensGene,\nTxDb.Hsapiens.UCSC.hg19.knownGene,\nTxDb.Mmusculus.UCSC.mm10.knownGene, RUnit, digest, knitr,\nBiocStyle","testthat, rgl, R.matlab, tripack","testthat (&gt;= 1.0.0), knitr, markdown","broom (&gt;= 0.4.0), chemometrics, geosphere (&gt;= 1.5-1), igraph\n(&gt;= 1.0.1), intergraph (&gt;= 2.0-2), maps (&gt;= 3.1.0), mapproj,\nnetwork (&gt;= 1.12.0), scagnostics, scales (&gt;= 0.4.0), sna (&gt;=\n2.3-2), survival, packagedocs (&gt;= 0.4.0), rmarkdown, roxygen2,\ntestthat, crosstalk","gridExtra","covr, dplyr, ggplot2movies, hexbin, Hmisc, lattice, mapproj,\nmaps, maptools, multcomp, munsell, nlme, testthat (&gt;= 0.11.0),\nvdiffr, quantreg, knitr, rgeos, rpart, rmarkdown, sf (&gt;=\n0.3-4), svglite (&gt;= 1.2.0.9001)","knitr, rmarkdown, testthat, gridExtra, devtools, prettydoc","MASS, mgcv, lubridate, testthat (&gt;= 0.8.1), knitr (&gt;= 1.6),\nrmarkdown","covr, pingr, testthat","getPass","BiocStyle, IRanges, GenomicRanges, SummarizedExperiment,\nDESeq2, limma, testthat, knitr, rmarkdown, pryr","testthat (&gt;= 1.0.0), knitr, markdown","testthat, covr, magrittr, crayon, knitr, rmarkdown, DBI,\nRSQLite, R.utils, forcats, microbenchmark, rprintf, stringr,\nggplot2, dplyr, withr","mvtnorm, car, stabledist, MASS, timeDate, timeSeries","DBI","AnnotationHub, BiocManager, clusterProfiler, DOSE, knitr,\norg.Hs.eg.db, prettydoc, testthat","testthat, knitr, rmarkdown","grid, MASS","SparseM (&gt;= 0.36), XML, RBGL, RUnit, cluster","lattice","ggplot2, egg, lattice, knitr, testthat","gridSVG, rsvg","boot, chron, grid, lattice, quantreg, svUnit, tcltk, zoo","testthat, covr",null,"covr, fs, knitr, rmarkdown, testthat","h5vcData, SummarizedExperiment (&gt;= 1.9.6), GenomicRanges,\nExperimentHub, BiocParallel, BiocStyle","marray, affy, Biobase, limma",null,"knitr, testit","chron, rms, mice, tables, knitr, ff, ffbase, plotly (&gt;=\n4.5.6)","crayon, lubridate, pillar (&gt;= 1.1.0), testthat","org.Hs.eg.db, GO.db, knitr, BiocStyle, testthat","testthat, XML, xtable, ztable, Hmisc, reshape, rmarkdown,\npander, chron, lubridate, tibble, tidyr (&gt;= 0.7.2), dplyr (&gt;=\n0.7.4)","markdown, testthat","knitr (&gt;= 1.8)","testthat, callr","covr, httpuv, jpeg, knitr, png, readr, rmarkdown, testthat\n(&gt;= 0.8.0), xml2",null,"ape, digest, graph, igraphdata, rgl, scales, stats4, tcltk,\ntestthat",null,null,"testthat","knitr","mvtnorm, mlbench, TH.data","XVector, GenomicRanges, Rsamtools, GenomicAlignments,\nGenomicFeatures, BSgenome.Celegans.UCSC.ce2, pasillaBamSubset,\nRUnit, BiocStyle","RUnit, foreach",null,"httr, curl, plyr, testthat, knitr, rmarkdown, R.rsp, sp",null,"knitr, rmarkdown","formatR, testit, digest, rgl (&gt;= 0.95.1201), codetools,\nrmarkdown, htmlwidgets (&gt;= 0.7), webshot, tikzDevice (&gt;= 0.10),\ntinytex, reticulate (&gt;= 1.4), JuliaCall (&gt;= 0.11.1), png, jpeg,\ngifski, xml2 (&gt;= 1.2.0), httr, DBI (&gt;= 0.4-1), showtext,\ntibble, styler",null,"testit","KernSmooth","knitr, rmarkdown, testthat","maps, mapproj, deldir, tripack, zoo, MASS, quantreg, mgcv","KernSmooth, Matrix, Rgraphviz, ascii, data.table, ellipse,\nfields, foreach, geepack, gof (&gt;= 0.9), graph, igraph (&gt;= 0.6),\nlava.tobit (&gt;= 0.4.7), lme4, mets (&gt;= 1.1), nlme, optimx,\npolycor, quantreg, rgl, testthat (&gt;= 0.11), visNetwork, zoo","knitr, rmarkdown (&gt;= 0.2.65), testthat, covr","affy, AnnotationDbi, BiasedUrn, Biobase, ellipse, GO.db,\ngplots, illuminaio, locfit, MASS, org.Hs.eg.db, splines,\nstatmod (&gt;= 1.2.2), vsn","akima, gam",null,"testthat, knitr, covr",null,"testthat, knitr","knitr, testthat (&gt;= 0.8)","dygraphs, leaflet, plotly, xts, rmarkdown, testthat, covr","knitr, RCurl","xcms, caTools","base64enc, ggplot2, knitr, microbenchmark, R.devices, R.rsp",null,"knitr (&gt;= 1.12), rmarkdown (&gt;= 0.9), mix (&gt;= 1.0), geometry\n(&gt;= 0.3-6), MASS","testthat, aws.s3, httr, covr",null,"ggplot2, multcomp",null,null,null,"lattice","class, e1071, ipred, randomForest, gpls, pamr, nnet, ALL,\nhgu95av2.db, som, hu6800.db, lattice, caret (&gt;= 5.07),\ngolubEsets, ada, keggorthology, kernlab, mboost, party",null,"testthat","compiler, covr, ggplot2, testthat",null,"xcms, mzR, MSnbase",null,null,null,"testthat, pryr, gridExtra, microbenchmark, zoo, knitr (&gt;=\n1.1.0), rols, Rdisop, pRoloc, pRolocdata (&gt;= 1.7.1), msdata (&gt;=\n0.19.3), roxygen2, rgl, rpx, AnnotationHub, BiocStyle (&gt;=\n2.5.19), rmarkdown, imputeLCMD, norm, gplots, shiny, magrittr,\nSummarizedExperiment","BiocStyle, msmsTests, ggplot2, RUnit, BiocGenerics","BiocStyle, knitr, MSstats","snow","ggplot2, testthat",null,"knitr, testthat","msdata (&gt;= 0.15.1), RUnit, mzID, BiocStyle (&gt;= 2.5.19),\nknitr, XML",null,"sna, statnet.common, testthat","knitr, rmarkdown, testthat","fastICA, doMPI, bigmemory (&gt;= 4.2), synchronicity(&gt;= 1.3.2),\ncorpcor, xtable, devtools, knitr, bibtex, RUnit, mail, Biobase",null,null,"testthat, devEMF, knitr,tibble,ggplot2, rmarkdown","testthat, digest, knitr, rmarkdown, jsonlite, jose","DBI, annotate, RUnit","DBI, annotate, RUnit","testthat (&gt;= 0.7), devtools, httr, knitr, rmarkdown",null,"matrixStats, lattice, ggplot2","knitr (&gt;= 1.19), lubridate (&gt;= 1.7.4), testthat (&gt;= 2.0.0),\nwithr (&gt;= 2.1.2)","Rcpp, testthat, covr","covr, testthat, disposables (&gt;= 1.0.3)","covr, devtools, htmlwidgets, jsonlite, knitr, leaflet,\ntestthat, rticles","bitops, covr, Rcpp, testthat","devtools (&gt;= 0.8), roxygen2, RUnit, testthat, knitr,\nrmarkdown, yaml, Biobase",null,"Rcpp","MASS, maps, ggthemes, GGally, testthat, knitr, devtools,\nshiny (&gt;= 1.1.0), curl, rmarkdown, Rserve, RSclient, Cairo,\nbroom, webshot, listviewer, dendextend, sf, maptools, rgeos,\npng, IRdisplay, processx, plotlyGeoAssets, forcats","MASS, parallel, Rmpi, testthat, RUnit","abind, testthat, tcltk, foreach, doParallel, itertools,\niterators, covr",null,null,"spdep, maptools, foreign, mvtnorm","testthat",null,"testthat","callr, covr, crayon, debugme, parallel, testthat, withr",null,"Rcpp, testthat, withr","testthat, rmarkdown, pRolocdata (&gt;= 1.9.4), roxygen2,\nsynapter, xtable, rgl, BiocStyle (&gt;= 2.5.19), hpar (&gt;= 1.15.3),\ndplyr, GO.db, AnnotationDbi, Rtsne (&gt;= 0.13), nipals","pRoloc (&gt;= 1.13.8), testthat","pRolocdata, knitr, BiocStyle (&gt;= 2.5.19), rmarkdown","testthat, future, knitr, rmarkdown",null,"testthat, covr","lattice, RUnit, xtable","cba","testthat (&gt;= 0.8.0)","callr, covr, curl, pingr, processx (&gt;= 3.1.0), R6, rlang,\ntestthat, tibble",null,"covr, dplyr (&gt;= 0.7.5), knitr, rmarkdown, testthat","knitr",null,null,"tools","digest (&gt;= 0.6.10)","knitr, microbenchmark, pryr, testthat, ggplot2, scales","RColorBrewer, MASS","testthat","Rgraphviz, XML, RUnit, BiocGenerics","covr, knitr, mockery, rmarkdown, testthat",null,"RUnit, inline, rbenchmark, knitr, rmarkdown, pinp, pkgKitten\n(&gt;= 0.1.2)","RUnit, Matrix, pkgKitten, reticulate, rmarkdown, knitr, pinp,\nslam","RUnit, inline, knitr, rmarkdown, pinp","inline, RUnit, pkgKitten, microbenchmark","knitr, rmarkdown, prettydoc, mvtnorm, RcppEigen","zoo, testthat","Rcompression, XML",null,"RUnit","curl, testthat, knitr, rmarkdown, stringi, covr, spelling","covr, knitr, rmarkdown, rprojroot (&gt;= 1.1), testthat","covr, ddalpha, dimRed (&gt;= 0.2.2), fastICA, ggplot2, igraph,\nkernlab, knitr, NMF, pls, RANN, rmarkdown, rpart, rsample,\nRSpectra, testthat","testthat, roxygen2",null,"covr, testthat","covr, testthat","curl, covr, git2r (&gt;= 0.23.0), knitr, mockery, pkgbuild (&gt;=\n1.0.1), pingr, rmarkdown, testthat, withr","covr, devtools, fortunes, knitr, miniUI, rprojroot,\nrstudioapi, shiny, styler (&gt;= 1.0.2), testthat (&gt;= 2.0.0)",null,"covr, lattice, testthat (&gt;= 0.8.0)","AnnotationDbi, rpx, DT, knitr, rmarkdown, BiocStyle, mzR,\nxcms, msdata, isobar, MALDIquant (&gt;= 1.12), MALDIquantForeign,\nreadBrukerFlexData, rTANDEM, synapter, synapterdata, IPPD,\nRdisop, OrgMassSpecR, SummarizedExperiment, BRAIN, rols, hpar,\nGO.db, org.Hs.eg.db, e1071, biomaRt, RColorBrewer, ggplot2,\nreshape2, xtable, lattice, mzID, pRoloc, pRolocdata, MSGFplus,\nMSGFgui, MSnID, msmsTests, msmsEDA, DEP, corrplot, beanplot,\nHeatplus, gplots, VennDiagram, protViz, genefilter, plotly,\ngridExtra, dplyr, lubridate","MASS, rmarkdown, deldir, orientlib, lattice, misc3d,\nrstudioapi, magick","bit64, BiocStyle, knitr, rmarkdown, testthat","BiocStyle, knitr, rmarkdown","knitr,\nrmarkdown",null,null,"crayon, magrittr, methods, pillar, testthat (&gt;= 2.0.0), covr","shiny (&gt;= 0.11), tufte, testthat, digest, dygraphs, tibble,\ncallr (&gt;= 2.0.0)","RUnit, testthat","grid, MASS, lattice, boot, cluster, Matrix, robust,\nfit.models, MPV, xtable, ggplot2, GGally, RColorBrewer,\nreshape2, sfsmisc, catdata, doParallel, foreach, skewt","GO.db, knitr (&gt;= 1.1.0), BiocStyle (&gt;= 2.5.19), testthat,\nlubridate, DT, rmarkdown,","covr, devtools, knitr, rmarkdown, testthat (&gt;= 0.8.0)","testthat, mockr, knitr, withr, rmarkdown","MSnbase, Biostrings, BiocStyle, testthat, knitr","knitr, testthat, rmarkdown (&gt;= 1.1), plumber (&gt;= 0.3.2),\nshiny, sourcetools, xtable","DBItest, knitr, rmarkdown, testthat","testthat, knitr, rmarkdown","spelling, svglite, png, jpeg, webp, ggplot2","irlba, testthat","XML (&gt;= 3.1.0)","testthat, knitr, png, stringi (&gt;= 0.3.1), rmarkdown, covr","testthat, BiocStyle, knitr, rmarkdown","IRanges, GenomicRanges, SummarizedExperiment, Matrix,\nDelayedArray, ShortRead, graph, data.table, RUnit, BiocStyle",null,"AER, car, geepack, lattice, lmtest, MASS, multiwayvcov,\nparallel, pcse, plm, pscl, scatterplot3d, stats4, strucchange,\nsurvival","dichromat, bit64, covr, hms, testthat (&gt;= 2.0)","BiocStyle, biomaRt, beachmat, cowplot, cluster, destiny,\nknitr, monocle, mvoutlier, rmarkdown, Rtsne, umap, testthat,\nmagrittr, pheatmap, irlba, tximport, tximportData, tximeta,\nBiobase, limma",null,"testthat, XML, xml2","tools, later, rstudioapi, knitr (&gt;= 1.9), rmarkdown","callr, covr, mockery, testthat","datasets, tcltk, cluster, lattice, MASS, Matrix, nlme, lokern",null,"datasets, Cairo (&gt;= 1.5-5), testthat, knitr (&gt;= 1.6),\nmarkdown, rmarkdown, ggplot2, magrittr",null,null,"testthat, BiocStyle, knitr, rmarkdown, scRNAseq, magrittr,\nRtsne","Rmpi,rlecuyer,nws","testthat",null,"RH2, RMySQL, RPostgreSQL, svUnit, tcltk, MASS","setRNG",null,"covr, htmltools, htmlwidgets, knitr, rmarkdown, testthat","BiocStyle, knitr, rmarkdown, testthat","annotate, AnnotationDbi, hgu95av2.db, GenomicFeatures,\nTxDb.Hsapiens.UCSC.hg19.knownGene, BiocStyle, knitr, rmarkdown,\ndigest, jsonlite, rhdf5, HDF5Array (&gt;= 1.7.5), airway, RUnit,\ntestthat","synapterdata (&gt;= 1.13.2), xtable, testthat (&gt;= 0.8), BRAIN,\nBiocStyle",null,"covr, devtools, knitr, rmarkdown, usethis, xml2","maps","covr, dplyr, import, knitr (&gt;= 1.5.32), microbenchmark,\nmockr, nycflights13, rmarkdown, testthat, withr","knitr, BiocStyle, biobroom","covr, gapminder, knitr, rmarkdown, testthat","covr, dplyr, testthat","feather (&gt;= 0.3.1), knitr (&gt;= 1.17), rmarkdown (&gt;= 1.7.4)","date, RUnit","testit, rstudioapi","lattice","ALL, hgu95av2.db, hgu133a.db, genefilter, xtable, multtest,\nRgraphviz, globaltest","fpc",null,"knitr","covr, knitr, magick, rmarkdown, roxygen2, spelling (&gt;= 1.2),\nstyler (&gt;= 1.0.2), testthat (&gt;= 2.0.0), withr","knitr, rmarkdown, testthat",null,"xtable","testthat, beeswarm, lattice, ggplot2, beanplot, vioplot,\nggbeeswarm,","hexbin (&gt;= 1.27.0), scales, MASS, knitr, dichromat,\ncolorspace, rasterVis, httr, mapproj, vdiffr, svglite (&gt;=\n1.2.0), testthat, covr, rmarkdown, rgdal","hexbin (&gt;= 1.27.0), ggplot2 (&gt;= 1.0.1), testthat, covr","knitr, rmarkdown, webshot, igraph, rpart, shiny,\nshinyWidgets, colourpicker, sparkline, ggraph, flashClust","affydata, hgu95av2cdf, BiocStyle, knitr, dplyr, testthat","fftw","httpuv, knitr, rmarkdown, shiny","markdown","covr, mockery, testthat","testthat, covr, lattice, DBI, RSQLite, methods, knitr,\nrmarkdown","rstudioapi, testit","BiocStyle, caTools, knitr (&gt;= 1.1.0), faahKO, msdata, ncdf4,\nrgl, microbenchmark, testthat, pander, magrittr, MALDIquant","testit, parallel, rstudioapi, tinytex, mime, markdown, knitr,\nhtmltools, base64enc, rmarkdown","bitops, RCurl","testthat, curl, covr, knitr, rmarkdown, magrittr, httr","ps, testthat","knitr, spdep, splm, sphet, plm, zoo, survival","Biostrings, drosophila2probe, RUnit","RUnit","testthat, knitr, rmarkdown, purrr, magrittr","covr, testthat, withr",null,"coda, chron, DAAG, fts, ggplot2, mondate, scales,\nstrucchange, timeDate, timeSeries, tis, tseries, xts","methods","MASS, survival",null,"MASS, Matrix",null,null,null,null,null,"KernSmooth","lattice","MASS","KernSmooth, MASS, latticeExtra","lattice, nlme, nnet, survival","expm, MASS","codetools","splines, parallel, survival, MASS","Hmisc, MASS","MASS","methods",null,"survival","MASS","Matrix, methods","MASS, Matrix, SuppDists, methods, stats4",null,null,null,"codetools, methods, xml2, curl","methods, xml2, commonmark"],[null,null,null,null,null,null,null,"AnnotationHubData",null,null,null,"png",null,null,null,null,null,"biganalytics, bigtabulate","bigmemory, synchronicity",null,null,null,null,null,null,null,null,null,null,null,null,null,"Rmpi",null,null,null,null,null,null,null,null,null,null,null,null,null,"zoo",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"ggdendro, dendroextras, Hmisc, data.table, rpart","robustbase",null,null,null,null,null,null,null,null,"compiler",null,null,null,null,null,null,null,null,"ExperimentHubData",null,null,null,null,null,null,null,"compiler, doMC, RUnit, doParallel",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"sp",null,null,null,null,null,null,null,null,null,null,null,null,"Rgraphviz",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"knitr","shiny (&gt;= 1.1)",null,null,null,null,null,null,null,"rstudioapi",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"chron, fts, timeSeries, timeDate, tis, tseries, xts, zoo",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"parallel",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"gmp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"cleaver",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"mgcv, rpart, nor1mix, polycor, sm, tikzDevice",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"knitr, shiny",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rgraphviz, Rmpi, XML",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"chron",null,"MatrixModels, graph, SparseM, sfsmisc",null,null,null,null,"snow, nws, Rmpi",null,null,null,null,null,null,null,null,null,null],["LGPL (&gt;= 2)","MIT + file LICENSE","LGPL (&gt;= 2.0)","LGPL (&gt;= 2)","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL-2","GPL-3","GPL-2","GPL-2 | GPL-3","GPL","GPL-3","Artistic-2.0","BSL-1.0","GPL (&gt;= 2)","LGPL-3 | Apache License 2.0","LGPL-3 | Apache License 2.0","MIT + file LICENSE","MIT + file LICENSE","Artistic-2.0","LGPL","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL-2 | GPL-3","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL-2","GPL-2","GPL (&gt;= 2)","GPL-3","GPL-3","GPL-2","GPL-2","MIT + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2)","GPL-3","MIT + file LICENSE","BSD_3_clause + file LICENSE","GPL-2","MIT + file LICENSE","GPL (&gt;= 3)","MIT + file LICENSE","GPL-3","MIT + file LICENSE","GPL (&gt;= 2)","BSD_3_clause + file LICENSE","BSD_2_clause + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2.0)","MPL-2.0 | file LICENSE","LGPL (&gt;= 2)","MIT + file LICENSE","GPL-2","Artistic-2.0","MIT + file LICENSE","GPL-2 | GPL-3","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 3)","LGPL (&gt;= 3)","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-3 | file LICENSE","GPL (&gt;= 2)","Artistic-2.0","GPL-2","Artistic-2.0","MIT + file LICENSE","GPL-3 | file LICENSE","GPL-3 | file LICENSE","GPL-2","GPL (&gt;=2)","GPL","MIT + file LICENSE","Artistic-2.0","LGPL","GPL (&gt;= 2)","GPL-2","MIT + file LICENCE","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-3","Apache License (== 2.0)","GPL","GPL-2 | GPL-3","GPL","GPL-3","LGPL-3","LGPL-3","GPL (&gt;= 2) | file LICENSE","GPL-2","Artistic-2.0","Artistic-2.0","GPL-2","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL (&gt;= 3) + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2.0)","GPL (&gt;= 2)","GPL-2 | file LICENSE","GPL-3 | file LICENSE","GPL-2 | file LICENSE","MIT + file LICENSE","GPL-2","GPL-3 | file LICENSE","MIT + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2)","Artistic-2.0","Artistic-2.0","GPL-3","GPL-2","Artistic-2.0","GPL","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-2","GPL-2","MIT + file LICENSE","Artistic-2.0","GPL-2","GPL (&gt;= 3)","GPL","GPL (&gt;= 2)","GPL-3","Artistic-2.0","GPL (&gt;= 3)","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 2) | file LICENSE","MIT + file LICENSE","LGPL-2.1","GPL (&gt;= 2)","GPL-2","GPL (&gt;= 2)","GPL-3","Artistic-2.0","GPL (&gt;= 2)","Artistic-2.0","Apache License (== 2.0)","GPL-2 | GPL-3","MIT + file LICENSE","GPL-2","GPL-3","GPL","MIT + file LICENSE | Unlimited","LGPL-3","MIT + file LICENSE","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-3","GPL-3","GPL (&gt;=2)","GPL (&gt;= 2)","LGPL-2","GPL (&gt;= 2)","GPL-2","MIT + file LICENSE","GPL (&gt;= 3)","GPL (&gt;= 2) | file LICENSE","GPL-2","LGPL (&gt;= 2)","Artistic-2.0","GPL (&gt;= 2)","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 2)","BSD_2_clause + file LICENSE","GPL","GPL-3","GPL (&gt;= 2)","GPL-2","LGPL","GPL-2 | GPL-3","GPL (&gt;= 2)","GPL-3","GPL-2","GPL (&gt;= 2)","GPL-3","GPL-2","GPL-2","Artistic-2.0","Artistic-2.0","Artistic-2.0","LGPL","MIT + file LICENSE","GPL-2","GPL (&gt;= 2)","Artistic-2.0","GPL (&gt;= 3)","GPL (&gt;= 2)","GPL-3","GPL (&gt;= 2)","file LICENSE","GPL-2","GPL-3","MIT + file LICENSE","Artistic-2.0","Artistic-2.0","GPL-2","MIT + file LICENSE","GPL (&gt;= 3)","GPL-3","GPL-3","MIT + file LICENSE","MIT + file LICENSE","GPL-3","GPL (&gt;= 2)","GPL-2 | GPL-3 | file LICENSE","MIT + file LICENSE","MIT + file LICENSE","GPL-2","MIT + file LICENSE","GPL-2 | GPL-3","GPL-2","GPL","MIT + file LICENSE","LGPL (&gt;= 2)","MIT + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","GPL-2","GPL-2","GPL-2","MIT + file LICENSE","Artistic-2.0","GPL-2","GPL-3","GPL-2","GPL-2","BSD_3_clause + file LICENSE","GPL (&gt;= 2)","GPL-3 | file LICENSE","LGPL","LGPL (&gt;= 2.1)","LGPL (&gt;= 2.1)","LGPL (&gt;= 2.1)","LGPL (&gt;= 2.1)","MIT + file LICENSE","GPL (&gt;= 2)","GPL (&gt;= 3)","Artistic-2.0","MIT + file LICENSE","Apache License 2.0","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 2) | file LICENSE","GPL (&gt;= 2)","GPL (&gt;= 2)","BSD","GPL (&gt;= 2)","GPL-2","GPL (&gt;= 2) | file LICENSE","GPL-3","GPL-2","MIT + file LICENSE","GPL-2","MIT + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","Artistic-2.0","GPL","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL-2","BSD_3_clause + file LICENSE","GPL-3","GPL-3","GPL-3","GPL (&gt;= 2)","GPL-2","GPL (&gt;= 2)","GPL-3","GPL-2","GPL-2","LGPL (&gt;= 2)","MIT + file LICENSE","MIT + file LICENSE","file LICENSE","GPL-2","GPL-3","MIT + file LICENSE","Artistic-2.0","GPL (&gt;= 2)","GPL-2 | GPL-3","MIT + file LICENSE","GPL-3","GPL","BSD_3_clause + file LICENCE","GPL","GPL-2","GPL (&gt;= 2)","GPL (&gt;= 3)","GPL-3 | file LICENSE","GPL-3","GPL (&gt;= 2) | file LICENSE","GPL-3","GPL","MIT + file LICENSE","GPL (&gt;= 2)","GPL-2","GPL (&gt;= 2)","file LICENSE","GPL-2 | file LICENSE","GPL-2","Artistic-2.0","GPL-2","GPL-2","MIT + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","Artistic-2.0","MIT + file LICENSE","GPL-3","GPL-3 | file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 2)","LGPL","GPL","GPL-3","MIT + file LICENSE","GPL-3","Apache License (== 2.0) | file LICENSE","MIT + file LICENSE","GPL","GPL (&gt;= 2)","MIT + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","Artistic-2.0","BSD_3_clause + file LICENSE","GPL-2","GPL-3","MIT + file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 2) + file LICENSE","MIT + file LICENSE","BSD_2_clause + file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 2)","Artistic-2.0","BSD_3_clause + file LICENSE","MIT + file LICENSE","CC0","Artistic-2.0 + file LICENSE","GPL-2 | GPL-3","Part of R 3.5.1","Unlimited","GPL-2 | GPL-3","GPL (&gt;= 2)","GPL","Part of R 3.5.1","Part of R 3.5.1","GPL (&gt;= 2)","Part of R 3.5.1","Part of R 3.5.1","Part of R 3.5.1","Unlimited","GPL (&gt;= 2)","GPL-2 | GPL-3","GPL (&gt;= 2) | file LICENCE","Part of R 3.5.1","GPL (&gt;= 2)","GPL (&gt;= 2) | file LICENCE","GPL-2 | GPL-3","Part of R 3.5.1","GPL-2","GPL-2 | GPL-3","GPL-2 | GPL-3","Part of R 3.5.1","Part of R 3.5.1","Part of R 3.5.1","LGPL (&gt;= 2)","Part of R 3.5.1","Part of R 3.5.1","Part of R 3.5.1"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"yes",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"yes",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"no",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["no","yes","yes","yes","no","no","no","yes","yes","no","yes","yes","no","yes","no","no","yes","yes","no","no","yes","yes","no","no","no","no","yes","no","no","no","no","no","yes","yes","yes","yes","no","no","no",null,"no","no","yes","yes","no","yes","yes","no","no","no","no","no","no","yes","yes","no","no","no","yes","no","yes","no","no","yes","yes","no","no","no","no","yes","yes","no","yes","yes","yes","no","no","no","yes","no","no","yes","yes","no","no","no","no","yes","yes","yes","no","yes","no","no","no","no","no","yes","no","no","yes","no","yes","no","no","no","no","yes","yes","no","no","no","no","yes","no","no","yes","no","no","yes","no","no","yes","yes","no","yes","no","no","no","no","no","yes","yes","no","yes","yes","no","yes","no","no","no","yes","no","yes","no","no","yes","yes","no","no","no","yes","yes","no","yes","yes","yes","no","no","no","no","no","yes","no","no","yes","yes","yes","yes","yes","no","no","yes","no","yes","yes","yes",null,"yes","no","no","yes","yes","no","yes",null,"no","yes","yes","no","no","no","no","no","no","yes","no","no","yes","no","yes","no","yes","yes","yes","no","yes","yes","no","yes","yes","no","no","no","no","yes","no","no","no","no","yes","no","yes","no","no","no","yes","yes","yes","no","no","yes","no","yes","yes","no","yes","no","no","yes","no","no","yes","yes","yes","yes",null,"yes","no","no","no","no","no","no","yes","yes","yes","no","no","yes","yes","yes","yes","yes","yes","yes","no","yes","yes","yes","no","no","no","no","no","no","no","yes","yes","no","yes","yes","yes",null,"yes","yes","yes","no","no","yes","no","yes","no","no","no","yes","no","yes","yes","no","no","no","yes","yes","no","yes","yes","no","no","no","no","no","no","no","no","no","no","no","yes","yes","no","no","yes","no","yes","no","no","no","yes","no","yes","no","yes","yes","no","no","no","yes","no","no","no","no","no","yes","yes","no","no","no","no","no","yes","yes","no","no","no","no","no","yes","no","yes","yes","no","no","yes","yes","no","yes","yes","yes",null,"no","yes","yes","no",null,null,"yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes",null,"yes","yes","yes","yes"],["3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>LibPath<\/th>\n      <th>Version<\/th>\n      <th>Priority<\/th>\n      <th>Depends<\/th>\n      <th>Imports<\/th>\n      <th>LinkingTo<\/th>\n      <th>Suggests<\/th>\n      <th>Enhances<\/th>\n      <th>License<\/th>\n      <th>License_is_FOSS<\/th>\n      <th>License_restricts_use<\/th>\n      <th>OS_type<\/th>\n      <th>MD5sum<\/th>\n      <th>NeedsCompilation<\/th>\n      <th>Built<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>It is always useful to finish a Rmarkdown report with a section
providing all the session information details with the
<code>sessionInfo()</code> function, such at the end of this material. This
allows readers to review the version of R itself and all the
packages that were used to produce the report.</li>
</ul>
<blockquote>
<p><strong>Challenge</strong></p>
<p>Prepare an Rmarkdown report summarising the portal ecology data. The
report should include a <em>Material and methods</em> section where the
data is read in (ideally from the online file) and briefly
described, a <em>Data preparation</em> section where rows with missing
values are filtered out, and a <em>Visualisation</em> section where one or
two plots are rendered. Finish your report with a <em>Session
information</em> section.</p>
</blockquote>
</div>
<div id="docker-containers" class="section level2">
<h2><span class="header-section-number">7.4</span> Docker containers</h2>

</div>
</div>
<p style="text-align: center;">
<a href="joining-tables.html"><button class="btn btn-default">Previous</button></a>
<a href="sec-bioinfo.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2019-01-07
</p>
</div>
</div>



</body>
</html>
