<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta property="og:title" content="Chapter 7 Reproducible research | Introduction to bioinformatics" />
<meta property="og:type" content="book" />


<meta property="og:description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2019-02-06" />


<meta name="description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain.">

<title>Chapter 7 Reproducible research | Introduction to bioinformatics</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/jquery-1.12.4/jquery.min.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.5/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>



</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Introduction to bioinformatics<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html">Preamble</a>
<a href="sec-dataorg.html"><span class="toc-section-number">1</span> Data organisation with Spreadsheets</a>
<a href="sec-startr.html"><span class="toc-section-number">2</span> Starting with R</a>
<a href="sec-startdata.html"><span class="toc-section-number">3</span> Starting with data</a>
<a href="sec-dplyr.html"><span class="toc-section-number">4</span> Manipulating and analyzing data with dplyr</a>
<a href="sec-vis.html"><span class="toc-section-number">5</span> Data visualization</a>
<a href="sec-join.html"><span class="toc-section-number">6</span> Joining tables</a>
<a id="active-page" href="sec-rr.html"><span class="toc-section-number">7</span> Reproducible research</a><ul class="toc-sections">
<li class="toc"><a href="#knitr-and-rmarkdown"> <code>knitr</code> and <code>rmarkdown</code></a></li>
<li class="toc"><a href="#additional-features"> Additional features</a></li>
<li class="toc"><a href="#docker-containers"> Docker containers</a></li>
</ul>
<a href="sec-bioinfo.html"><span class="toc-section-number">8</span> Bioinformatics</a>
<a href="sec-prog.html"><span class="toc-section-number">9</span> Additional programming concepts</a>
<a href="sec-ccl.html"><span class="toc-section-number">10</span> Conclusions</a>
<a href="session-information.html">Session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="sec:rr" class="section level1">
<h1><span class="header-section-number">Chapter 7</span> Reproducible research</h1>
<p><strong>Learning Objectives</strong></p>
<ul>
<li>Understand the concept of reproducible research and reproducible
documents.</li>
<li>Undertand the process by which a source document in compiled into a
final report.</li>
<li>Generate a reproducible report in html or pdf from an Rmarkdown
document using RStudio.</li>
</ul>
<p>For a general introduction on the topic in French, see <span class="citation">Pouzat, Davison, and Hinsen (<label for="tufte-mn-20" class="margin-toggle">&#8853;</label><input type="checkbox" id="tufte-mn-20" class="margin-toggle">2015<span class="marginnote">Pouzat, Christophe, Andrew Davison, and Konrad Hinsen. 2015. “La Recherche Reproductible : Une Communication Scientifique Explicite.” <em>Statistique et Société</em> 3 (1). <a href="http://www.publications-sfds.fr/index.php/stat_soc/article/view/448">http://www.publications-sfds.fr/index.php/stat_soc/article/view/448</a>.</span>)</span>.</p>
<p>Reproducible research refers to research that can be reproduced under
various conditions and by different people. It applies to every area
of research, both experimental and computational, but is often (but
not always) easier to implement for computational work. The different
levels of reproducibility can formalised<label for="tufte-sn-6" class="margin-toggle sidenote-number">6</label><input type="checkbox" id="tufte-sn-6" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">6</span> The nomenclature is taken from <a href="https://lgatto.github.io/rr-what-should-be-our-goals/">this blog post</a> that provides links and highlights that confusion that exists around the terms and concepts of reproducibility.</span> as follows:</p>
<ul>
<li><p><strong>Repeat</strong> my experiment, i.e. obtain the same tables/graphs/results
using the same setup (data, software, …) in the same lab or on the
same computer. That’s basically re-running one of my analysis some
time after I original developed it.</p></li>
<li><p><strong>Reproduce</strong> an experiment (not ones own), i.e. obtain the same
tables/graphs/results in a different lab or on a different computer,
using the same setup (the data would be downloaded from a public
repository and the same software, but possibly using a different
version, or a different operation system).</p></li>
<li><p><strong>Replicate</strong> an experiment, i.e. obtain the same (similar enough)
tables/graphs/results in a different set up. The data could still be
downloaded from the public repository, or possibly
re-generated/re-simulated, and the analysis would be re-implemented
based on the original description.</p></li>
<li><p>Finally, <strong>re-use</strong> the information/knowledge from one experiment to
run a different experiment with the aim to <strong>confirm</strong> results from
scratch.</p></li>
</ul>
<p>The table below summerised these concepts focusing on data and code in
computational projects.</p>
<table>
<thead>
<tr class="header">
<th></th>
<th align="center">Same data</th>
<th align="center">Different data</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Same code</strong></td>
<td align="center">Repeat</td>
<td align="center">Reproduce</td>
</tr>
<tr class="even">
<td><strong>Different code</strong></td>
<td align="center">Reproduce</td>
<td align="center">Replicate</td>
</tr>
</tbody>
</table>
<p>There are many reasons to work reproducibly, and <span class="citation">Markowetz (<label for="tufte-mn-21" class="margin-toggle">&#8853;</label><input type="checkbox" id="tufte-mn-21" class="margin-toggle">2015<span class="marginnote">Markowetz, Florian. 2015. “Five Selfish Reasons to Work Reproducibly.” <em>Genome Biology</em> 16 (1): 274. <a href="https://doi.org/10.1186/s13059-015-0850-7">https://doi.org/10.1186/s13059-015-0850-7</a>.</span>)</span>
nicely summarises 5 good reasons. Importantly, he stressed out that
the first beneficiary of reproducible work are the student/research
that apply these principles:</p>
<ol style="list-style-type: decimal">
<li>Reproducibility helps to avoid disaster.</li>
<li>Reproducibility makes it easier to write papers<label for="tufte-sn-7" class="margin-toggle sidenote-number">7</label><input type="checkbox" id="tufte-sn-7" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">7</span> And course reports! The exam of this course will consist in a reproducible report using the tools described below.</span>.</li>
<li>Reproducibility helps reviewers see it your way<label for="tufte-sn-8" class="margin-toggle sidenote-number">8</label><input type="checkbox" id="tufte-sn-8" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">8</span> Not reviewers, also professors that read exams. See previous footnote.</span>.</li>
<li>Reproducibility enables continuity of your work.</li>
<li>Reproducibility helps to build your reputation.</li>
</ol>
<div id="knitr-and-rmarkdown" class="section level2">
<h2><span class="header-section-number">7.1</span> <code>knitr</code> and <code>rmarkdown</code></h2>
<p>Reproducible research is an essential part of any data analysis. With
the tools that are available, one can argue that it has become more
difficult not to produce reproducible reports than to producing then.</p>
<p>Reproducible documents have been a part of R since the very
beginning. See for example <span class="citation">R. Gentleman and Temple Lang (<label for="tufte-mn-22" class="margin-toggle">&#8853;</label><input type="checkbox" id="tufte-mn-22" class="margin-toggle">2004<span class="marginnote">Gentleman, Robert, and Duncan Temple Lang. 2004. “Statistical Analyses and Reproducible Research.” <em>Bioconductor Project Working Papers. Working Paper 2</em>. <a href="https://biostats.bepress.com/bioconductor/paper2">https://biostats.bepress.com/bioconductor/paper2</a>.</span>)</span>, to see how such <em>compendia</em> play a
central role within the <a href="https://bioconductor.org/">Bioconductor</a>
project (more about Bioconductor in it’s dedicated
chapter). Originally, these were written in LaTeX, interleaved with R
code chunks, forming so called Sweave documents (with extension
<code>.Rnw</code>).</p>
<p>More recently, it has become to use the
<a href="https://daringfireball.net/projects/markdown/">markdown</a> syntax
markup language, rather than LaTeX. Once interleaved with R code
chunks, these documents become Rmarkdown files (<code>.Rmd</code>). The can be
converted into markdown using <code>knitr::knit</code>, that executes the code
chunk and incorporates their output in the resulting markdown
documents, which itself is converted to one of many output formats,
typically pdf of html using <a href="http://pandoc.org/">pandoc</a>. In R, this
final conversion is done using <code>rmarkdown::render</code> (that relies on
pandoc). The <a href="http://rmarkdown.rstudio.com/">rmarkdown</a> package is
developed and maintained by RStudio and benefits from excellent
documentation, support and integrates into the RStudio editor.</p>
<p><img src="figs/rmd.png" /></p>
<p>RStudio now also support <em>R Notebook</em> documents<label for="tufte-sn-9" class="margin-toggle sidenote-number">9</label><input type="checkbox" id="tufte-sn-9" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">9</span> See also <a href="https://jupyter.org/">Jupyter notebooks</a>
initially developed for Python, but that can run R code as well.</span> that execute
individual code chunks that are displayed directly in the source
document.</p>
<p><img src="figs/rnb.png" /></p>
</div>
<div id="additional-features" class="section level2">
<h2><span class="header-section-number">7.2</span> Additional features</h2>
<ul>
<li><p>Among the most options that can set for code chunks is
<code>cache</code>. Setting <code>cache=TRUE</code> will avoid that specific code chunk to
be cached and not recomputed every time the documented in <em>knitted</em>,
unless the code chunk was modified. This is an important feature
when long computations are necessary.</p></li>
<li><p>The <code>DT::datatable</code> function allows to create dynamic tables
directly from R, as show below.</p></li>
</ul>
<pre class="sourceCode r"><code class="sourceCode r">ip &lt;-<span class="st"> </span><span class="kw">installed.packages</span>()
DT<span class="op">::</span><span class="kw">datatable</span>(ip)</code></pre>
<div id="htmlwidget-602b47c1ae4e0aa1655d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-602b47c1ae4e0aa1655d">{"x":{"filter":"none","data":[["abind","acepack","ada","addinexamples","additivityTests","ade4","affy","affydata","affyio","affyPLM","AhoCorasickTrie","aLFQ","ALL","amap","animation","annotate","AnnotationDbi","AnnotationFilter","AnnotationForge","AnnotationHub","AnnotationHubData","ape","apercu","askpass","assertthat","backports","base2grob","base64enc","BASiCS","BBmisc","beachmat","beanplot","beeswarm","BH","bibtex","biclust","biglm","bigmemory","bigmemory.sri","bindr","bindrcpp","bio3d","Biobase","biobroom","BiocFileCache","BiocGenerics","BiocInstaller","BiocManager","BiocNeighbors","BiocParallel","biocpkgs","BiocStyle","BiocVersion","biocViews","BiocWorkflowTools","bioDist","biomaRt","Biostrings","biovizBase","bit","bit64","bitops","blob","blogdown","bmp","bookdown","BRAIN","brew","broom","BSgenome","BSgenome.Hsapiens.NCBI.GRCh38","C50","Cairo","callr","car","carData","Cardinal","caret","Category","caTools","cellranger","checkmate","circlize","classInt","cleaver","cli","clipr","clisymbols","clue","clusterProfiler","coda","CoGAPS","coin","colorspace","colourpicker","combinat","commonmark","ComplexHeatmap","conflicted","corrplot","covr","cowplot","cp4p","cputools","crayon","crosstalk","Cubist","curatedTCGAData","curl","CVST","d3heatmap","daff","DAPAR","DAPARdata","datapasta","data.table","data.tree","DBI","dbplyr","ddalpha","debugme","DelayedArray","DelayedMatrixStats","dendextend","dendsort","DEoptimR","DEP","desc","DESeq2","devtools","DiagrammeR","dichromat","digest","dimRed","diptest","distr","DO.db","doMC","doParallel","DOSE","doSNOW","downloader","dplyr","DRR","DT","dynamicTreeCut","e1071","EBImage","edgeR","egg","ellipse","emojifont","enc","enrichplot","EnsDb.Hsapiens.v86","ensembldb","europepmc","evaluate","evd","exampletestr","ExperimentHub","expm","faahKO","factoextra","FactoMineR","fansi","farver","fastcluster","fastmatch","fcuk","fdrtool","ff","ffbase","fftwtools","fgsea","filesstrings","fireData","flashClust","flexclust","flexdashboard","flexmix","FNN","forcats","foreach","formatR","Formula","fpc","fs","futile.logger","futile.options","future","gapminder","gbm","gclus","gcrma","gdata","gdtools","geeM","geepack","genefilter","geneplotter","generics","GenomeInfoDb","GenomeInfoDbData","GenomicAlignments","GenomicFeatures","GenomicRanges","geometry","GEOquery","geosphere","GetoptLong","getPass","GGally","ggbeeswarm","ggbio","ggdendro","ggExtra","ggforce","ggimage","ggiraph","gglabeller","ggmap","ggplot2","ggplotify","ggpubr","ggraph","ggrepel","ggridges","ggsci","ggsignif","ggstance","ggtree","ggvis","gh","git2r","glmnet","GlobalOptions","globals","glue","gmm","GO.db","golubEsets","googlesheets","GOSemSim","gower","gplots","gpls","graph","gridBase","gridExtra","gridGraphics","gridSVG","GSEABase","gsubfn","gtable","gtools","Gviz","gWidgets","haven","HDF5Array","heatmaply","Heatplus","here","hexbin","hexSticker","hgu133plus2cdf","hgu95av2cdf","hgu95av2.db","highcharter","highlight","highr","Hmisc","hms","Homo.sapiens","hpar","htmlTable","htmltools","htmlwidgets","httpuv","httr","hu6800.db","hunspell","hwriter","icon","igraph","iheatmapr","imager","imp4p","impute","imputeLCMD","influenceR","ini","interactiveDisplay","interactiveDisplayBase","inum","IPPD","ipred","IRanges","irlba","Iso","isobar","ISOcodes","iterators","janeaustenr","jpeg","jsonlite","KEGGgraph","keggorthology","KEGGREST","kernlab","kinship2","klaR","kmknn","knitr","labeling","labelled","lambda.r","LaplacesDemon","later","latticeExtra","lava","lazyeval","LCA","leaflet","leaps","libcoin","limma","listenv","lme4","lobstr","locfit","lpSolve","lubridate","magic","magick","magrittr","MALDIquant","MALDIquantForeign","manipulateWidget","mapproj","maps","maptools","MAQCsubsetAFX","markdown","marray","MassSpecWavelet","MatrixModels","matrixStats","matter","mboost","mcclust","mclust","mcmc","MCMCpack","memoise","MESS","MetaMass","metRology","microbenchmark","mime","miniUI","minpack.lm","minqa","mixtools","mlbench","MLInterfaces","mlr","mnormt","ModelMetrics","modelr","modeltools","msdata","MSGFgui","MSGFplus","msmbstyle","msmsEDA","msmsTests","MSnbase","MSnID","MSqRob","MSstats","MSstatsBioData","mstut","multcomp","MultiAssayExperiment","multtest","munsell","mvtnorm","myp","myrp","mzID","mzR","naivebayes","ncdf4","nipals","nloptr","NMF","nnls","norm","Nozzle.R1","numDeriv","officer","openssl","openxlsx","ore","OrganismDbi","org.Hs.eg.db","OrgMassSpecR","org.Mm.eg.db","packrat","pamr","pander","parallelMap","ParamHelpers","parsedate","party","partykit","patchwork","pathview","Pbase","pbkrtest","pcaMethods","pheatmap","pillar","pingr","pkgbuild","pkgconfig","pkgDepTools","pkgdown","pkgload","pkgmaker","PKI","plogr","plotly","pls","plyr","png","polspline","polynom","PolynomF","prabclus","praise","precrec","preprocessCore","prettyunits","processx","prodlim","progress","pRoloc","pRolocdata","pRolocGUI","promises","ProteomicsAnnotationHubData","ProtGenerics","protiq","proto","protViz","proxy","prozor","pryr","ps","psych","purrr","Pviz","pwr","PXD000001","qap","qcmetrics","quadprog","quantmod","quantreg","questionr","qvalue","R6","rafalib","RamiGO","randomForest","ranger","RANN","rappdirs","raster","rayshader","RBGL","rBiopaxParser","R.cache","rcartocolor","rcmdcheck","RColorBrewer","Rcpp","RcppAnnoy","RcppArmadillo","RcppClassic","RcppEigen","RcppParallel","RcppRoll","RCurl","RCytoscape","rda","Rdisop","Rdsm","readbitmap","readBrukerFlexData","readMzXmlData","readr","readxl","recipes","RecordLinkage","registry","rematch","rematch2","remotes","renjin","reprex","repurrrsive","reshape","reshape2","revealjs","rex","RforProteomics","rgexf","rgl","RgoogleMaps","Rgraphviz","rhdf5","Rhdf5lib","rhub","rio","rJava","rjson","RJSONIO","rlang","rlist","rmarkdown","rmdexamples","rmdshower","R.methodsS3","Rmpi","rms","rmsfact","RMySQL","rngtools","robustbase","ROCR","rols","R.oo","Rook","roxygen2","rpart.plot","rprojroot","rpx","Rsamtools","rsconnect","RSQLite","rstudioapi","rTANDEM","rtracklayer","rtslprot","Rtsne","RUnit","R.utils","rvcheck","rvest","rvg","rWSBIM1207","S4Vectors","sampling","sandwich","scales","scater","scatterplot3d","scholar","scran","secret","segmented","selectr","seqinr","sequences","seriation","servr","sessioninfo","sfsmisc","shape","shiny","shinydashboard","shinyFiles","shinyjs","shinythemes","showtext","showtextdb","siggenes","signal","simpleaffy","SingleCellExperiment","snow","SnowballC","som","sourcetools","sp","SparseM","spData","speedtest","SQUAREM","stabs","startupmsg","statmod","stopwords","strex","strict","stringdist","stringi","stringr","strucchange","styler","sugsvarsel","SummarizedExperiment","sva","svDialogs","svglite","svGUI","SweaveListingUtils","synapter","synapterdata","sys","sysfonts","TeachingDemos","testthat","TH.data","threejs","tibble","tidies","tidygraph","tidyr","tidyselect","tidytext","tidytree","tidyverse","tiff","timeDate","tinytex","tmvtnorm","TnT","tokenizers","treeio","triebeard","trimcluster","truncnorm","tsne","TSP","TTR","tufte","tweenr","TxDb.Hsapiens.UCSC.hg19.knownGene","TxDb.Hsapiens.UCSC.hg38.knownGene","tximport","udapi","udunits2","unimod","UniProt.REST","units","UpSetR","urltools","usethis","utf8","uuid","V8","VariantAnnotation","VennDiagram","Vennerable","venneuler","vioplot","vipor","viridis","viridisLite","visNetwork","vsn","waveslim","webshot","whisker","whoami","withr","xaringan","xcms","xfun","xlsx","xlsxjars","XML","xml2","XMLRPC","xopen","XR","xtable","xts","XVector","yagop","yaml","yaqcaffy","yesno","zip","zlibbioc","zoo","base","boot","class","cluster","codetools","compiler","datasets","foreign","graphics","grDevices","grid","KernSmooth","lattice","MASS","Matrix","methods","mgcv","nlme","nnet","parallel","rpart","spatial","splines","stats","stats4","survival","tcltk","tools","utils"],["abind","acepack","ada","addinexamples","additivityTests","ade4","affy","affydata","affyio","affyPLM","AhoCorasickTrie","aLFQ","ALL","amap","animation","annotate","AnnotationDbi","AnnotationFilter","AnnotationForge","AnnotationHub","AnnotationHubData","ape","apercu","askpass","assertthat","backports","base2grob","base64enc","BASiCS","BBmisc","beachmat","beanplot","beeswarm","BH","bibtex","biclust","biglm","bigmemory","bigmemory.sri","bindr","bindrcpp","bio3d","Biobase","biobroom","BiocFileCache","BiocGenerics","BiocInstaller","BiocManager","BiocNeighbors","BiocParallel","biocpkgs","BiocStyle","BiocVersion","biocViews","BiocWorkflowTools","bioDist","biomaRt","Biostrings","biovizBase","bit","bit64","bitops","blob","blogdown","bmp","bookdown","BRAIN","brew","broom","BSgenome","BSgenome.Hsapiens.NCBI.GRCh38","C50","Cairo","callr","car","carData","Cardinal","caret","Category","caTools","cellranger","checkmate","circlize","classInt","cleaver","cli","clipr","clisymbols","clue","clusterProfiler","coda","CoGAPS","coin","colorspace","colourpicker","combinat","commonmark","ComplexHeatmap","conflicted","corrplot","covr","cowplot","cp4p","cputools","crayon","crosstalk","Cubist","curatedTCGAData","curl","CVST","d3heatmap","daff","DAPAR","DAPARdata","datapasta","data.table","data.tree","DBI","dbplyr","ddalpha","debugme","DelayedArray","DelayedMatrixStats","dendextend","dendsort","DEoptimR","DEP","desc","DESeq2","devtools","DiagrammeR","dichromat","digest","dimRed","diptest","distr","DO.db","doMC","doParallel","DOSE","doSNOW","downloader","dplyr","DRR","DT","dynamicTreeCut","e1071","EBImage","edgeR","egg","ellipse","emojifont","enc","enrichplot","EnsDb.Hsapiens.v86","ensembldb","europepmc","evaluate","evd","exampletestr","ExperimentHub","expm","faahKO","factoextra","FactoMineR","fansi","farver","fastcluster","fastmatch","fcuk","fdrtool","ff","ffbase","fftwtools","fgsea","filesstrings","fireData","flashClust","flexclust","flexdashboard","flexmix","FNN","forcats","foreach","formatR","Formula","fpc","fs","futile.logger","futile.options","future","gapminder","gbm","gclus","gcrma","gdata","gdtools","geeM","geepack","genefilter","geneplotter","generics","GenomeInfoDb","GenomeInfoDbData","GenomicAlignments","GenomicFeatures","GenomicRanges","geometry","GEOquery","geosphere","GetoptLong","getPass","GGally","ggbeeswarm","ggbio","ggdendro","ggExtra","ggforce","ggimage","ggiraph","gglabeller","ggmap","ggplot2","ggplotify","ggpubr","ggraph","ggrepel","ggridges","ggsci","ggsignif","ggstance","ggtree","ggvis","gh","git2r","glmnet","GlobalOptions","globals","glue","gmm","GO.db","golubEsets","googlesheets","GOSemSim","gower","gplots","gpls","graph","gridBase","gridExtra","gridGraphics","gridSVG","GSEABase","gsubfn","gtable","gtools","Gviz","gWidgets","haven","HDF5Array","heatmaply","Heatplus","here","hexbin","hexSticker","hgu133plus2cdf","hgu95av2cdf","hgu95av2.db","highcharter","highlight","highr","Hmisc","hms","Homo.sapiens","hpar","htmlTable","htmltools","htmlwidgets","httpuv","httr","hu6800.db","hunspell","hwriter","icon","igraph","iheatmapr","imager","imp4p","impute","imputeLCMD","influenceR","ini","interactiveDisplay","interactiveDisplayBase","inum","IPPD","ipred","IRanges","irlba","Iso","isobar","ISOcodes","iterators","janeaustenr","jpeg","jsonlite","KEGGgraph","keggorthology","KEGGREST","kernlab","kinship2","klaR","kmknn","knitr","labeling","labelled","lambda.r","LaplacesDemon","later","latticeExtra","lava","lazyeval","LCA","leaflet","leaps","libcoin","limma","listenv","lme4","lobstr","locfit","lpSolve","lubridate","magic","magick","magrittr","MALDIquant","MALDIquantForeign","manipulateWidget","mapproj","maps","maptools","MAQCsubsetAFX","markdown","marray","MassSpecWavelet","MatrixModels","matrixStats","matter","mboost","mcclust","mclust","mcmc","MCMCpack","memoise","MESS","MetaMass","metRology","microbenchmark","mime","miniUI","minpack.lm","minqa","mixtools","mlbench","MLInterfaces","mlr","mnormt","ModelMetrics","modelr","modeltools","msdata","MSGFgui","MSGFplus","msmbstyle","msmsEDA","msmsTests","MSnbase","MSnID","MSqRob","MSstats","MSstatsBioData","mstut","multcomp","MultiAssayExperiment","multtest","munsell","mvtnorm","myp","myrp","mzID","mzR","naivebayes","ncdf4","nipals","nloptr","NMF","nnls","norm","Nozzle.R1","numDeriv","officer","openssl","openxlsx","ore","OrganismDbi","org.Hs.eg.db","OrgMassSpecR","org.Mm.eg.db","packrat","pamr","pander","parallelMap","ParamHelpers","parsedate","party","partykit","patchwork","pathview","Pbase","pbkrtest","pcaMethods","pheatmap","pillar","pingr","pkgbuild","pkgconfig","pkgDepTools","pkgdown","pkgload","pkgmaker","PKI","plogr","plotly","pls","plyr","png","polspline","polynom","PolynomF","prabclus","praise","precrec","preprocessCore","prettyunits","processx","prodlim","progress","pRoloc","pRolocdata","pRolocGUI","promises","ProteomicsAnnotationHubData","ProtGenerics","protiq","proto","protViz","proxy","prozor","pryr","ps","psych","purrr","Pviz","pwr","PXD000001","qap","qcmetrics","quadprog","quantmod","quantreg","questionr","qvalue","R6","rafalib","RamiGO","randomForest","ranger","RANN","rappdirs","raster","rayshader","RBGL","rBiopaxParser","R.cache","rcartocolor","rcmdcheck","RColorBrewer","Rcpp","RcppAnnoy","RcppArmadillo","RcppClassic","RcppEigen","RcppParallel","RcppRoll","RCurl","RCytoscape","rda","Rdisop","Rdsm","readbitmap","readBrukerFlexData","readMzXmlData","readr","readxl","recipes","RecordLinkage","registry","rematch","rematch2","remotes","renjin","reprex","repurrrsive","reshape","reshape2","revealjs","rex","RforProteomics","rgexf","rgl","RgoogleMaps","Rgraphviz","rhdf5","Rhdf5lib","rhub","rio","rJava","rjson","RJSONIO","rlang","rlist","rmarkdown","rmdexamples","rmdshower","R.methodsS3","Rmpi","rms","rmsfact","RMySQL","rngtools","robustbase","ROCR","rols","R.oo","Rook","roxygen2","rpart.plot","rprojroot","rpx","Rsamtools","rsconnect","RSQLite","rstudioapi","rTANDEM","rtracklayer","rtslprot","Rtsne","RUnit","R.utils","rvcheck","rvest","rvg","rWSBIM1207","S4Vectors","sampling","sandwich","scales","scater","scatterplot3d","scholar","scran","secret","segmented","selectr","seqinr","sequences","seriation","servr","sessioninfo","sfsmisc","shape","shiny","shinydashboard","shinyFiles","shinyjs","shinythemes","showtext","showtextdb","siggenes","signal","simpleaffy","SingleCellExperiment","snow","SnowballC","som","sourcetools","sp","SparseM","spData","speedtest","SQUAREM","stabs","startupmsg","statmod","stopwords","strex","strict","stringdist","stringi","stringr","strucchange","styler","sugsvarsel","SummarizedExperiment","sva","svDialogs","svglite","svGUI","SweaveListingUtils","synapter","synapterdata","sys","sysfonts","TeachingDemos","testthat","TH.data","threejs","tibble","tidies","tidygraph","tidyr","tidyselect","tidytext","tidytree","tidyverse","tiff","timeDate","tinytex","tmvtnorm","TnT","tokenizers","treeio","triebeard","trimcluster","truncnorm","tsne","TSP","TTR","tufte","tweenr","TxDb.Hsapiens.UCSC.hg19.knownGene","TxDb.Hsapiens.UCSC.hg38.knownGene","tximport","udapi","udunits2","unimod","UniProt.REST","units","UpSetR","urltools","usethis","utf8","uuid","V8","VariantAnnotation","VennDiagram","Vennerable","venneuler","vioplot","vipor","viridis","viridisLite","visNetwork","vsn","waveslim","webshot","whisker","whoami","withr","xaringan","xcms","xfun","xlsx","xlsxjars","XML","xml2","XMLRPC","xopen","XR","xtable","xts","XVector","yagop","yaml","yaqcaffy","yesno","zip","zlibbioc","zoo","base","boot","class","cluster","codetools","compiler","datasets","foreign","graphics","grDevices","grid","KernSmooth","lattice","MASS","Matrix","methods","mgcv","nlme","nnet","parallel","rpart","spatial","splines","stats","stats4","survival","tcltk","tools","utils"],["/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/home/lg390/R/x86_64-pc-linux-gnu-library/3.5","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library","/usr/local/lib64/R/library"],["1.4-5","1.4.1","2.0-5","0.1.0","1.1-4","1.7-13","1.60.0","1.30.0","1.52.0","1.58.0","0.1.0","1.3.4","1.24.0","0.8-16","2.6","1.60.0","1.44.0","1.6.0","1.24.0","2.14.2","1.12.0","5.2","0.2.3","1.1","0.2.0","1.1.3","0.0.3","0.1-3","1.4.1","1.11","1.4.0","1.2","0.2.3","1.69.0-1","0.4.2","2.0.1","0.9-1","4.5.33","0.1.3","0.1.1","0.2.2","2.3-4","2.42.0","1.14.0","1.6.0","0.28.0","1.32.1","1.30.4","1.0.0","1.16.5","0.1.2","2.10.0","3.8.0","1.50.10","1.8.0","1.54.0","2.38.0","2.50.2","1.30.1","1.1-14","0.9-7","1.0-6","1.1.1","0.10","0.3","0.9","1.28.1","1.0-6","0.5.1","1.50.0","1.3.1000","0.1.2","1.5-9","3.1.1","3.0-2","3.0-2","2.0.3","6.0-81","2.48.0","1.17.1.1","1.1.0","1.9.1","0.4.5","0.3-1","1.20.0","1.0.1","0.5.0","1.2.0","0.3-56","3.10.1","0.19-2","3.2.2","1.2-2","1.4-0","1.0","0.0-8","1.7","1.20.0","1.0.1","0.84","3.2.1","0.9.4","0.3.5","0.2.0","1.3.4","1.0.0","0.2.2","1.4.3","3.3","0.2-2","0.6.1.2","0.3.0","1.14.4","1.12.1","3.0.0","1.12.0","0.7.8","1.0.0","1.3.0","1.3.8","1.1.0","0.8.0","1.4.0","1.9.0","0.3.3","1.0-8","1.4.1","1.2.0","1.22.2","2.0.1","1.0.0","2.0-0","0.6.18","0.2.2","0.75-7","2.7.0","2.9","1.3.5","1.0.14","3.8.2","1.0.16","0.4","0.7.8","0.0.3","0.5","1.63-1","1.7-0.1","4.24.0","3.24.3","0.4.2","0.4.1","0.5.2","0.2.0","1.2.0","2.99.0","2.6.3","0.3","0.12","2.3-3","1.4.1","1.8.0","0.999-3","1.22.0","1.0.5","1.41","0.4.0","1.1.0","1.1.25","1.1-0","0.1.21","1.2.15","2.2-14","0.12.7","0.9-8","1.8.0","3.0.0","1.1","1.01-2","1.4-0","0.5.1.1","2.3-14","1.1.2.2","0.3.0","1.4.4","1.5","1.2-3","2.1-11.1","1.2.6","1.4.3","1.0.1","1.10.0","0.3.0","2.1.5","1.3.2","2.54.0","2.18.0","0.1.7","0.10.1","1.2-1","1.64.0","1.60.0","0.0.2","1.18.1","1.2.0","1.18.1","1.34.1","1.34.0","0.3-6","2.50.5","1.5-7","0.1.7","0.2-2","1.4.0","0.6.0","1.30.0","0.1-20","0.8","0.1.3","0.2.1","0.6.0","0.1.0","2.6.1","3.1.0","0.0.3","0.2","1.0.2","0.8.0","0.5.1","2.9","0.4.0","0.3.1","1.14.6","0.4.4","1.0.1","0.24.0","2.0-16","0.1.0","0.12.4","1.3.0","1.6-2","3.7.0","1.24.0","0.3.0","2.8.0","0.1.2","3.0.1","1.54.0","1.60.0","0.4-7","2.3","0.3-0","1.6-0","1.44.0","0.7","0.2.0","3.8.1","1.26.4","0.0-54","2.0.0","1.10.1","0.15.2","2.28.0","0.1","1.27.2","0.4.5","2.18.0","2.18.0","3.2.3","0.7.0","0.4.7.2","0.7","4.1-1","0.4.2","1.3.1","1.25.1","1.13.1","0.3.6","1.3","1.4.5.1","1.4.0","3.2.3","3.0","1.3.2","0.1.0","1.2.2","0.4.3","0.41.2","0.7","1.56.0","2.0","0.1.0","0.3.1","1.20.0","1.20.0","1.0-0","1.30.0","0.9-8","2.16.0","2.3.2","0.0-17","1.28.1","2018.06.29","1.0.10","0.1.5","0.1-8","1.6","1.42.0","2.34.0","1.22.0","0.9-27","1.6.4","0.6-14","0.99.15","1.21","0.3","2.0.2","1.2.3","16.1.1","0.7.5","0.6-28","1.6.4","0.2.1","0.1","2.0.2","3.0","1.0-2","3.38.3","0.7.0","1.1-19","1.0.1","1.5-9.1","5.6.13","1.7.4","1.5-9","1.9","1.5","1.18","0.11.5","0.10.0","1.2.6","3.3.0","0.9-4","1.20.0","0.9","1.60.0","1.48.1","0.4-1","0.54.0","1.8.3","2.9-1","1.0","5.4.2","0.9-5","1.4-4","1.1.0","0.5.5","0.1-1","0.9-28-1","1.4-6","0.6","0.1.1.1","1.2-1","1.2.4","1.1.0","2.1-1","1.62.0","2.13","1.5-5","1.2.2","0.1.2","0.2-22","0.22.0","1.16.1","1.16.1","0.0.3","1.20.1","1.20.1","2.9.3","1.16.1","0.7.5","3.14.1","1.4.0","0.1.0","1.4-8","1.8.1","2.38.0","0.5.0","1.0-8","0.1.0","0.1.0","1.20.1","2.16.1","0.9.3","1.16","0.5","1.2.1","0.21.0","1.4","1.0-9.5","1.1-1","2016.8-1","0.3.2","1.2.1","4.1.0","1.6.2","1.24.0","3.7.0","0.5-3","3.7.0","0.5.0","1.55","0.6.3","1.3","1.12","1.1.3","1.3-1","1.2-2","0.0.1","1.22.1","0.22.1","0.4-7","1.74.0","1.0.12","1.3.1","1.1.2","1.0.2","2.0.2","1.48.0","1.3.0","1.0.2","0.27","0.1-5.1","0.2.0","4.8.0","2.7-0","1.8.4","0.1-7","1.1.13","1.3-9","1.0-2","2.2-7","1.0.0","0.9.1","1.44.0","1.0.2","3.2.1","2018.04.18","1.2.0","1.23.1","1.21.0","1.16.0","1.0.1","1.12.0","1.14.0","1.2","1.0.0","0.4.0","0.4-22","0.2.11","0.1.4","1.3.0","1.8.12","0.3.0","1.16.0","1.2-2","0.3.1","0.1-1","1.20.1","1.5-5","0.4-13","5.38","0.7.0","2.14.1","2.3.0","1.0.0","1.28.0","4.6-14","0.11.0","2.6.1","0.3.1","2.8-19","0.5.1","1.58.1","2.22.0","0.13.0","0.0.22","1.3.2","1.1-2","1.0.0","0.0.11","0.9.200.7.0","0.9.11","0.3.3.5.0","4.4.2","0.3.0","1.95-4.11","1.27.1","1.0.2-2.1","1.42.1","2.1.1","0.1.5","1.8.5","2.8.1","1.3.1","1.2.0","0.1.4","0.4-10","0.5","1.0.1","2.0.1","2.0.2","0.8.2353","0.2.1","0.1.0","0.8.8","1.4.3","0.9","1.1.2","1.20.0","0.15.3","0.99.16","1.4.3","2.26.0","2.27.3","1.4.2","1.0.2","0.5.16","0.9-10","0.2.20","1.3-1.1","0.3.1","0.4.6.1","1.11","1.0.1","2.1.1","1.7.1","0.6-9","5.1-2","0.0.3","0.10.16","1.3.1","0.93-3","1.0-7","2.10.1","1.22.0","1.1-1","6.1.1","3.0.6","1.3-2","1.18.1","1.34.0","0.8.13","2.1.1","0.9.0","1.22.1","1.42.1","0.2.2","0.15","0.4.32","2.7.0","0.1.3","0.3.2","0.1.9","0.1.0","0.20.1","2.8","2.5-0","1.0.0","1.10.1","0.3-41","0.1.7","1.10.2","1.0.0","0.5-3.0","0.4-1","3.4-5","0.5.9","1.2-3","0.11","1.1.1","1.1-3","1.4.4","1.2.0","0.7.1","0.7.2","1.0","1.1.2","0.6","2.0","1.56.0","0.7-6","2.58.0","1.4.1","0.4-3","0.6.0","0.3-5.1","0.1.7","1.3-1","1.77","0.3.0","0.1.0","2017.10-1","0.6-3","0.9.5","1.4.30","0.9.0","0.1.3","0.0.0.9000","0.9.5.1","1.2.4","1.3.1","1.5-1","1.1.0","0.1.0","1.12.0","3.30.1","1.0.0","1.2.1","1.0.0","0.7.7","2.6.1","1.20.0","2.1","0.8","2.10","2.0.1","1.0-10","0.3.1","2.0.1","0.0.3","1.1.1","0.8.2","0.2.5","0.2.0","0.2.1","1.2.1","0.1-5","3043.102","0.10","1.4-10","1.4.0","0.2.1","1.6.1","0.3.0","0.1-2.1","1.0-8","0.1-3","1.1-6","0.23-4","0.4","1.0.1","3.2.2","3.4.0","1.10.1","0.1.3","0.13","0.1","0.1.0","0.6-2","1.3.3","1.7.1","1.4.0","1.1.4","0.1-2","1.5","1.28.8","1.6.20","3.1.0.9000","1.1-0","0.2","0.4.5","0.5.1","0.3.0","2.0.5","3.50.0","1.7.5","0.5.1","0.3-2","1.2.0","2.1.2","0.8","3.4.2","0.4","0.6.1","0.6.1","3.98-1.16","1.2.0","0.3-0","1.0.0","0.7.2","1.8-3","0.11-2","0.22.0","0.1.0","2.2.0","1.42.0","0.1.0","1.0.0","1.28.0","1.8-4","3.6.0","1.3-20","7.3-14","2.0.7-1","0.2-15","3.6.0","3.6.0","0.8-70","3.6.0","3.6.0","3.6.0","2.23-15","0.20-35","7.3-50","1.2-14","3.6.0","1.8-24","3.1-137","7.3-12","3.6.0","4.1-13","7.3-11","3.6.0","3.6.0","3.6.0","2.42-4","3.6.0","3.6.0","3.6.0"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"base","recommended","recommended","recommended","recommended","base","base","recommended","base","base","base","recommended","recommended","recommended","recommended","base","recommended","recommended","recommended","base","recommended","recommended","base","base","base","recommended","base","base","base"],["R (&gt;= 1.5.0)",null,"R(&gt;= 2.10),rpart","R (&gt;= 3.0.0)",null,"R (&gt;= 2.10)","R (&gt;= 2.8.0), BiocGenerics (&gt;= 0.1.12), Biobase (&gt;= 2.5.5)","R (&gt;= 2.4.0), affy (&gt;= 1.23.4)","R (&gt;= 2.6.0)","R (&gt;= 2.6.0), BiocGenerics (&gt;= 0.3.2), affy (&gt;= 1.11.0),\nBiobase (&gt;= 2.17.8), gcrma, stats, preprocessCore (&gt;= 1.5.1)",null,"R (&gt;= 2.15.0)","R (&gt;= 2.10), Biobase (&gt;= 2.5.5)","R (&gt;= 2.10.0)","R (&gt;= 2.14.0)","R (&gt;= 2.10), AnnotationDbi (&gt;= 1.27.5), XML","R (&gt;= 2.7.0), methods, utils, stats4, BiocGenerics (&gt;=\n0.23.1), Biobase (&gt;= 1.17.0), IRanges","R (&gt;= 3.4.0)","R (&gt;= 2.7.0), methods, utils, BiocGenerics (&gt;= 0.15.10),\nBiobase (&gt;= 1.17.0), AnnotationDbi (&gt;= 1.33.14)","BiocGenerics (&gt;= 0.15.10)","R (&gt;= 3.2.2), methods, utils, S4Vectors (&gt;= 0.7.21), IRanges\n(&gt;= 2.3.23), GenomicRanges, AnnotationHub (&gt;= 2.9.14)","R (&gt;= 3.2.0)",null,null,null,"R (&gt;= 3.0.0)","R (&gt;= 3.4.0)","R (&gt;= 2.9.0)","R (&gt;= 3.5), SingleCellExperiment, KernSmooth, ggplot2",null,"R (&gt;= 3.5)",null,null,null,"R (&gt;= 3.0.2)","R (&gt;= 2.10), MASS, grid, colorspace, lattice","DBI, methods","R (&gt;= 3.2.0),",null,null,null,"R (&gt;= 3.1.0)","R (&gt;= 2.10), BiocGenerics (&gt;= 0.27.1), utils","R (&gt;= 3.0.0), broom","R (&gt;= 3.4.0), dbplyr (&gt;= 1.0.0)","methods, utils, graphics, stats, parallel","R (&gt;= 3.5.0), R (&lt; 3.6.0)","R (&gt;= 3.5.0)","R (&gt;= 3.5), BiocParallel","methods","R (&gt;= 3.4.0)",null,"R (&gt;= 3.5.0), R (&lt; 3.6.0)","R (&gt;= 3.5.0)","R (&gt;= 3.4)","R (&gt;= 2.0), methods, Biobase,KernSmooth","methods","R (&gt;= 3.5.0), methods, BiocGenerics, S4Vectors, IRanges,\nXVector","R (&gt;= 2.10), methods","R (&gt;= 2.9.2)","R (&gt;= 3.0.1), bit (&gt;= 1.1-12), utils, methods, stats",null,null,null,null,null,"R (&gt;= 2.8.1), PolynomF, Biostrings, lattice",null,"R (&gt;= 3.1)","R (&gt;= 2.8.0), methods, BiocGenerics (&gt;= 0.13.8), S4Vectors (&gt;=\n0.17.28), IRanges (&gt;= 2.13.16), GenomeInfoDb (&gt;= 1.15.2),\nGenomicRanges (&gt;= 1.31.10), Biostrings (&gt;= 2.47.6), rtracklayer\n(&gt;= 1.39.7)","BSgenome (&gt;= 1.33.5)","R (&gt;= 2.10.0)","R (&gt;= 2.4.0)",null,"R (&gt;= 3.2.0), carData (&gt;= 3.0-0)","R (&gt;= 3.0)","BiocGenerics, BiocParallel, EBImage, graphics, methods,\nS4Vectors, stats, ProtGenerics","R (&gt;= 2.10), lattice (&gt;= 0.20), ggplot2","methods, stats4, BiocGenerics, AnnotationDbi, Biobase, Matrix","R (&gt;= 2.2.0)","R (&gt;= 3.0.0)","R (&gt;= 3.0.0)","R (&gt;= 3.0.0), graphics","R (&gt;= 2.2)","R (&gt;= 3.0.0), methods, Biostrings (&gt;= 1.29.8)","R (&gt;= 2.10)",null,null,"R (&gt;= 3.2.0)","R (&gt;= 3.4.0)","R (&gt;= 2.14.0)","R (&gt;= 3.5.0)","R (&gt;= 2.14.0), methods, survival","R (&gt;= 3.0.0), methods","R (&gt;= 3.1.0)",null,null,"R (&gt;= 3.1.2), methods, grid, graphics, stats, grDevices","R (&gt;= 3.1)",null,"R (&gt;= 3.1.0), methods","R (&gt;= 3.3.0), ggplot2 (&gt;= 2.1.0),","R (&gt;= 3.2.0), MESS, graphics, stats, multtest, qvalue, limma",null,null,null,"lattice","R (&gt;= 3.5.0), MultiAssayExperiment","R (&gt;= 3.0.0)","kernlab,Matrix",null,null,"R (&gt;= 3.5), foreach, parallel, doParallel","R (&gt;= 3.5)","R (&gt;= 3.3.0)","R (&gt;= 3.1.0)","R (&gt;= 3.0)","R (&gt;= 3.0.0), methods","R (&gt;= 3.1)","R (&gt;= 2.10), stats, utils, graphics, grDevices, MASS, class,\nrobustbase, sfsmisc, geometry",null,"R (&gt;= 3.4), methods, stats4, matrixStats, BiocGenerics (&gt;=\n0.27.1), S4Vectors (&gt;= 0.19.15), IRanges (&gt;= 2.11.17),\nBiocParallel","DelayedArray (&gt;= 0.7.37)","R (&gt;= 3.0.0)",null,null,"R (&gt;= 3.5)","R (&gt;= 3.1.0)","S4Vectors (&gt;= 0.9.25), IRanges, GenomicRanges,\nSummarizedExperiment (&gt;= 1.1.6)","R (&gt;= 3.0.2)","R (&gt;= 3.2.0)","R (&gt;= 2.10), stats","R (&gt;= 3.1.0)","R (&gt;= 3.0.0), DRR",null,"R(&gt;= 2.14.0), methods, graphics, startupmsg, sfsmisc","R (&gt;= 2.7.0), methods, AnnotationDbi (&gt;= 1.9.7)","R (&gt;= 2.14.0), foreach(&gt;= 1.2.0), iterators(&gt;= 1.0.0),\nparallel","R (&gt;= 2.14.0), foreach(&gt;= 1.2.0), iterators(&gt;= 1.0.0),\nparallel, utils","R (&gt;= 3.4.0)","R (&gt;= 2.5.0), foreach(&gt;= 1.2.0), iterators(&gt;= 1.0.0), snow(&gt;=\n0.3.0), utils",null,"R (&gt;= 3.1.2)","kernlab, CVST, Matrix",null,"R (&gt;= 2.3.0), stats",null,"methods","R (&gt;= 2.15.0), limma (&gt;= 3.34.5)","gridExtra (&gt;= 2.3), ggplot2","R (&gt;= 2.0.0),graphics,stats","R (&gt;= 3.4.0)","R (&gt;= 3.1)","R (&gt;= 3.4.0)","ensembldb","BiocGenerics (&gt;= 0.15.10), GenomicRanges (&gt;= 1.31.18),\nGenomicFeatures (&gt;= 1.29.10), AnnotationFilter (&gt;= 1.5.2)","R (&gt;= 3.00)","R (&gt;= 3.0.2)",null,"R (&gt;= 3.5.0)","methods, BiocGenerics (&gt;= 0.15.10), AnnotationHub (&gt;= 2.13.4)","Matrix","xcms","R (&gt;= 3.1.2), ggplot2 (&gt;= 2.2.0)","R (&gt;= 3.0.0)","R (&gt;= 3.1.0)",null,"R (&gt;= 3.0.0)",null,null,"R (&gt;= 3.0.2)","R (&gt;= 2.10.1), bit (&gt;= 1.1-13), utils","R (&gt;= 2.12.0),ff(&gt;= 2.2-11)","R (&gt;= 2.15.2)","R (&gt;= 3.3), Rcpp","stringr","R (&gt;= 3.3.3)","R (&gt;= 2.3.0)","R (&gt;= 2.14.0), graphics, grid, lattice, modeltools","R (&gt;= 3.0.2)","R (&gt;= 2.15.0), lattice","R (&gt;= 3.0.0)","R (&gt;= 3.1)","R (&gt;= 2.5.0)","R (&gt;= 3.0.2)","R (&gt;= 2.0.0), stats","R (&gt;= 2.0)","R (&gt;= 3.1)","R (&gt;= 3.0.0)","R (&gt;= 2.8.0)",null,"R (&gt;= 3.1.0)","R (&gt;= 2.9.0)","R (&gt;= 2.10), cluster","R (&gt;= 2.6.0), affy (&gt;= 1.23.2), graphics, methods, stats,\nutils","R (&gt;= 2.3.0)",null,"Matrix",null,null,"R (&gt;= 2.10), methods, Biobase, BiocGenerics, lattice, annotate","R (&gt;= 3.1)","R (&gt;= 3.1), methods, BiocGenerics (&gt;= 0.13.8), S4Vectors (&gt;=\n0.17.25), IRanges (&gt;= 2.13.12)","R (&gt;= 3.3)","R (&gt;= 2.10), methods, BiocGenerics (&gt;= 0.15.3), S4Vectors (&gt;=\n0.19.11), IRanges (&gt;= 2.15.12), GenomeInfoDb (&gt;= 1.13.1),\nGenomicRanges (&gt;= 1.33.4), SummarizedExperiment (&gt;= 1.9.13),\nBiostrings (&gt;= 2.47.6), Rsamtools (&gt;= 1.31.2)","BiocGenerics (&gt;= 0.1.0), S4Vectors (&gt;= 0.17.29), IRanges (&gt;=\n2.13.23), GenomeInfoDb (&gt;= 1.15.4), GenomicRanges (&gt;= 1.31.17),\nAnnotationDbi (&gt;= 1.41.4)","R (&gt;= 2.10), methods, stats4, BiocGenerics (&gt;= 0.25.3),\nS4Vectors (&gt;= 0.19.11), IRanges (&gt;= 2.15.12), GenomeInfoDb (&gt;=\n1.15.2)","R (&gt;= 2.5.0), magic","methods, Biobase","R (&gt;= 3.0.0)","R (&gt;= 3.0.0)","R (&gt;= 3.0.0)","R (&gt;= 3.1), ggplot2 (&gt; 2.2.0)","R (&gt;= 3.0.0), ggplot2 (&gt;= 2.0)","methods, BiocGenerics, ggplot2 (&gt;= 1.0.0)",null,"R (&gt;= 3.1.0)","ggplot2 (&gt;= 2.2.0), R (&gt;= 3.3.0)","R (&gt;= 3.4.0), ggplot2",null,"R (&gt;= 3.3)","R (&gt;= 2.14.0), ggplot2 (&gt;= 2.0.0)","R (&gt;= 3.1)","R (&gt;= 3.4.0)","R (&gt;= 3.1.0), ggplot2, magrittr","R (&gt;= 2.10), ggplot2 (&gt;= 2.0.0)","R (&gt;= 3.0.0), ggplot2 (&gt;= 2.2.0)","R (&gt;= 3.2)","R (&gt;= 3.0.2)",null,"R (&gt;= 3.1.0)","R (&gt;= 3.4.0)","R (&gt;= 3.0)",null,"R (&gt;= 3.1)","Matrix (&gt;= 1.0-6), utils, foreach","R (&gt;= 3.3.0), methods","R (&gt;= 3.1.2)","R (&gt;= 3.1)","R (&gt;= 2.10.0), sandwich","R (&gt;= 2.7.0), methods, AnnotationDbi (&gt;= 1.43.1)","R (&gt;= 2.14.0), Biobase (&gt;= 2.5.5)","R (&gt;= 3.2.0)","R (&gt;= 3.4.0)",null,"R (&gt;= 3.0)",null,"R (&gt;= 2.10), methods, BiocGenerics (&gt;= 0.13.11)","R (&gt;= 2.3.0)",null,"grid, graphics","R (&gt;= 3.1)","R (&gt;= 2.6.0), BiocGenerics (&gt;= 0.13.8), Biobase (&gt;= 2.17.8),\nannotate (&gt;= 1.45.3), methods, graph (&gt;= 1.37.2)","proto","R (&gt;= 2.14)","methods, stats, utils","R (&gt;= 2.10.0), methods, S4Vectors (&gt;= 0.9.25), IRanges (&gt;=\n1.99.18), GenomicRanges (&gt;= 1.17.20), grid","methods, utils","R (&gt;= 3.1)","R (&gt;= 3.4), methods, DelayedArray (&gt;= 0.7.41), rhdf5 (&gt;=\n2.25.6)","R (&gt;= 3.0.0), plotly (&gt;= 4.7.1), viridis",null,null,"R (&gt;= 2.0.1), methods","R (&gt;= 3.4.0)","utils","utils","R (&gt;= 2.7.0), methods, AnnotationDbi (&gt;= 1.34.3), org.Hs.eg.db\n(&gt;= 3.3.0)","R (&gt;= 2.10)","R (&gt;= 3.1.3)","R (&gt;= 3.0.2)","lattice, survival (&gt;= 2.40-1), Formula, ggplot2 (&gt;= 2.2)",null,"R (&gt;= 1.6), AnnotationDbi (&gt;= 1.17.11), methods, OrganismDbi\n(&gt;= 1.11.39), GenomicFeatures, GO.db, org.Hs.eg.db,\nTxDb.Hsapiens.UCSC.hg19.knownGene","R (&gt;= 2.15)",null,"R (&gt;= 2.14.1)",null,"R (&gt;= 2.15.1), methods","R (&gt;= 3.1)","R (&gt;= 2.7.0), methods, AnnotationDbi (&gt;= 1.34.3), org.Hs.eg.db\n(&gt;= 3.3.0)","R (&gt;= 3.0.2)","R (&gt;= 2.6.0)","R (&gt;= 3.1.3)","methods","R (&gt;= 3.2.0)","R (&gt;= 2.10.0),magrittr","R (&gt;= 3.3.0), Iso, stats, truncnorm, norm","R (&gt;= 2.10)","R (&gt;= 2.10), tmvtnorm, norm, pcaMethods, impute","R (&gt;= 3.2.0)",null,"R (&gt;= 2.10), methods, BiocGenerics, grid","R (&gt;= 2.10), methods, BiocGenerics","R (&gt;= 3.3.0)","R (&gt;= 2.12.0), MASS, Matrix, XML, digest, bitops","R (&gt;= 2.10)","R (&gt;= 3.1.0), methods, utils, stats, BiocGenerics (&gt;= 0.25.3),\nS4Vectors (&gt;= 0.19.11)","Matrix","R (&gt;= 1.7.0)","R (&gt;= 2.10.0), Biobase, stats, methods","R (&gt;= 2.10.0)","R (&gt;= 2.5.0), utils","R (&gt;= 3.1.2)","R (&gt;= 2.9.0)","methods","R (&gt;= 2.10.0)","R (&gt;= 2.5.0),stats,graph,hgu95av2.db",null,"R (&gt;= 2.10)","R (&gt;= 3.1.0), Matrix, quadprog","R (&gt;= 2.10.0), MASS","R (&gt;= 3.5), BiocParallel","R (&gt;= 3.1.0)",null,null,"R (&gt;= 3.0.0)","R (&gt;= 3.0.0)",null,"R (&gt;= 2.10.0), lattice, RColorBrewer","R (&gt;= 3.0)","R (&gt;= 3.1.0)","R (&gt;= 2.15.0)","R (&gt;= 3.1.0)","","R (&gt;= 3.4.0)","R (&gt;= 2.3.0)","R (&gt;= 3.1.2)","R (&gt;= 3.2.0), Matrix (&gt;= 1.2-1), methods, stats","R (&gt;= 3.1)","R (&gt;= 2.0.1)",null,"methods, R (&gt;= 3.0.0)","R (&gt;= 2.10), abind",null,null,"R (&gt;= 3.2.0), methods","R (&gt;= 3.2.2), methods, MALDIquant (&gt;= 1.16.4)",null,"R (&gt;= 3.0.0), maps (&gt;= 2.3-0)","R (&gt;= 3.0.0)","R (&gt;= 2.10), sp (&gt;= 1.0-11)","R (&gt;= 2.10), affy (&gt;= 1.23.4), Biobase (&gt;= 2.5.5)","R (&gt;= 2.11.1)","R (&gt;= 2.10.0), limma, methods","waveslim","R (&gt;= 3.0.1)","R (&gt;= 2.12.0)","R (&gt;= 3.5), methods, stats, biglm","R (&gt;= 3.2.0), methods, stats, parallel, stabs (&gt;= 0.5-0)","R (&gt;= 2.10), lpSolve","R (&gt;= 3.0)","R (&gt;= 3.0.2)","R (&gt;= 2.10.0), coda (&gt;= 0.11-3), MASS, stats",null,"R (&gt;= 3.1), geepack, geeM,","R (&gt;= 3.0.0), cluster,amap","R (&gt;= 2.14.0), base, stats",null,null,null,null,null,"R (&gt;= 3.2.0)","R (&gt;= 2.10)","R (&gt;= 3.5), methods, BiocGenerics (&gt;= 0.13.11), Biobase,\nannotate, cluster","R (&gt;= 3.0.2), ParamHelpers (&gt;= 1.10)","R (&gt;= 2.2.0)","R (&gt;= 3.2.2)","R (&gt;= 3.1)","stats, stats4","R (&gt;= 2.10)","mzR, xlsx","methods","bookdown","R (&gt;= 3.0.1), MSnbase","R (&gt;= 3.0.1), MSnbase, msmsEDA","R (&gt;= 3.1), methods, BiocGenerics (&gt;= 0.7.1), Biobase (&gt;=\n2.15.2), mzR (&gt;= 2.15.1), S4Vectors, ProtGenerics (&gt;= 1.9.1)","R (&gt;= 2.10), Rcpp","R (&gt;= 3.0.1),\nlme4","R (&gt;= 3.4)","R (&gt;= 3.4)","R (&gt;= 3.4.0)","stats, graphics, mvtnorm (&gt;= 1.0-3), survival (&gt;= 2.39-4),\nTH.data (&gt;= 1.0-2)","R (&gt;= 3.5.0), SummarizedExperiment (&gt;= 1.3.81)","R (&gt;= 2.10), methods, BiocGenerics, Biobase",null,"R(&gt;= 1.9.0)",null,null,"methods","Rcpp (&gt;= 0.10.1), methods, utils",null,null,"R (&gt;= 3.4.0)",null,"R (&gt;= 3.0.0), methods, utils, pkgmaker (&gt;= 0.20), registry,\nrngtools (&gt;= 1.2.3), cluster",null,null,null,"R (&gt;= 2.11.1)",null,null,"R (&gt;= 3.3.0)",null,"R (&gt;= 2.14.0), methods, BiocGenerics (&gt;= 0.15.10),\nAnnotationDbi (&gt;= 1.33.15), GenomicFeatures (&gt;= 1.23.31)","R (&gt;= 2.7.0), methods, AnnotationDbi (&gt;= 1.43.1)","R (&gt;= 3.0.0), grid","R (&gt;= 2.7.0), methods, AnnotationDbi (&gt;= 1.43.1)","R (&gt;= 3.0.0)","R (&gt;= 2.10), cluster, survival","R (&gt;= 2.15.0)","R (&gt;= 3.0.0)",null,"methods","R (&gt;= 3.0.0), methods, grid, stats, mvtnorm (&gt;= 1.0-2),\nmodeltools (&gt;= 0.2-21), strucchange","R (&gt;= 3.1.0), graphics, grid, libcoin (&gt;= 1.0-0), mvtnorm",null,"R (&gt;= 2.10), org.Hs.eg.db","R (&gt;= 2.10), methods, BiocGenerics, Rcpp, Gviz","R (&gt;= 3.2.3), lme4 (&gt;= 1.1.10)","Biobase, methods","R (&gt;= 2.0)",null,null,"R (&gt;= 3.1)",null,"methods, graph, RBGL","R (&gt;= 3.1.0)",null,"R (&gt;= 3.0.0), methods, stats, registry","R (&gt;= 2.9.0), base64enc",null,"R (&gt;= 3.2.0), ggplot2 (&gt;= 3.0.0)","R (&gt;= 2.10)","R (&gt;= 3.1.0)","R (&gt;= 2.9.0)",null,null,"R (&gt;= 3.0.0), graphics","R (&gt;= 2.10), MASS, mclust",null,"R (&gt;= 3.2.1)",null,null,null,"R (&gt;= 2.9.0)",null,"R (&gt;= 2.15), MSnbase (&gt;= 1.19.20), MLInterfaces (&gt;= 1.37.1),\nmethods, Rcpp (&gt;= 0.10.3), BiocParallel","R (&gt;= 2.15), MSnbase","methods, R (&gt;= 3.1.0), pRoloc (&gt;= 1.11.1), Biobase, MSnbase\n(&gt;= 2.1.11)",null,"AnnotationHub (&gt;= 2.1.45), AnnotationHubData,","methods","methods",null,"R (&gt;= 3.3), methods","R (&gt;= 3.4.0)",null,"R (&gt;= 3.1.0)","R (&gt;= 3.1)","R (&gt;= 2.10)","R (&gt;= 3.1)","R(&gt;= 3.0.0), Gviz(&gt;= 1.7.10)",null,"mzR, MSnbase",null,"R (&gt;= 3.3)","R (&gt;= 2.15.0)","R (&gt;= 3.2.0), xts(&gt;= 0.9-0), zoo, TTR(&gt;= 0.2), methods","R (&gt;= 2.6), stats, SparseM","R (&gt;= 2.10)","R(&gt;= 2.10)","R (&gt;= 3.0)",null,"gsubfn,methods","R (&gt;= 3.2.2), stats","R (&gt;= 3.1)",null,"R (&gt;= 2.14), methods","sp (&gt;= 1.2-0), R (&gt;= 3.0.0)","R (&gt;= 3.0.2)","graph, methods","R (&gt;= 3.0.0), data.table","R (&gt;= 2.14.0)","R (&gt;= 3.1.0)",null,"R (&gt;= 2.0.0)","R (&gt;= 3.0.0)","R (&gt;= 3.1)","R (&gt;= 3.3.0)","R (&gt;= 2.12.0)","R (&gt;= 2.15.1)","R (&gt;= 3.0.2)","R (&gt;= 2.15.1)","R (&gt;= 3.0.0), methods, bitops","R (&gt;= 2.14.0), graph (&gt;= 1.31.0), XMLRPC (&gt;= 0.2.4)","R (&gt;= 2.10)","R (&gt;= 2.0.0), RcppClassic",null,null,"R (&gt;= 3.3.0)","R (&gt;= 2.15.0)","R (&gt;= 3.1)",null,"R (&gt;= 3.1), dplyr","R (&gt;= 2.10.0), DBI, RSQLite(&gt;= 1.0.0), ff, ffbase","R (&gt;= 2.6.0)",null,null,"R (&gt;= 3.0.0)",null,"R (&gt;= 3.1)","R (&gt;= 2.10)","R (&gt;= 2.6.1)","R (&gt;= 3.1)","R (&gt;= 3.0)",null,"MSnbase (&gt;= 2.5.3)","XML, Rook, igraph","R (&gt;= 3.2.0)","R (&gt;= 2.10)","R (&gt;= 2.6.0), methods, utils, graph, grid","R (&gt;= 3.5.0), methods",null,null,"R (&gt;= 2.15.0)","R (&gt;= 2.5.0), methods","R (&gt;= 3.1.0)",null,"R (&gt;= 3.1.0)","R (&gt;= 2.15)","R (&gt;= 3.0)","methods","R (&gt;= 3.0)","R (&gt;= 2.13.0)","R (&gt;= 2.15.1)","Hmisc (&gt;= 4.1-0), survival (&gt;= 2.40-1), lattice, ggplot2 (&gt;=\n2.2), SparseM",null,"R (&gt;= 2.8.0), DBI (&gt;= 0.4)","R (&gt;= 3.0.0), methods, pkgmaker (&gt;= 0.20)","R (&gt;= 3.1.0)","gplots, methods","methods","R (&gt;= 2.13.0), R.methodsS3 (&gt;= 1.7.1)","R (&gt;= 2.13.0)","R (&gt;= 3.1)","R (&gt;= 3.2.0), rpart (&gt;= 4.1-10)","R (&gt;= 3.0.0)","methods","methods, GenomeInfoDb (&gt;= 1.1.3), GenomicRanges (&gt;= 1.31.8),\nBiostrings (&gt;= 2.47.6)","R (&gt;= 3.0.0)","R (&gt;= 3.1.0)",null,"XML, Rcpp, data.table (&gt;= 1.8.8)","R (&gt;= 3.3), methods, GenomicRanges (&gt;= 1.31.8)","MSnbase",null,"R (&gt;= 2.5.0), utils (&gt;= 2.5.0), methods (&gt;= 2.5.0), graphics\n(&gt;= 2.5.0)","R (&gt;= 2.14.0), R.oo (&gt;= 1.21.0)","R (&gt;= 3.4.0)","R (&gt;= 3.0.1), xml2","R (&gt;= 3.0)",null,"R (&gt;= 3.3.0), methods, utils, stats, stats4, BiocGenerics (&gt;=\n0.23.3)",null,"R (&gt;= 2.10.0)","R (&gt;= 3.1)","R (&gt;= 3.5), SingleCellExperiment, ggplot2, BiocParallel","R (&gt;= 2.7.0)","R (&gt;= 3.4.0)","R (&gt;= 3.5), BiocParallel, SingleCellExperiment",null,null,"R (&gt;= 3.0)","R (&gt;= 2.10.0)","methods, Rcpp","R (&gt;= 2.14.0)","R (&gt;= 3.0.0)",null,"R (&gt;= 3.2.0)","R (&gt;= 2.01)","R (&gt;= 3.0.2), methods","R (&gt;= 3.0)",null,"R (&gt;= 3.1.0)","R (&gt;= 3.0.0)","sysfonts (&gt;= 0.7.1), showtextdb (&gt;= 2.0)",null,"methods, Biobase, multtest, splines, graphics","R (&gt;= 2.14.0)","R (&gt;= 2.0.0), methods, utils, grDevices, graphics, stats,\nBiocGenerics (&gt;= 0.1.12), Biobase, affy (&gt;= 1.33.6),\ngenefilter, gcrma","R (&gt;= 3.5), SummarizedExperiment","R (&gt;= 2.13.1), utils",null,"R (&gt;= 2.10)","R (&gt;= 3.0.2)","R (&gt;= 3.0.0), methods","R (&gt;= 2.15), methods","R (&gt;= 3.3.0)","R (&gt;= 3.2.0)","R (&gt;= 3.0)","R (&gt;= 2.14.0), methods, stats, parallel","R(&gt;= 1.8.0)","R (&gt;= 1.6.1)","R (&gt;= 2.10)","stringr",null,"R (&gt;= 2.15.3)","R (&gt;= 2.14)","R (&gt;= 3.1)","R (&gt;= 2.10.0), zoo, sandwich",null,"metRology, Matrix, BiocParallel, Rcpp, mnormt, mcclust","R (&gt;= 3.2), methods, GenomicRanges (&gt;= 1.33.6), Biobase,\nDelayedArray (&gt;= 0.3.20)","R (&gt;= 3.2), mgcv, genefilter, BiocParallel","R (&gt;= 2.6.0)","R (&gt;= 3.0.0)","R (&gt;= 2.6.0)","R(&gt;= 2.14.0), startupmsg","R (&gt;= 3.1.0), methods, MSnbase (&gt;= 2.1.2)","R (&gt;= 2.10), synapter (&gt;= 1.99.2)",null,null,null,"R (&gt;= 3.1)","R (&gt;= 2.10.0), survival, MASS","R (&gt;= 3.0.0), igraph (&gt;= 1.0.0)","R (&gt;= 3.1.0)","R (&gt;= 3.4.0), Biobase, MSnbase, dplyr, methods",null,"R (&gt;= 3.1)","R (&gt;= 3.1)","R (&gt;= 2.10)","R (&gt;= 3.4.0)",null,"R (&gt;= 2.9.0)","R (&gt;= 2.15.1), graphics, utils, stats, methods",null,"R (&gt;= 1.9.0), mvtnorm, utils, Matrix, stats4, gmm","R (&gt;= 3.4), GenomicRanges","R (&gt;= 3.1.3)","R (&gt;= 3.4.0)",null,"R (&gt;= 1.9.0)","R (&gt;= 3.4.0)",null,"R (&gt;= 2.14.0)",null,null,"R (&gt;= 3.2.0)","GenomicFeatures (&gt;= 1.21.30)","GenomicFeatures (&gt;= 1.25.18)",null,null,"R (&gt;= 2.10.0)","R (&gt;= 3.3)",null,"R (&gt;= 3.0.2)","R (&gt;= 3.0)","R (&gt;= 2.10)","R (&gt;= 3.1)","R (&gt;= 2.10)","R (&gt;= 2.9.0)",null,"R (&gt;= 2.8.0), methods, BiocGenerics (&gt;= 0.15.3), GenomeInfoDb\n(&gt;= 1.15.2), GenomicRanges (&gt;= 1.31.8), SummarizedExperiment\n(&gt;= 1.9.9), Rsamtools (&gt;= 1.33.6)","R (&gt;= 2.14.1), grid (&gt;= 2.14.1), futile.logger","methods, R (&gt;= 2.10)","rJava","sm","R (&gt;= 3.0.0)","R (&gt;= 2.10), viridisLite (&gt;= 0.3.0)","R (&gt;= 2.10)","R (&gt;= 3.0)","R (&gt;= 3.4.0), Biobase","R (&gt;= 2.11.0), graphics, grDevices, stats, utils","R (&gt;= 3.0)",null,null,"R (&gt;= 3.0.2)",null,"R (&gt;= 2.14.0), methods, Biobase, BiocParallel (&gt;= 1.8.0),\nMSnbase (&gt;= 2.7.8)",null,null,"rJava","R (&gt;= 2.13.0), methods, utils","R (&gt;= 3.1.0)",null,"R (&gt;= 3.1)",null,"R (&gt;= 2.10.0)","zoo (&gt;= 1.7-12)","R (&gt;= 2.8.0), methods, BiocGenerics (&gt;= 0.19.2), S4Vectors (&gt;=\n0.19.15), IRanges (&gt;= 2.15.12)",null,null,"simpleaffy (&gt;= 2.19.3), methods","R (&gt;= 3.1.0)",null,null,"R (&gt;= 3.1.0), stats",null,"R (&gt;= 3.0.0), graphics, stats","R (&gt;= 3.0.0), stats, utils","R (&gt;= 3.2.0)","R (&gt;= 2.1)",null,null,"R (&gt;= 3.0.0)",null,null,null,"R (&gt;= 2.5.0), stats","R (&gt;= 3.0.0)","R (&gt;= 3.1.0), grDevices, graphics, stats, utils","R (&gt;= 3.2.0)",null,"R (&gt;= 2.14.0), nlme (&gt;= 3.1-64)","R (&gt;= 3.4.0)","R (&gt;= 2.14.0), stats, utils",null,"R (&gt;= 2.15.0), graphics, stats, grDevices","R (&gt;= 3.0.0), graphics, stats, utils",null,null,null,"R (&gt;= 2.13.0)",null,null,null],["methods, utils",null,null,"shiny (&gt;= 0.13),\nminiUI (&gt;= 0.1.1),\nrstudioapi (&gt;= 0.4),\nformatR",null,"graphics, grDevices, methods, stats, utils, MASS","affyio (&gt;= 1.13.3), BiocManager, graphics, grDevices, methods,\npreprocessCore, stats, utils, zlibbioc","methods","zlibbioc, methods","zlibbioc, graphics, grDevices, methods","Rcpp (&gt;= 0.12.5)","data.table, plyr, caret, seqinr, lattice, randomForest, ROCR,\nreshape2, protiq, bio3d",null,null,"magick","Biobase, DBI, xtable, graphics, utils, stats, methods,\nBiocGenerics (&gt;= 0.13.8), RCurl","DBI, RSQLite, S4Vectors (&gt;= 0.9.25)","utils, methods, GenomicRanges, lazyeval","DBI, RSQLite, XML, S4Vectors, RCurl","utils, methods, grDevices, RSQLite, BiocManager, curl,\nAnnotationDbi (&gt;= 1.31.19), S4Vectors, interactiveDisplayBase,\nhttr, yaml","GenomicFeatures, Rsamtools, rtracklayer, BiocGenerics,\njsonlite, BiocManager, biocViews, AnnotationDbi, Biobase,\nBiostrings, DBI, GenomeInfoDb (&gt;= 1.15.4), OrganismDbi,\nRSQLite, rBiopaxParser, AnnotationForge, futile.logger (&gt;=\n1.3.0), XML, RCurl","nlme, lattice, graphics, methods, stats, tools, utils,\nparallel, Rcpp (&gt;= 0.12.0)","pls","sys (&gt;= 2.1)","tools","utils","ggplotify",null,"BiocGenerics, methods, coda, data.table, graphics, grDevices,\nMASS, matrixStats, Rcpp (&gt;= 0.11.3), S4Vectors, scran, stats,\nSummarizedExperiment, testthat, utils","utils, methods, stats, checkmate (&gt;= 1.8.0)","utils, methods, Rhdf5lib (&gt;= 1.1.4), rhdf5, HDF5Array (&gt;=\n1.9.5), DelayedArray (&gt;= 0.7.38), Rcpp (&gt;= 0.12.14),\nBiocGenerics",null,"stats, graphics, grDevices, utils",null,"stringr, utils","methods, flexclust, additivityTests, tidyr, ggplot2",null,"methods, Rcpp, utils, bigmemory.sri","methods",null,"bindr (&gt;= 0.1.1), Rcpp (&gt;= 0.12.16)","Rcpp, parallel, grid, graphics, grDevices, stats, utils","methods","dplyr, tidyr, Biobase","methods, stats, utils, dplyr, RSQLite, DBI, rappdirs, curl,\nhttr","methods, utils, graphics, stats, parallel",null,"utils","Rcpp, S4Vectors, stats, methods","stats, utils, futile.logger, parallel, snow","BiocInstaller, pkgDepTools, graph, Rgraphviz, readr, lubridate","bookdown, knitr (&gt;= 1.12), rmarkdown (&gt;= 1.2), stats, utils,\nyaml, BiocManager",null,"Biobase, graph (&gt;= 1.9.26), methods, RBGL (&gt;= 1.13.5), tools,\nutils, XML, RCurl, RUnit","BiocStyle, bookdown, devtools, git2r, httr, knitr, rmarkdown,\nrstudioapi, stringr, tools, utils",null,"utils, XML, RCurl, AnnotationDbi, progress, stringr, httr","graphics, methods, stats, utils","grDevices, stats, scales, Hmisc, RColorBrewer, dichromat,\nBiocGenerics, S4Vectors (&gt;= 0.9.25), IRanges (&gt;= 1.99.28),\nGenomeInfoDb (&gt;= 1.5.14), GenomicRanges (&gt;= 1.23.21),\nSummarizedExperiment, Biostrings (&gt;= 2.33.11), Rsamtools (&gt;=\n1.17.28), GenomicAlignments (&gt;= 1.1.16), GenomicFeatures (&gt;=\n1.21.19), AnnotationDbi, VariantAnnotation (&gt;= 1.11.4),\nensembldb (&gt;= 1.99.13), AnnotationFilter (&gt;= 0.99.8), rlang",null,null,null,"methods, prettyunits","rmarkdown, bookdown (&gt;= 0.9), knitr (&gt;= 1.20), htmltools, yaml\n(&gt;= 2.1.19), httpuv (&gt;= 1.4.0), xfun (&gt;= 0.4), servr (&gt;= 0.11)",null,"htmltools (&gt;= 0.3.6), knitr (&gt;= 1.16), rmarkdown (&gt;= 1.9),\nxfun (&gt;= 0.2), tinytex, yaml (&gt;= 2.1.14)",null,null,"backports, dplyr, generics (&gt;= 0.0.2), methods, nlme, purrr,\nreshape2, stringr, tibble, tidyr","methods, utils, stats, BiocGenerics, S4Vectors, IRanges,\nXVector, GenomeInfoDb, GenomicRanges, Biostrings, Rsamtools,\nrtracklayer","BSgenome","partykit, Cubist (&gt;= 0.2.1)","grDevices, graphics","processx (&gt;= 3.2.1), R6, utils","abind, MASS, mgcv, nnet, pbkrtest (&gt;= 0.4-4), quantreg,\ngrDevices, utils, stats, graphics, maptools, rio, lme4, nlme",null,"Biobase, dplyr, grid, irlba, lattice, magrittr, matter,\nsignal, sp, stats4, utils","foreach, methods, plyr, ModelMetrics (&gt;= 1.1.0), nlme,\nreshape2, stats, stats4, utils, grDevices, recipes (&gt;= 0.1.4),\nwithr (&gt;= 2.0.0)","utils, stats, graph, RBGL, GSEABase, genefilter, annotate, DBI","bitops","rematch, tibble","backports (&gt;= 1.1.0), utils","GlobalOptions (&gt;= 0.1.0), shape, grDevices, utils, stats,\ncolorspace, methods, grid","grDevices, stats, graphics, e1071, class","S4Vectors, IRanges","assertthat, crayon (&gt;= 1.3.4), methods, utils","utils",null,"stats, cluster, graphics, methods","AnnotationDbi, DOSE (&gt;= 3.5.1), enrichplot (&gt;= 0.99.7),\nggplot2, GO.db, GOSemSim, magrittr, methods, plyr, qvalue,\nrvcheck, stats, tidyr, utils","lattice","BiocParallel, cluster, data.table, methods, gplots, graphics,\ngrDevices, RColorBrewer, Rcpp, S4Vectors, SingleCellExperiment,\nstats, SummarizedExperiment, tools, utils","stats, modeltools (&gt;= 0.2-9), mvtnorm (&gt;= 1.0-5), multcomp","graphics, grDevices, stats","ggplot2, htmltools, htmlwidgets (&gt;= 0.7), jsonlite, miniUI (&gt;=\n0.1.1), shiny (&gt;= 0.11.1), shinyjs, utils",null,null,"circlize (&gt;= 0.4.1), GetoptLong, colorspace, RColorBrewer,\nGlobalOptions (&gt;= 0.1.0)","rlang, memoise",null,"digest, stats, utils, jsonlite, rex, httr, crayon, withr (&gt;=\n1.0.2)","grid (&gt;= 3.0.0), gtable (&gt;= 0.1.2), plyr (&gt;= 1.8.2),\ngrDevices, methods, scales, utils",null,"grDevices, graphics, utils, scales","grDevices, methods, utils","htmltools (&gt;= 0.3.5), jsonlite, lazyeval, R6, shiny (&gt;= 0.11),\nggplot2","reshape2","AnnotationHub, ExperimentHub, S4Vectors, utils",null,null,"scales (&gt;= 0.2.5), htmlwidgets, png, base64enc, dendextend (&gt;=\n0.18.0), stats, grDevices","V8 (&gt;= 0.6), jsonlite, utils","MSnbase, RColorBrewer,stats,preprocessCore,Cairo,png,\nlattice,reshape2,gplots,pcaMethods,ggplot2,\nlimma,knitr,tmvtnorm,norm,impute, stringr, grDevices, graphics,\nopenxlsx, utils, cp4p (&gt;= 0.3.5), scales, Matrix, vioplot,\nimp4p (&gt;= 0.7), highcharter (&gt;= 0.5.0), DAPARdata (&gt;= 1.11.2),\nsiggenes, graph, lme4, readxl, clusterProfiler, dplyr,\ntidyr,AnnotationDbi, tidyverse, vsn, FactoMineR, factoextra","utils, knitr, MSnbase","readr (&gt;= 1.0.0), clipr (&gt;= 0.3.0), rstudioapi (&gt;= 0.6)","methods","R6, stringr, methods, DiagrammeR (&gt;= 1.0.0)",null,"assertthat (&gt;= 0.2.0), DBI (&gt;= 1.0.0), dplyr (&gt;= 0.7.5), glue\n(&gt;= 1.2.0), methods, purrr (&gt;= 0.2.5), R6 (&gt;= 2.2.2), rlang (&gt;=\n0.2.0), tibble (&gt;= 1.4.2), tidyselect (&gt;= 0.2.4), utils","Rcpp (&gt;= 0.11.0)","crayon, grDevices","stats","methods, matrixStats (&gt;= 0.53.1), Matrix, S4Vectors (&gt;=\n0.17.5), IRanges, HDF5Array (&gt;= 1.7.10), BiocParallel","utils, stats, datasets, magrittr (&gt;= 1.0.1), ggplot2, fpc,\nwhisker, viridis",null,"stats","ggplot2, dplyr, purrr, readr, tibble, tidyr,\nSummarizedExperiment (&gt;= 1.11.5), MSnbase, limma, vsn, fdrtool,\nggrepel, ComplexHeatmap, RColorBrewer, circlize, shiny,\nshinydashboard, DT, rmarkdown, assertthat, gridExtra, grid,\nstats, imputeLCMD, cluster","assertthat, utils, R6, crayon, rprojroot","BiocGenerics (&gt;= 0.7.5), Biobase, BiocParallel, genefilter,\nmethods, locfit, geneplotter, ggplot2, Hmisc, Rcpp (&gt;= 0.11.0)","callr, cli, digest, git2r (&gt;= 0.23.0), httr (&gt;= 0.4),\njsonlite, memoise (&gt;= 1.0.0), pkgbuild (&gt;= 1.0.2), pkgload (&gt;=\n1.0.1), rcmdcheck (&gt;= 1.3.0), remotes (&gt;= 2.0.0), rstudioapi\n(&gt;= 0.6.0.9000), sessioninfo, stats, tools, usethis (&gt;= 1.4.0),\nutils, withr","dplyr (&gt;= 0.7.4), downloader (&gt;= 0.4), glue (&gt;= 1.2.0),\nhtmltools (&gt;= 0.3.6), htmlwidgets (&gt;= 1.0), igraph (&gt;= 1.1.2),\ninfluenceR (&gt;= 0.1.0), magrittr (&gt;= 1.5), purrr (&gt;= 0.2.4),\nRColorBrewer (&gt;= 1.1-2), readr (&gt;= 1.1.1), rlang (&gt;= 0.2.0),\nrstudioapi (&gt;= 0.7), rgexf (&gt;= 0.15.3), scales (&gt;= 0.5.0),\nstringr (&gt;= 1.3.0), tibble (&gt;= 1.4.2), tidyr (&gt;= 0.8.0),\nviridis (&gt;= 0.5.0), visNetwork (&gt;= 2.0.3)",null,null,"magrittr, methods",null,"stats, grDevices, utils, MASS","methods, AnnotationDbi","utils",null,"AnnotationDbi, BiocParallel, DO.db, fgsea, ggplot2, GOSemSim\n(&gt;= 2.0.0), methods, qvalue, reshape2, S4Vectors, stats, utils",null,"utils, digest","assertthat (&gt;= 0.2.0), bindrcpp (&gt;= 0.2.0.9000), glue (&gt;=\n1.1.1), magrittr (&gt;= 1.5), methods, pkgconfig (&gt;= 2.0.1), R6\n(&gt;= 2.2.2), Rcpp (&gt;= 0.12.19), rlang (&gt;= 0.3.0), tibble (&gt;=\n1.4.2), tidyselect (&gt;= 0.2.3), utils","stats, methods","htmltools (&gt;= 0.3.6), htmlwidgets (&gt;= 1.3), magrittr,\ncrosstalk, promises",null,"graphics, grDevices, class, stats, methods, utils","BiocGenerics (&gt;= 0.7.1), graphics, grDevices, stats, abind,\ntiff, jpeg, png, locfit, fftwtools (&gt;= 0.9-7), utils,\nhtmltools, htmlwidgets, RCurl","graphics, stats, utils, methods, locfit, Rcpp","gtable, grid, grDevices, stats, utils",null,"ggplot2, proto, showtext, sysfonts, utils","methods","AnnotationDbi, cowplot, DOSE (&gt;= 3.5.1), europepmc, ggplot2,\nggplotify, ggraph, ggridges, GOSemSim, graphics, grDevices,\ngrid, gridExtra, igraph, methods, purrr, RColorBrewer,\nreshape2, stats, UpSetR, utils",null,"methods, RSQLite (&gt;= 1.1), DBI, Biobase, GenomeInfoDb,\nAnnotationDbi (&gt;= 1.31.19), rtracklayer, S4Vectors, Rsamtools,\nIRanges (&gt;= 2.13.24), ProtGenerics, Biostrings (&gt;= 2.47.9),\ncurl","httr, jsonlite, plyr, dplyr, progress, urltools, purrr, xml2","methods","stats, grDevices, graphics","filesstrings (&gt;= 3.0.0), magrittr, stringr, styler (&gt;= 1.0.2),\npurrr, roxygen2, checkmate, readr, utils, ore (&gt;= 1.4.0),\nrprojroot, usethis, clisymbols, crayon, fs, glue, rlang","utils, S4Vectors, BiocManager, curl","methods",null,"abind, cluster, dendextend, FactoMineR, ggpubr(&gt;= 0.1.5),\ngrid, stats, reshape2, ggrepel, tidyr","car,cluster,ellipse,flashClust,graphics,grDevices,lattice,leaps,MASS,scatterplot3d,stats,utils",null,"Rcpp (&gt;= 0.12.15)",null,null,"stringdist, purrr, magrittr, tibble","graphics, grDevices, stats",null,"fastmatch, bit",null,"data.table, BiocParallel, stats, ggplot2 (&gt;= 2.2.0),\ngridExtra, grid, fastmatch, Matrix, utils","magrittr, tools, purrr, checkmate, strex (&gt;= 0.1.2),\nmatrixStats, Rcpp, ore (&gt;= 1.4.0), tibble, rlang","httr (&gt;= 1.2.1),\njsonlite (&gt;= \t1.4),\ncurl (&gt;= 2.3)",null,"methods, parallel, stats, stats4, class","tools, jsonlite, htmltools, knitr (&gt;= 1.13), htmlwidgets (&gt;=\n0.6), rmarkdown (&gt;= 1.0), shiny (&gt;= 0.13)","graphics, grid, grDevices, methods, modeltools (&gt;= 0.2-16),\nnnet, stats, stats4",null,"magrittr, rlang, tibble","codetools, utils, iterators",null,null,"MASS, cluster, mclust, flexmix, prabclus, class, diptest,\nmvtnorm, robustbase, kernlab, trimcluster, grDevices, graphics,\nmethods, stats, utils","methods, Rcpp","utils, lambda.r (&gt;= 1.1.0), futile.options",null,"digest, globals (&gt;= 0.12.1), listenv (&gt;= 0.7.0), parallel,\nutils","tibble","gridExtra, lattice, parallel, survival",null,"Biobase, affy (&gt;= 1.23.2), affyio (&gt;= 1.13.3), XVector,\nBiostrings (&gt;= 2.11.32), splines, BiocManager","gtools, stats, methods, utils","Rcpp (&gt;= 0.12.12), withr","stats, methods",null,"S4Vectors (&gt;= 0.9.42), AnnotationDbi, annotate, Biobase,\ngraphics, methods, stats, survival","AnnotationDbi, graphics, grDevices, grid, RColorBrewer, stats,\nutils","methods","stats, stats4, utils, RCurl, GenomeInfoDbData",null,"methods, utils, stats, BiocGenerics, S4Vectors, IRanges,\nGenomicRanges, Biostrings, Rsamtools, BiocParallel","methods, utils, stats, tools, DBI, RSQLite (&gt;= 2.0), RCurl,\nXVector (&gt;= 0.19.7), Biostrings (&gt;= 2.47.6), rtracklayer (&gt;=\n1.39.7), biomaRt (&gt;= 2.17.1), Biobase (&gt;= 2.15.1)","utils, stats, XVector (&gt;= 0.19.8)",null,"httr, readr (&gt;= 1.3.1), xml2, dplyr, tidyr, magrittr, limma","sp","rjson, GlobalOptions (&gt;= 0.1.0), methods","utils, rstudioapi (&gt;= 0.5)","grDevices, grid, gtable (&gt;= 0.2.0), plyr (&gt;= 1.8.3), progress,\nRColorBrewer, reshape (&gt;= 0.8.5), utils, rlang","beeswarm, vipor","grid, grDevices, graphics, stats, utils, gridExtra, scales,\nreshape2, gtable, Hmisc, biovizBase (&gt;= 1.29.2), Biobase,\nS4Vectors (&gt;= 0.13.13), IRanges (&gt;= 2.11.16), GenomeInfoDb (&gt;=\n1.1.3), GenomicRanges (&gt;= 1.29.14), SummarizedExperiment,\nBiostrings, Rsamtools (&gt;= 1.17.28), GenomicAlignments (&gt;=\n1.1.16), BSgenome, VariantAnnotation (&gt;= 1.11.4), rtracklayer\n(&gt;= 1.25.16), GenomicFeatures (&gt;= 1.29.11), OrganismDbi,\nGGally, ensembldb (&gt;= 1.99.13), AnnotationDbi,\nAnnotationFilter, rlang","MASS, ggplot2(&gt;= 0.9.2)","colourpicker (&gt;= 1.0), ggplot2 (&gt;= 2.2.0), grDevices, grid (&gt;=\n3.1.3), gtable (&gt;= 0.2.0), miniUI (&gt;= 0.1.1), scales (&gt;=\n0.2.0), shiny (&gt;= 0.13.0), shinyjs (&gt;= 0.5.2), utils","Rcpp (&gt;= 0.12.2), grid, dplyr, scales, MASS, tweenr (&gt;=\n0.1.5), units (&gt;= 0.4), gtable, lazyeval","ggplotify, grid, jsonlite, magick, methods, rvcheck (&gt;=\n0.1.0), scales, tibble, tools","grid, ggplot2 (&gt;= 3.0.0), htmlwidgets (&gt;= 0.6), stats, xml2\n(&gt;= 1.0.0), htmltools, Rcpp (&gt;= 0.12.12), gdtools (&gt;= 0.1.6),","shiny (&gt;= 0.13),\nminiUI (&gt;= 0.1.1),\nggplot2 (&gt;= 2.2.0),\nggrepel (&gt;= 0.6.9),\nstats,\nutils","proto, RgoogleMaps, png, plyr, reshape2, rjson, mapproj, jpeg,\ngeosphere, digest, scales","digest, grid, gtable (&gt;= 0.1.1), lazyeval, MASS, mgcv, plyr\n(&gt;= 1.7.1), reshape2, rlang (&gt;= 0.2.1), scales (&gt;= 0.5.0),\nstats, tibble, viridisLite, withr (&gt;= 2.0.0)","ggplot2, graphics, grDevices, grid, gridGraphics, rvcheck","ggrepel, grid, ggsci, stats, utils, tidyr, purrr, dplyr(&gt;=\n0.7.1), cowplot, ggsignif, scales, gridExtra, glue, polynom","Rcpp (&gt;= 0.12.2), dplyr, plyr, ggforce, grid, igraph (&gt;=\n1.0.0), scales, MASS, digest, gtable, ggrepel, utils, stats,\nviridis","grid, Rcpp, scales (&gt;= 0.3.0)","ggplot2 (&gt;= 2.2.0), grid (&gt;= 3.0.0), plyr (&gt;= 1.8.0), scales\n(&gt;= 0.4.1), withr (&gt;= 2.1.1)","grDevices, scales, ggplot2 (&gt;= 2.0.0)","ggplot2 (&gt;= 2.0.0)","ggplot2 (&gt;= 3.0.0), plyr, rlang, withr (&gt;= 2.0.0)","ape, dplyr, ggplot2 (&gt;= 3.0.0), grDevices, grid, magrittr,\nmethods, purrr, rlang, rvcheck (&gt;= 0.1.0), scales, tibble,\ntidyr, tidytree (&gt;= 0.1.9), treeio (&gt;= 1.3.14), utils","assertthat, jsonlite (&gt;= 0.9.11), shiny (&gt;= 0.11.1), magrittr,\ndplyr (&gt;= 0.4.0), lazyeval, htmltools (&gt;= 0.2.4), methods","ini, jsonlite, httr","graphics, utils","methods","utils","codetools","methods","stats, methods, grDevices, graphics","methods, AnnotationDbi",null,"cellranger (&gt;= 1.0.0), dplyr (&gt;= 0.4.2), httr (&gt;= 1.1.0),\njsonlite, purrr, readr (&gt;= 0.2.2), stats, stringr, tibble,\ntidyr, utils, xml2 (&gt;= 1.0.0)","AnnotationDbi, GO.db, methods, utils",null,"gtools, gdata, stats, caTools, KernSmooth","stats","stats, stats4, utils","graphics, grid","gtable, grid, grDevices, graphics, utils","grDevices","grDevices, graphics, utils, methods, grid, jsonlite, XML","AnnotationDbi, XML",null,"grid",null,"XVector (&gt;= 0.5.7), rtracklayer (&gt;= 1.25.13), lattice,\nRColorBrewer, biomaRt (&gt;= 2.11.0), AnnotationDbi (&gt;= 1.27.5),\nBiobase (&gt;= 2.15.3), GenomicFeatures (&gt;= 1.17.22), BSgenome (&gt;=\n1.33.1), Biostrings (&gt;= 2.33.11), biovizBase (&gt;= 1.13.8),\nRsamtools (&gt;= 1.17.28), latticeExtra (&gt;= 0.6-26), matrixStats\n(&gt;= 0.8.14), GenomicAlignments (&gt;= 1.1.16), GenomeInfoDb (&gt;=\n1.1.3), BiocGenerics (&gt;= 0.11.3), digest(&gt;= 0.6.8)",null,"forcats (&gt;= 0.2.0), hms, Rcpp (&gt;= 0.11.4), readr (&gt;= 0.1.0),\ntibble","utils, tools, BiocGenerics (&gt;= 0.25.1), S4Vectors, IRanges","ggplot2 (&gt;= 2.2.0), dendextend (&gt;= 1.4.0), magrittr (&gt;=\n1.0.1), reshape2, scales, seriation, utils, stats, grDevices,\nmethods, colorspace, RColorBrewer, htmlwidgets, webshot,\ngplots, assertthat","graphics, grDevices, stats, RColorBrewer","rprojroot (&gt;= 1.2)","lattice, grid, graphics, grDevices, stats, utils","ggimage, ggplot2, grDevices, hexbin, showtext, stats, sysfonts","AnnotationDbi","AnnotationDbi","methods, AnnotationDbi","htmlwidgets, magrittr, purrr, rlist, assertthat, zoo, dplyr\n(&gt;= 0.7.0), tibble (&gt;= 1.1), stringr, broom, xts, quantmod,\ntidyr, htmltools, jsonlite, igraph, lubridate, crosstalk, yaml,\nrlang(&gt;= 0.1.1), whisker","grDevices, tools",null,"methods, latticeExtra, cluster, rpart, nnet, acepack, foreign,\ngtable, grid, gridExtra, data.table, htmlTable (&gt;= 1.11.0),\nviridis, htmltools, base64enc","methods, pkgconfig, rlang","GenomicFeatures, AnnotationDbi","utils","stringr, knitr (&gt;= 1.6), magrittr (&gt;= 1.5), methods,\ncheckmate, htmlwidgets, htmltools, rstudioapi (&gt;= 0.6)","utils, digest, Rcpp","grDevices, htmltools (&gt;= 0.3), jsonlite (&gt;= 0.9.16), yaml","Rcpp (&gt;= 0.11.0), utils, promises, later (&gt;= 0.7.3)","curl (&gt;= 0.9.1), jsonlite, mime, openssl (&gt;= 0.8), R6","methods, AnnotationDbi","Rcpp, digest",null,"rmarkdown (&gt;= 1.6.0),\nknitr,\nhtmltools","graphics, grDevices, magrittr, Matrix, pkgconfig (&gt;= 2.0.0),\nstats, utils","methods, plyr, utils, magrittr, S4Vectors, stats, ggdendro,\nfastcluster, RColorBrewer, htmlwidgets, jsonlite, scales,\nknitr, tools","Rcpp (&gt;=\n0.11.5),methods,stringr,png,jpeg,readbitmap,grDevices,purrr,Cairo,downloader,igraph,plyr","Rcpp (&gt;= 0.12.8)",null,null,"igraph (&gt;= 1.0.1), Matrix (&gt;= 1.1-4), methods, utils",null,"interactiveDisplayBase (&gt;= 1.7.3), shiny, RColorBrewer,\nggplot2, reshape2, plyr, gridSVG, XML, Category, AnnotationDbi","shiny","stats, libcoin (&gt;= 1.0-0)","methods, stats, graphics","rpart (&gt;= 3.1-8), MASS, survival, nnet, class, prodlim","stats4","stats, methods",null,"distr, plyr, biomaRt, ggplot2",null,null,null,null,null,"methods, XML (&gt;= 2.3-0), graph, utils","AnnotationDbi,graph,DBI, graph, grDevices, methods, stats,\ntools, utils","methods, httr, png, Biostrings","methods, stats, grDevices, graphics","graphics, stats, methods","combinat, questionr, grDevices, stats, utils, graphics","Rcpp, DelayedArray, stats","evaluate (&gt;= 0.10), highr, markdown, stringr (&gt;= 0.6), yaml\n(&gt;= 2.1.19), methods, xfun, tools",null,"haven (&gt;= 2.0.0), dplyr, stats","formatR","parallel, grDevices, graphics, stats, utils","Rcpp (&gt;= 0.12.9), rlang","grid, stats, utils, grDevices","grDevices, graphics, methods, numDeriv, stats, survival,\nSQUAREM, utils",null,null,"base64enc, crosstalk, htmlwidgets, htmltools, magrittr,\nmarkdown, methods, png, RColorBrewer, raster, scales (&gt;=\n1.0.0), sp, stats, viridis (&gt;= 0.5.1)",null,"stats, mvtnorm","grDevices, graphics, stats, utils, methods",null,"graphics, grid, splines, utils, parallel, MASS, lattice, nlme\n(&gt;= 3.1-123), minqa (&gt;= 1.1.15), nloptr (&gt;= 1.0.4)","crayon, Rcpp, rlang (&gt;= 0.3.0)","lattice",null,"stringr, Rcpp (&gt;= 0.12.13),",null,"Rcpp (&gt;= 0.12.12), magrittr, knitr, curl",null,"parallel","base64enc, digest, readBrukerFlexData (&gt;= 1.7), readMzXmlData\n(&gt;= 2.7), XML","shiny (&gt;= 1.0.3), miniUI, htmltools, htmlwidgets, knitr,\nmethods, tools, base64enc, grDevices, codetools, webshot","stats, graphics","graphics, utils","foreign (&gt;= 0.8), methods, grid, lattice, stats, utils,\ngrDevices",null,"utils, mime (&gt;= 0.3)",null,null,"stats, methods, Matrix (&gt;= 1.1-5)",null,"BiocGenerics, digest, irlba, utils","Matrix, survival, splines, lattice, nnls, quadprog, utils,\ngraphics, grDevices, partykit (&gt;= 1.2-1)",null,"stats, utils, graphics, grDevices","stats, compiler","graphics, grDevices, lattice, methods, utils, mcmc, quantreg","digest (&gt;= 0.6.3)","MASS, Matrix, Rcpp, glmnet, kinship2, methods, mvtnorm,\nparallel",null,"graphics, MASS, numDeriv, robustbase","graphics, stats","tools","shiny (&gt;= 0.13), htmltools (&gt;= 0.3), utils",null,"Rcpp (&gt;= 0.9.10)","MASS, segmented, stats, survival",null,"gdata, pls, sfsmisc, MASS, rpart, rda, genefilter, fpc, ggvis,\nshiny, gbm, RColorBrewer, hwriter, threejs (&gt;= 0.2.2), mlbench,\nstats4, tools, grDevices, graphics, stats","BBmisc (&gt;= 1.11), backports (&gt;= 1.1.0), ggplot2, stats,\nstringi, checkmate (&gt;= 1.8.2), data.table, methods, parallelMap\n(&gt;= 1.3), survival, utils, XML",null,"Rcpp, data.table","broom, dplyr, magrittr, purrr (&gt;= 0.2.2), tibble, tidyr (&gt;=\n0.8.0), rlang (&gt;= 0.2.0)","methods",null,"shiny, mzID (&gt;= 1.2), MSGFplus, shinyFiles (&gt;= 0.4.0), tools","mzID, ProtGenerics","tufte, htmltools, xfun, knitr (&gt;= 1.18), rmarkdown, stringr,\nmagrittr","MASS, gplots, RColorBrewer","edgeR, qvalue","BiocParallel, IRanges (&gt;= 2.13.28), plyr, preprocessCore, vsn,\ngrid, stats4, affy, impute, pcaMethods, MALDIquant (&gt;= 1.16),\nmzID (&gt;= 1.5.2), digest, lattice, ggplot2, XML, scales, MASS,\nRcpp","MSnbase (&gt;= 1.12.1), mzID (&gt;= 1.3.5), R.cache, foreach,\ndoParallel, parallel, methods, iterators, data.table, Biobase,\nProtGenerics, reshape2, dplyr, mzR","Biobase,\nDT,\nfdrtool,\ngraphics,\ngrDevices,\nlimma,\nMASS,\nMatrix,\nmethods,\nMSnbase,\nnumDeriv,\nopenxlsx,\nplyr,\npreprocessCore,\nshiny,\nshinyjs,\nshinythemes,\nstatmod,\nsvDialogs,\nutils,\nvsn,\nminiUI,\nhtmltools,\nmzR,\nProtGenerics,\nBiocInstaller,\nBiocParallel,\nsnow,\nhttpuv,\nxtable,\nhtmlwidgets,\ncolorspace,\nRColorBrewer,\ndichromat,\nmunsell,\nlabeling,\naffyio,\nzlibbioc,\ndoParallel,\ngtable,\nscales,\ntibble,\nlazyeval,\nreshape2,\naffy,\nimpute,\npcaMethods,\nmzID,\nMALDIquant,\nggplot2,\nXML,\nminqa,\nnloptr,\nmagrittr,\nyaml,\nstringr,\nstringi,\nforeach,\niterators,\nRcpp,\nBiocGenerics","lme4, marray, limma, gplots, ggplot2, methods, grid, ggrepel,\npreprocessCore, reshape2, survival, minpack.lm, utils,\ngrDevices, graphics, stats, doSNOW, snow, foreach, data.table,\nMASS, dplyr, tidyr, stringr, randomForest",null,null,"sandwich (&gt;= 2.3-0), codetools","methods, GenomicRanges (&gt;= 1.25.93), BiocGenerics, S4Vectors\n(&gt;= 0.17.5), IRanges, Biobase, stats, tidyr, utils","survival, MASS, stats4","colorspace, methods","stats, methods",null,null,"XML, plyr, parallel, doParallel, foreach, iterators,\nProtGenerics","Biobase, BiocGenerics (&gt;= 0.13.6), ProtGenerics (&gt;= 1.9.1),\nncdf4",null,null,null,null,"graphics, stats, stringr (&gt;= 1.0.0), digest, grid, grDevices,\ngridBase, colorspace, RColorBrewer, foreach, doParallel,\nggplot2, reshape2",null,null,null,null,"Rcpp (&gt;= 0.12.12), R6, grDevices, base64enc, zip,\ndigest,uuid,utils,stats, magrittr,htmltools, xml2 (&gt;= 1.1.0)","askpass","methods, Rcpp, grDevices, stats, utils, zip",null,"Biobase, BiocManager, GenomicRanges (&gt;= 1.31.13), graph,\nIRanges, RBGL, DBI, S4Vectors (&gt;= 0.9.25), stats","methods, AnnotationDbi",null,"methods, AnnotationDbi","tools, utils",null,"grDevices, graphics, methods, utils, stats, digest, tools,\nRcpp","BBmisc (&gt;= 1.8), checkmate (&gt;= 1.5.1), parallel","backports, BBmisc (&gt;= 1.10), checkmate (&gt;= 1.8.2), fastmatch,\nmethods",null,"survival (&gt;= 2.37-7), coin (&gt;= 1.1-0), zoo, sandwich (&gt;=\n1.1-1)","grDevices, stats, utils, survival, Formula (&gt;= 1.2-1), inum\n(&gt;= 1.0-0), rpart (&gt;= 4.1-11)","ggplot2,\ngtable,\ngrid,\nstats,\ngrDevices,\nutils","KEGGgraph, XML, Rgraphviz, graph, png, AnnotationDbi,\nKEGGREST, methods, utils","cleaver (&gt;= 1.3.6), Biobase, Biostrings (&gt;= 2.47.5), IRanges\n(&gt;= 2.13.11), S4Vectors (&gt;= 0.17.24), mzID, mzR (&gt;= 1.99.1),\nMSnbase (&gt;= 1.15.5), Pviz, biomaRt, GenomicRanges (&gt;= 1.31.7),\nrtracklayer (&gt;= 1.39.6), ensembldb (&gt;= 1.99.13), BiocParallel,\nAnnotationFilter","Matrix (&gt;= 1.2.3), parallel, MASS, methods","BiocGenerics, Rcpp (&gt;= 0.11.3), MASS","grid, RColorBrewer, scales, gtable, stats, grDevices, graphics","cli (&gt;= 1.0.0), crayon (&gt;= 1.3.4), fansi (&gt;= 0.4.0), methods,\nrlang (&gt;= 0.3.0.1), utf8 (&gt;= 1.1.3)",null,"callr (&gt;= 2.0.0), cli, crayon, desc, prettyunits, R6,\nrprojroot, withr (&gt;= 2.1.2)","utils","graph, RBGL","callr (&gt;= 2.0.2), cli, crayon, desc, digest, evaluate, fs (&gt;=\n1.2.0), fansi, highlight, httr, magrittr, MASS, memoise,\nopenssl, pkgload (&gt;= 1.0.2), purrr, processx, R6, rematch2,\nrlang (&gt;= 0.2.0), rmarkdown (&gt;= 1.1.9007), roxygen2, rsconnect,\nrstudioapi, tibble, tools, whisker, withr, xml2 (&gt;= 1.1.1),\nyaml","desc, methods, pkgbuild, rlang, rprojroot, rstudioapi, utils,\nwithr","tools, grDevices, utils, graphics, codetools, digest, stringi,\nmagrittr, stringr, xtable, withr, bibtex (&gt;= 0.4)",null,null,"tools, scales, httr, jsonlite, magrittr, digest, viridisLite,\nbase64enc, htmltools, htmlwidgets (&gt;= 1.2), tidyr, hexbin,\nRColorBrewer, dplyr, tibble, lazyeval (&gt;= 0.2.0), rlang,\ncrosstalk, purrr, data.table, promises","grDevices, graphics, methods, stats","Rcpp (&gt;= 0.11.0)",null,"stats, graphics","stats, graphics","stats",null,null,"Rcpp (&gt;= 0.12.2), ggplot2 (&gt;= 2.1.0), assertthat (&gt;= 0.1),\ngrid, gridExtra (&gt;= 2.0.0), methods, data.table (&gt;= 1.10.4)","stats","magrittr, assertthat, methods","ps, R6, utils","Rcpp (&gt;= 0.11.5), stats, graphics, survival, KernSmooth, lava","hms, prettyunits, R6, crayon","Biobase, mclust (&gt;= 4.3), caret, e1071, sampling, class,\nkernlab, lattice, nnet, randomForest, proxy, FNN, hexbin,\nBiocGenerics, stats, dendextend, RColorBrewer, scales, MASS,\nknitr, mvtnorm, LaplacesDemon, coda, mixtools, gtools, plyr,\nggplot2, biomaRt, utils, grDevices, graphics","Biobase, utils","shiny (&gt;= 0.9.1), scales, dplyr, DT (&gt;= 0.1.40), graphics,\nutils, ggplot2","R6, Rcpp, later, rlang, stats, magrittr","mzR (&gt;= 2.3.2), MSnbase, Biostrings, GenomeInfoDb, utils,\nBiobase, BiocManager, RCurl",null,"graph, RBGL, mvtnorm",null,"Rcpp (&gt;= 0.12.8)","stats, utils","AhoCorasickTrie, Matrix, doParallel, foreach, plyr, readr,\nseqinr, stringr, dplyr","stringr, codetools, methods, Rcpp (&gt;= 0.11.0)","utils","mnormt,parallel,stats,graphics,grDevices,methods,foreign,lattice,nlme","magrittr (&gt;= 1.5), rlang (&gt;= 0.3.1)","biovizBase, Biostrings, GenomicRanges, IRanges, data.table,\nmethods","stats, graphics","msdata (&gt;= 0.17.1)",null,"Biobase, methods, knitr, tools, Nozzle.R1, xtable, pander,\nS4Vectors",null,"curl","methods, graphics, Matrix, MatrixModels","shiny (&gt;= 1.0.5), miniUI, rstudioapi, highr, classInt,\nhtmltools, graphics, stats, utils, labelled (&gt;= 2.0.0)","splines, ggplot2, grid, reshape2",null,"RColorBrewer","igraph,RCurl,png,graph",null,"Rcpp (&gt;= 0.11.2), Matrix",null,null,"Rcpp, methods","doParallel, foreach, Rcpp, progress, raster, scales, png,\nmagrittr, rgl, imager","methods","XML","utils, R.methodsS3 (&gt;= 1.7.1), R.oo (&gt;= 1.21.0), R.utils (&gt;=\n2.6.0), digest (&gt;= 0.6.13)","ggplot2, scales","callr (&gt;= 2.0.0), cli, crayon, desc (&gt;= 1.2.0), digest,\npkgbuild, prettyunits, R6, rprojroot, sessioninfo (&gt;= 1.1.1),\nutils, withr, xopen",null,"methods, utils","methods, Rcpp (&gt;= 0.11.3)","Rcpp (&gt;= 0.11.0), stats, utils, methods","Rcpp (&gt;= 0.10.2), methods","Matrix (&gt;= 1.1-0), Rcpp (&gt;= 0.11.0), stats, utils",null,"Rcpp",null,"methods, XMLRPC, BiocGenerics",null,null,"bigmemory (&gt;= 4.0.0), parallel","bmp, jpeg, png, tiff",null,"base64enc, digest, XML","Rcpp (&gt;= 0.12.0.5), tibble, hms (&gt;= 0.4.1), R6, clipr, crayon,\nmethods","cellranger, progress, Rcpp (&gt;= 0.12.18), tibble (&gt;= 1.3.1),\nutils","generics, gower, ipred, lubridate, magrittr, Matrix, purrr (&gt;=\n0.2.3), RcppRoll, rlang (&gt;= 0.3.0.1), stats, tibble, tidyr,\ntidyselect (&gt;= 0.1.1), timeDate, utils, withr","e1071, rpart, ada, ipred, stats, evd, methods, data.table (&gt;=\n1.7.8), nnet, xtable","utils",null,"tibble","methods, utils","rJava","callr (&gt;= 2.0.0), clipr (&gt;= 0.4.0), fs, rlang, rmarkdown,\ntools, utils, whisker, withr","tibble, utils","plyr","plyr (&gt;= 1.8.1), Rcpp, stringr","rmarkdown (&gt;= 1.0)","magrittr, lazyeval","R.utils, shiny, biocViews, BiocManager",null,"graphics, grDevices, stats, utils, htmlwidgets, htmltools,\nknitr, jsonlite (&gt;= 0.9.20), shiny, magrittr, crosstalk,\nmanipulateWidget (&gt;= 0.9.0)","graphics, stats, utils, grDevices, methods, png","stats4, graphics, grDevices","Rhdf5lib (&gt;= 1.3.2)",null,"assertthat, callr, clisymbols, crayon, desc, httr, jsonlite,\nparsedate, prettyunits, R6, rappdirs, rcmdcheck (&gt;= 1.2.1),\nrematch, utils, whoami, withr","tools, stats, utils, foreign, haven (&gt;= 1.1.0), curl (&gt;= 0.6),\ndata.table (&gt;= 1.9.8), readxl (&gt;= 0.1.1), openxlsx, tibble",null,null,"methods",null,"yaml, jsonlite, XML, data.table","tools, utils, knitr (&gt;= 1.18), yaml (&gt;= 2.1.19), htmltools (&gt;=\n0.3.5), evaluate (&gt;= 0.8), base64enc, jsonlite, mime, tinytex\n(&gt;= 0.3), methods, stringr (&gt;= 1.2.0)","shiny (&gt;= 0.10.0),\nhtmltools (&gt;= 0.2.4),\nknitr (&gt;= 1.5.33),\nRJSONIO","rmarkdown (&gt;= 0.3.3)","utils","parallel","methods, quantreg, rpart, nlme (&gt;= 3.1-123), polspline,\nmultcomp, htmlTable (&gt;= 1.11.0), htmltools",null,"methods","stringr, digest, utils, stats, parallel","stats, graphics, utils, methods, DEoptimR",null,"httr, progress, jsonlite, utils, Biobase, BiocGenerics (&gt;=\n0.23.1)","methods, utils","utils, tools, methods, brew","brew, commonmark, desc (&gt;= 1.2.0), digest, methods, pkgload\n(&gt;= 1.0.2), purrr, R6 (&gt;= 2.1.2), Rcpp (&gt;= 0.11.0), stringi,\nstringr (&gt;= 1.0.0), utils, xml2",null,"backports","xml2, RCurl, utils","utils, BiocGenerics (&gt;= 0.25.1), S4Vectors (&gt;= 0.17.25),\nIRanges (&gt;= 2.13.12), XVector (&gt;= 0.19.7), zlibbioc, bitops,\nBiocParallel","openssl, RCurl, jsonlite, packrat (&gt;= 0.4.8-1), yaml (&gt;=\n2.1.5), rstudioapi (&gt;= 0.5)","bit64, blob (&gt;= 1.1.1), DBI (&gt;= 1.0.0), memoise, methods,\npkgconfig, Rcpp (&gt;= 0.12.7)",null,"methods","XML (&gt;= 1.98-0), BiocGenerics (&gt;= 0.25.1), S4Vectors (&gt;=\n0.19.22), IRanges (&gt;= 2.13.13), XVector (&gt;= 0.19.7),\nGenomeInfoDb (&gt;= 1.15.2), Biostrings (&gt;= 2.47.6), zlibbioc,\nRCurl (&gt;= 1.4-2), Rsamtools (&gt;= 1.31.2), GenomicAlignments (&gt;=\n1.15.6), tools","msmsTests","Rcpp (&gt;= 0.11.0), stats",null,"methods, utils, tools, R.methodsS3 (&gt;= 1.7.1)","utils, rlang","httr (&gt;= 0.5), selectr, magrittr","grDevices, Rcpp (&gt;= 0.12.12), officer (&gt;= 0.2.0), gdtools (&gt;=\n0.1.6), xml2 (&gt;= 1.0.0)","MSnbase, Biobase, SummarizedExperiment, pRolocdata,\ncuratedTCGAData, ALL, tidyverse, utils, rmarkdown",null,"MASS, lpSolve","stats, utils, zoo","labeling, munsell (&gt;= 0.5), R6, RColorBrewer, Rcpp,\nviridisLite","BiocGenerics, SummarizedExperiment, Matrix, dplyr, ggbeeswarm,\ngrid, DelayedMatrixStats, methods, plyr, reshape2, S4Vectors,\nstats, utils, viridis, Rcpp (&gt;= 0.12.14), DelayedArray","grDevices, graphics, stats","R.cache, dplyr, httr, rvest, stringr, xml2, tidygraph, ggraph,\nggplot2","SummarizedExperiment, S4Vectors, BiocGenerics, Rcpp (&gt;=\n0.12.14), stats, methods, utils, Matrix, scater, edgeR, limma,\ndynamicTreeCut, BiocNeighbors, igraph, statmod, DelayedArray,\nDelayedMatrixStats","assertthat, openssl, rprojroot, curl, jsonlite",null,"methods, stringr, R6","ade4,segmented",null,"TSP, qap, grid, cluster, gclus, dendextend, colorspace, MASS,\ngplots, registry, methods, stats, grDevices","mime (&gt;= 0.2), httpuv (&gt;= 1.4.0), xfun, jsonlite","cli, tools, utils, withr","grDevices, methods, utils, stats","stats, graphics, grDevices","utils, grDevices, httpuv (&gt;= 1.4.4), mime (&gt;= 0.3), jsonlite\n(&gt;= 0.9.16), xtable, digest, htmltools (&gt;= 0.3.5), R6 (&gt;= 2.0),\nsourcetools, later (&gt;= 0.7.2), promises (&gt;= 1.0.1), tools,\ncrayon, rlang","utils, shiny (&gt;= 1.0.0), htmltools (&gt;= 0.2.6), promises","htmltools, jsonlite, tools, shiny (&gt;= 1.1.0), fs (&gt;= 1.2.6),\ntibble (&gt;= 1.4.2)","digest (&gt;= 0.6.8), htmltools (&gt;= 0.2.9), jsonlite, shiny (&gt;=\n1.0.0)","shiny (&gt;= 0.11)","grDevices","sysfonts (&gt;= 0.7.1), utils","stats4","MASS, graphics, grDevices, stats, utils","methods, utils, grDevices, graphics, stats, BiocGenerics,\nBiobase, affy, genefilter, gcrma","S4Vectors, methods, BiocGenerics, utils, stats",null,null,null,null,"utils, stats, graphics, grDevices, lattice, grid","graphics, stats, utils",null,"curl,\npurrr,\ndplyr,\nxml2,\nutils,\npingr,\nurltools,\njsonlite,\ncrayon,\ncli",null,"graphics, grDevices, utils",null,"stats, graphics","ISOcodes","Rcpp, magrittr, checkmate, matrixStats, ore (&gt;= 1.4.0), purrr,\nrlang, tibble, glue (&gt;= 1.3.0)","rlang,\ncrayon","parallel","tools, utils, stats","glue (&gt;= 1.2.0), magrittr, stringi (&gt;= 1.1.7)","graphics, stats","backports, cli, magrittr, purrr (&gt;= 0.2.3), rematch2, rlang,\nrprojroot, tibble (&gt;= 1.4.2), tools, withr, xfun","Rcpp","utils, stats, tools, Matrix, BiocGenerics (&gt;= 0.15.3),\nS4Vectors (&gt;= 0.17.25), IRanges (&gt;= 2.13.16), GenomeInfoDb (&gt;=\n1.13.1)","matrixStats, stats, graphics, utils, limma,","svGUI (&gt;= 1.0.0), utils, methods, rstudioapi (&gt;= 0.7)","Rcpp, gdtools (&gt;= 0.1.6)",null,"methods","RColorBrewer, lattice, qvalue, multtest, utils, tools,\nBiobase, knitr, Biostrings, cleaver (&gt;= 1.3.3), readr (&gt;= 0.2),\nrmarkdown (&gt;= 1.0)","utils",null,null,null,"cli, crayon, digest, magrittr, methods, praise, R6 (&gt;= 2.2.0),\nrlang, withr (&gt;= 2.0.0)",null,"htmlwidgets (&gt;= 0.3.2), base64enc, crosstalk, methods, stats","cli (&gt;= 1.0.1), crayon (&gt;= 1.3.4), fansi (&gt;= 0.4.0), methods,\npillar (&gt;= 1.3.1), pkgconfig (&gt;= 2.0.2), rlang (&gt;= 0.3.1),\nutils","tidyr, tibble","tibble, dplyr (&gt;= 0.7), igraph, magrittr, utils, rlang, R6,\nRcpp, tools, stats, tidyr, pillar","dplyr (&gt;= 0.7.0), glue, magrittr, purrr, Rcpp, rlang, stringi,\ntibble, tidyselect (&gt;= 0.2.5)","glue (&gt;= 1.3.0), purrr, rlang (&gt;= 0.2.2), Rcpp (&gt;= 0.12.0)","rlang, dplyr, stringr, hunspell, broom, Matrix, tokenizers,\njaneaustenr, purrr (&gt;= 0.1.1), methods, stopwords","ape, dplyr, lazyeval, magrittr, methods, rlang, tibble, utils","broom (&gt;= 0.4.2), cli (&gt;= 1.0.0), crayon (&gt;= 1.3.4), dplyr (&gt;=\n0.7.4), dbplyr (&gt;= 1.1.0), forcats (&gt;= 0.2.0), ggplot2 (&gt;=\n2.2.1), haven (&gt;= 1.1.0), hms (&gt;= 0.3), httr (&gt;= 1.3.1),\njsonlite (&gt;= 1.5), lubridate (&gt;= 1.7.1), magrittr (&gt;= 1.5),\nmodelr (&gt;= 0.1.1), purrr (&gt;= 0.2.4), readr (&gt;= 1.1.1), readxl\n(&gt;= 1.0.0), reprex (&gt;= 0.1.1), rlang (&gt;= 0.1.4), rstudioapi (&gt;=\n0.7), rvest (&gt;= 0.3.2), stringr (&gt;= 1.2.0), tibble (&gt;= 1.3.4),\ntidyr (&gt;= 0.7.2), xml2 (&gt;= 1.1.1)",null,null,"xfun (&gt;= 0.3)","stats, methods","methods, stats, utils, grDevices, htmlwidgets, jsonlite,\ndata.table, Biobase, GenomeInfoDb, IRanges, S4Vectors, knitr","stringi (&gt;= 1.0.1), Rcpp (&gt;= 0.12.3), SnowballC (&gt;= 0.5.1)","ape, dplyr, jsonlite, magrittr, methods, rlang, rvcheck,\ntibble, tidytree (&gt;= 0.1.7), utils","Rcpp",null,null,null,"graphics, foreach, utils, methods, stats, grDevices","xts (&gt;= 0.10-0), zoo, curl","htmltools, xfun, knitr (&gt;= 1.18), rmarkdown","Rcpp (&gt;= 0.12.3), grDevices, farver, magrittr, rlang","AnnotationDbi","AnnotationDbi","utils, stats","httr, curl",null,null,"httr,\nutils,\njsonlite,\nxml2,\nmethods","Rcpp","ggplot2, gridExtra, plyr, utils, stats, methods, grDevices,\nscales","Rcpp, methods, triebeard","clipr (&gt;= 0.3.0), clisymbols, crayon, curl (&gt;= 2.7), desc, fs\n(&gt;= 1.2.0), gh, git2r (&gt;= 0.23), glue, rlang, rprojroot (&gt;=\n1.2), rstudioapi, utils, whisker",null,null,"Rcpp (&gt;= 0.12), jsonlite (&gt;= 1.0), curl (&gt;= 1.0), utils","utils, DBI, zlibbioc, Biobase, S4Vectors (&gt;= 0.17.24), IRanges\n(&gt;= 2.13.13), XVector (&gt;= 0.19.7), Biostrings (&gt;= 2.47.6),\nAnnotationDbi (&gt;= 1.27.9), rtracklayer (&gt;= 1.39.7), BSgenome\n(&gt;= 1.47.3), GenomicFeatures (&gt;= 1.31.3)",null,"graph, RBGL, grid, lattice, RColorBrewer, reshape2,  gtools",null,null,"stats, graphics","stats, ggplot2 (&gt;= 1.0.1), gridExtra",null,"htmlwidgets, htmltools, jsonlite, magrittr, utils, methods,\ngrDevices, stats","methods, affy, limma, lattice, ggplot2",null,"magrittr, jsonlite, callr",null,"httr, jsonlite, utils","stats,\ngraphics,\ngrDevices","htmltools, knitr (&gt;= 1.16), servr (&gt;= 0.5), xfun (&gt;= 0.2),\nrmarkdown","mzR (&gt;= 2.13.1), BiocGenerics, ProtGenerics, lattice,\nRColorBrewer, plyr, RANN, multtest, MassSpecWavelet (&gt;= 1.5.2),\nS4Vectors, robustbase","tools","rJava, xlsxjars, grDevices, utils",null,null,"Rcpp","methods, RCurl, XML","processx","methods, utils, jsonlite","stats, utils","methods","methods, utils, zlibbioc, BiocGenerics, S4Vectors, IRanges","GO.db, RamiGO, AnnotationDbi, graphics, utils, DBI",null,"stats4","utils",null,null,"utils, graphics, grDevices, lattice (&gt;= 0.20-27)",null,null,"MASS","graphics, grDevices, stats, utils",null,null,null,"methods, utils, stats","grDevices",null,"grDevices, utils",null,"grid, grDevices, graphics, stats, utils","methods","methods, graphics, grid, stats, utils, lattice","utils, stats","methods, stats, graphics, Matrix","graphics, stats, utils, lattice",null,"tools, compiler",null,null,"graphics, stats","utils, grDevices, graphics","graphics, methods, stats","graphics, Matrix, methods, splines, stats, utils","utils",null,null],[null,null,null,null,null,null,"preprocessCore",null,null,"preprocessCore","Rcpp",null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,"Rcpp, RcppArmadillo",null,"Rcpp, Rhdf5lib",null,null,null,null,null,null,"BH, Rcpp",null,null,"plogr, Rcpp","Rcpp",null,null,null,null,null,null,"Rcpp, RcppAnnoy","BH",null,null,null,null,null,null,null,"S4Vectors, IRanges, XVector (&gt;= 0.21.4)",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp, BH",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"BH, Rcpp",null,"S4Vectors",null,null,null,null,null,null,"Rcpp, RcppArmadillo",null,null,null,null,null,null,null,null,null,null,null,null,null,"BH (&gt;= 1.58.0-1), bindrcpp (&gt;= 0.2.0.9000), plogr (&gt;=\n0.1.10), Rcpp (&gt;= 0.12.15)",null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,"Rcpp","Rcpp",null,null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,"S4Vectors, IRanges",null,"S4Vectors, IRanges",null,null,null,null,null,null,null,null,null,null,"Rcpp",null,"Rcpp, gdtools",null,null,null,null,null,"Rcpp","Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp",null,"Rcpp, BH, later",null,null,"Rcpp (&gt;= 0.12.12)",null,null,null,null,"Rcpp","Rcpp",null,null,null,null,null,null,null,null,null,"S4Vectors","Matrix",null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,"Rcpp, BH",null,null,null,null,null,null,"mvtnorm",null,null,"Rcpp (&gt;= 0.10.5), RcppEigen","Rcpp",null,null,"Rcpp,",null,"Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp, RcppArmadillo",null,null,null,null,null,null,"Rcpp",null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp, zlibbioc, Rhdf5lib (&gt;= 1.1.4)",null,null,null,null,null,null,null,null,null,"Rcpp",null,"Rcpp",null,null,null,null,null,null,null,"Rcpp",null,null,null,"mvtnorm",null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,"Rcpp",null,null,null,"Rcpp",null,"Rcpp, RcppArmadillo",null,null,"later, Rcpp",null,null,null,null,"Rcpp",null,null,"Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp, RcppEigen",null,null,"Rcpp","Rcpp, progress",null,null,null,null,null,null,null,"Rcpp","Rcpp","Rcpp","Rcpp",null,"Rcpp",null,null,null,"RcppClassic, Rcpp",null,null,null,null,"Rcpp, BH","progress, Rcpp",null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,"Rhdf5lib",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,"S4Vectors, IRanges, XVector, Biostrings",null,"BH, plogr (&gt;= 0.2.0), Rcpp",null,"Rcpp","S4Vectors, IRanges, XVector",null,"Rcpp",null,null,null,null,"Rcpp, gdtools",null,null,null,null,"Rcpp","Rhdf5lib, Rcpp, beachmat",null,null,"beachmat, Rcpp, Rhdf5lib",null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,"Rcpp, RcppArmadillo",null,null,null,"Rcpp, gdtools, BH",null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp","Rcpp","Rcpp (&gt;= 0.12.0),",null,null,null,null,null,null,null,null,"Rcpp",null,"Rcpp",null,null,null,null,"xts",null,"Rcpp",null,null,null,null,null,null,null,"Rcpp (&gt;= 0.12.10)",null,"Rcpp",null,null,null,"Rcpp","S4Vectors, IRanges, XVector, Biostrings, Rsamtools",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp (&gt;= 0.12.12)",null,null,null,null,"zoo","S4Vectors, IRanges",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,"testthat",null,null,"knitr","ade4TkGUI, adegraphics, adephylo, ape, CircStats, deldir,\nlattice, pixmap, sp, spdep, splancs, waveslim","tkWidgets (&gt;= 1.19.0), affydata, widgetTools","hgu95av2cdf, hgu133acdf",null,"affydata, MASS","Biostrings, microbenchmark","testthat","rpart","Biobase","MASS, class, testit","hgu95av2.db, genefilter, Biostrings (&gt;= 2.25.10), IRanges,\nrae230a.db, rae230aprobe, tkWidgets, GO.db, org.Hs.eg.db,\norg.Mm.eg.db, hom.Hs.inp.db, humanCHRLOC, Rgraphviz, RUnit,","hgu95av2.db, GO.db, org.Sc.sgd.db, org.At.tair.db, KEGG.db,\nRUnit, TxDb.Hsapiens.UCSC.hg19.knownGene, hom.Hs.inp.db,\norg.Hs.eg.db, reactome.db, AnnotationForge, graph,\nEnsDb.Hsapiens.v75, BiocStyle, knitr","BiocStyle, knitr, testthat, RSQLite, org.Hs.eg.db","biomaRt, httr, GenomeInfoDb (&gt;= 1.17.1), Biostrings, affy,\nhgu95av2.db, human.db0, org.Hs.eg.db, Homo.sapiens,\nhom.Hs.inp.db, GO.db, BiocStyle, knitr, BiocManager","IRanges, GenomicRanges, GenomeInfoDb, VariantAnnotation,\nRsamtools, rtracklayer, BiocStyle, knitr, AnnotationForge,\nrBiopaxParser, RUnit, GenomicFeatures, MSnbase, mzR,\nBiostrings, SummarizedExperiment, ExperimentHub, gdsfmt","RUnit, knitr, BiocStyle, grasp2db, GenomeInfoDbData","gee, expm, igraph","testthat","testthat","testthat",null,"cowplot, ggplot2, grid, knitr, prettydoc",null,"BiocStyle, knitr, rmarkdown","testthat, microbenchmark, codetools","testthat, BiocStyle, knitr, rmarkdown, Matrix, devtools","vioplot, lattice",null,null,"testthat","isa2","RSQLite, RODBC","knitr, testthat",null,"testthat","testthat","XML, RCurl, lattice, ncdf4, igraph, bigmemory, knitr,\ntestthat (&gt;= 0.9.1), httr","tools, tkWidgets, ALL, RUnit, golubEsets","limma, DESeq2, airway, ggplot2, plyr, GenomicRanges,\ntestthat, magrittr, edgeR, qvalue, knitr, data.table, MSnbase,\nSummarizedExperiment","testthat, knitr, BiocStyle, rmarkdown, rtracklayer","Biobase, S4Vectors, IRanges, GenomicRanges, Rsamtools,\nAnnotationDbi, oligoClasses, oligo, affyPLM, flowClust, affy,\nDESeq2, MSnbase, annotate, RUnit","remotes, RUnit, BiocGenerics","BiocStyle, BiocVersion, remotes, testthat, knitr, withr","testthat, BiocStyle, knitr, rmarkdown, FNN, RcppAnnoy","BiocGenerics, tools, foreach, BatchJobs, BBmisc, doParallel,\nRmpi, GenomicRanges, RNAseqData.HNRNPC.bam.chr14,\nTxDb.Hsapiens.UCSC.hg19.knownGene, VariantAnnotation,\nRsamtools, GenomicAlignments, ShortRead, codetools, RUnit,\nBiocStyle, knitr, batchtools, data.table","testthat, Biobase","BiocGenerics, RUnit, htmltools",null,"BiocGenerics, knitr, commonmark",null,"locfit","annotate, BiocStyle, knitr, rmarkdown, testthat","BSgenome (&gt;= 1.13.14), BSgenome.Celegans.UCSC.ce2 (&gt;=\n1.3.11), BSgenome.Dmelanogaster.UCSC.dm3 (&gt;= 1.3.11),\nBSgenome.Hsapiens.UCSC.hg18, drosophila2probe, hgu95av2probe,\nhgu133aprobe, GenomicFeatures (&gt;= 1.3.14), hgu95av2cdf, affy\n(&gt;= 1.41.3), affydata (&gt;= 1.11.5), RUnit","BSgenome.Hsapiens.UCSC.hg19,\nTxDb.Hsapiens.UCSC.hg19.knownGene, BSgenome, rtracklayer,\nEnsDb.Hsapiens.v75, RUnit",null,null,null,"covr, pillar (&gt;= 1.2.1), testthat","testit, shiny, miniUI, stringr, rstudioapi, tools, processx,\nlater","pixmap, testthat","htmlwidgets, rstudioapi, miniUI, rsconnect (&gt;= 0.4.3), servr\n(&gt;= 0.3), shiny, testit (&gt;= 0.5), tufte, webshot",null,null,"AER, akima, AUC, bbmle, betareg, biglm, binGroup, boot, brms,\nbtergm, car, caret, coda, covr, e1071, emmeans, ergm, gam (&gt;=\n1.15), gamlss, gamlss.data, gamlss.dist, geepack, ggplot2,\nglmnet, gmm, Hmisc, irlba, joineRML, Kendall, knitr, ks,\nLahman, lavaan, lfe, lme4, lmodel2, lmtest, lsmeans, maps,\nmaptools, MASS, Matrix, mclust, mgcv, muhaz, multcomp, network,\nnnet, orcutt (&gt;= 2.2), ordinal, plm, plyr, poLCA, psych,\nquantreg, rgeos, rmarkdown, robust, rsample, rstan, rstanarm,\nsp, speedglm, statnet.common, survey, survival, testthat,\ntseries, xergm, zoo","BiocManager, Biobase, BSgenome.Celegans.UCSC.ce2,\nBSgenome.Hsapiens.UCSC.hg38,\nBSgenome.Hsapiens.UCSC.hg38.masked,\nBSgenome.Mmusculus.UCSC.mm10, BSgenome.Rnorvegicus.UCSC.rn5,\nBSgenome.Scerevisiae.UCSC.sacCer1,\nTxDb.Hsapiens.UCSC.hg38.knownGene,\nTxDb.Mmusculus.UCSC.mm10.knownGene,\nSNPlocs.Hsapiens.dbSNP144.GRCh38,\nXtraSNPlocs.Hsapiens.dbSNP144.GRCh38, hgu95av2probe, RUnit","","recipes, knitr","png","cliapp, covr, crayon, pingr, ps, testthat, withr","alr4, boot, coxme, leaps, lmtest, Matrix, MatrixModels, rgl\n(&gt;= 0.93.960), sandwich, SparseM, survival, survey",null,"BiocStyle, testthat","BradleyTerry2, e1071, earth (&gt;= 2.2-3), fastICA, gam (&gt;=\n1.15), ipred, kernlab, knitr, klaR, MASS, ellipse, mda, mgcv,\nmlbench, MLmetrics, nnet, party (&gt;= 0.9-99992), pls, pROC,\nproxy, randomForest, RANN, spls, subselect, pamr, superpc,\nCubist, testthat (&gt;= 0.9.1), rpart, dplyr","EBarrays, ALL, Rgraphviz, RColorBrewer, xtable (&gt;= 1.4-6),\nhgu95av2.db, KEGG.db, SNPchip, geneplotter, limma, lattice,\nRUnit, org.Sc.sgd.db, GOstats, GO.db","MASS, rpart","covr, testthat (&gt;= 1.0.0), knitr, rmarkdown","R6, bit, fastmatch, data.table (&gt;= 1.9.8), devtools, ggplot2,\nknitr, magrittr, microbenchmark, rmarkdown, testthat (&gt;=\n0.11.0), tibble","knitr, dendextend (&gt;= 1.0.1), ComplexHeatmap (&gt;= 1.13.2),\ngridBase, png","spData (&gt;= 0.2.6.2)","testthat (&gt;= 0.8), knitr, BiocStyle (&gt;= 0.0.14), BRAIN,\nUniProt.ws (&gt;= 2.1.4)","covr, fansi, mockery, testthat, webshot, withr","covr, rstudioapi (&gt;= 0.5), testthat (&gt;= 2.0.0)","testthat","e1071, lpSolve (&gt;= 5.5.7), quadprog (&gt;= 1.4-8), relations","AnnotationHub, GSEABase, KEGG.db, knitr, org.Hs.eg.db,\nprettydoc, pathview, ReactomePA, testthat",null,"testthat, knitr, rmarkdown, BiocStyle","parallel, xtable, e1071, vcd, TH.data (&gt;= 1.0-7)","datasets, utils, KernSmooth, MASS, kernlab, mvtnorm, vcd,\ntcltk, shiny, shinyjs, ggplot2, dplyr, scales, grid, png, jpeg,\nknitr, rmarkdown, RColorBrewer, rcartocolor, scico, viridis","knitr (&gt;= 1.7), rmarkdown, rstudioapi (&gt;= 0.5)",null,"curl, testthat, xml2","testthat (&gt;= 0.3), knitr, markdown, cluster, MASS, pvclust,\ndendsort, HilbertCurve, Cairo, png, jpeg, tiff, fastcluster,\ndendextend (&gt;= 1.0.1)","covr, crayon, dplyr, pkgdown, testthat","knitr, RColorBrewer, testthat","R6, knitr, rmarkdown, htmltools, DT (&gt;= 0.2), testthat,\nrstudioapi (&gt;= 0.2), xml2 (&gt;= 1.0.0), parallel, memoise,\nmockery","covr, gridGraphics, knitr, rmarkdown, magick, maps, dplyr,\ntidyr, testthat, vdiffr, viridis",null,"testthat, BiocStyle, httr, knitr, rmarkdown","mockery, rstudioapi, testthat, withr",null,"mlbench, caret, knitr","BiocStyle, knitr, readr, rmarkdown, testthat","spelling, testthat (&gt;= 1.0.0), knitr, jsonlite, rmarkdown,\nmagrittr, httpuv (&gt;= 1.4.4), webutils",null,"shiny, knitr, rmarkdown","testthat","BiocGenerics, Biobase, testthat, BiocStyle","DAPAR, Prostar","tibble (&gt;= 1.2), testthat, knitr, rmarkdown, covr","bit64, curl, R.utils, knitr, xts, nanotime, zoo","Formula, graphics, testthat, knitr, rmarkdown, ape, yaml,\nnetworkD3, jsonlite, igraph, treemap, party, partykit,\ndoParallel, foreach, htmlwidgets","blob, covr, hms, knitr, magrittr, rprojroot, rmarkdown,\nRSQLite (&gt;= 1.1-2), testthat, xml2","bit64, covr, knitr, Lahman, nycflights13, RMariaDB (&gt;=\n1.0.2), rmarkdown, RMySQL (&gt;= 0.10.11), RPostgreSQL (&gt;= 0.4.1),\nRSQLite (&gt;= 2.1.0), testthat (&gt;= 2.0.0), withr (&gt;= 2.1.2)",null,"covr, mockery, R6, testthat, withr","Matrix, HDF5Array, genefilter, SummarizedExperiment, airway,\npryr, DelayedMatrixStats, knitr, BiocStyle, RUnit","testthat, knitr, rmarkdown, covr, BiocStyle, microbenchmark,\nprofmem","knitr, rmarkdown, testthat, seriation, colorspace, plyr, ape,\nprofdpm, microbenchmark, gplots, heatmaply, d3heatmap,\ndynamicTreeCut, pvclust, corrplot, DendSer, MASS, cluster,\ncirclize (&gt;= 0.2.5), covr","knitr, RColorBrewer, gplots, heatmap.plus, seriation, gapmap",null,"testthat, enrichR, knitr, BiocStyle","covr, testthat, whoami, withr","testthat, knitr, rmarkdown, vsn, pheatmap, RColorBrewer, IHW,\napeglm, ashr, tximport, tximportData, readr, pbapply, airway,\npasilla (&gt;= 0.2.10)","BiocInstaller, BiocManager, bitops, covr (&gt;= 3.2.0), crayon,\ncurl (&gt;= 0.9), evaluate, foghorn (&gt;= 1.0.1), gmailr (&gt; 0.7.0),\nknitr, lintr (&gt;= 0.2.1), mockery, pingr, MASS, pkgdown, Rcpp\n(&gt;= 0.10.0), rhub, rmarkdown, roxygen2 (&gt;= 6.1.0), rversions,\nspelling (&gt;= 1.1), testthat (&gt;= 1.0.2), whisker","covr, DiagrammeRsvg, rsvg, knitr, testthat",null,"knitr, rmarkdown","NMF, MASS, Matrix, RANN, RSpectra, Rtsne, cccd, coRanking,\ndiffusionMap, energy, fastICA, ggplot2, graphics, igraph,\nkeras, kernlab, knitr, lle, loe, optimx, pcaL1, pcaPP,\nreticulate, rgl, scales, scatterplot3d, stats, tensorflow,\ntestthat, tidyr, tinytex, umap, vegan",null,"distrEx, svUnit (&gt;= 0.7-11), knitr",null,null,"caret, mlbench, rpart, RUnit","BiocStyle, clusterProfiler, knitr, org.Hs.eg.db, testthat","compiler, RUnit, caret, mlbench, rpart, parallel","testthat","bit64 (&gt;= 0.9.7), callr, covr (&gt;= 3.0.1), DBI (&gt;= 0.7.14),\ndbplyr (&gt;= 1.2.0), dtplyr (&gt;= 0.0.2), ggplot2 (&gt;= 2.2.1), hms\n(&gt;= 0.4.1), knitr (&gt;= 1.19), Lahman (&gt;= 3.0-1), lubridate,\nMASS, mgcv (&gt;= 1.8.23), microbenchmark (&gt;= 1.4.4), nycflights13\n(&gt;= 0.2.2), rmarkdown (&gt;= 1.8), RMySQL (&gt;= 0.10.13),\nRPostgreSQL (&gt;= 0.6.2), RSQLite (&gt;= 2.0), testthat (&gt;= 2.0.0),\nwithr (&gt;= 2.1.1)","knitr, rmarkdown","jsonlite (&gt;= 0.9.16), knitr (&gt;= 1.8), rmarkdown, shiny (&gt;=\n0.12.1)",null,"cluster, mlbench, nnet, randomForest, rpart, SparseM, xtable,\nMatrix, MASS","BiocStyle, digest, knitr, rmarkdown, shiny","AnnotationDbi, jsonlite, org.Hs.eg.db, readr, rhdf5, splines","knitr, png","MASS","colorspace, knitr, prettydoc","digest, pillar, readr, rlang, testthat, withr","clusterProfiler, knitr, org.Hs.eg.db, prettydoc",null,"BiocStyle, knitr, EnsDb.Hsapiens.v86 (&gt;= 0.99.8), testthat,\nBSgenome.Hsapiens.NCBI.GRCh38, ggbio (&gt;= 1.24.0), Gviz (&gt;=\n1.20.0), magrittr, AnnotationHub","testthat, knitr, rmarkdown, ggplot2, readr","testthat, lattice, ggplot2","akima","testthat, covr, knitr, rmarkdown, spelling, pacman","knitr, BiocStyle","RColorBrewer, sfsmisc, Rmpfr",null,"ade4, ca, igraph, MASS, knitr, mclust","missMDA,knitr","unitizer, knitr, rmarkdown","testthat, covr",null,null,"testthat, knitr, rmarkdown","","biglm","testthat, parallel, LaF, biglm","fftw","testthat, knitr, rmarkdown, reactome.db, AnnotationDbi,\nparallel, org.Mm.eg.db, limma, GEOquery","testthat, covr, knitr, rmarkdown, spelling","testthat (&gt;= 1.0),\ncovr (&gt;= 2.2.2)",null,"ellipse, clue, cluster, seriation, skmeans","testthat","actuar, codetools, diptest, ellipse, gclus, glmnet, lme4 (&gt;=\n1.0), MASS, mgcv (&gt;= 1.8-0), mlbench, mlogit, multcomp,\nmvtnorm, SuppDists, survival","chemometrics, mvtnorm","covr, ggplot2, testthat","randomForest","codetools, shiny, testit, rmarkdown, knitr",null,"tclust, pdfCluster","testthat, covr, pillar (&gt;= 1.0.0), crayon, withr","testthat, jsonlite",null,"R.rsp, markdown","dplyr, ggplot2, testthat","knitr, pdp, RUnit, splines, viridis","knitr, rmarkdown","affydata, tools, splines, hgu95av2cdf, hgu95av2probe","RUnit","htmltools, testthat, fontquiver (&gt;= 0.2.0), curl","geepack, testthat, MuMIn",null,"class, hgu95av2.db, tkWidgets, ALL, ROC, DESeq, pasilla,\nRColorBrewer, BiocStyle, knitr","Rgraphviz, fibroEset, hgu95av2.db, hu6800.db, hgu133a.db","covr, pkgload, testthat, tibble","GenomicRanges, Rsamtools, GenomicAlignments, BSgenome,\nGenomicFeatures, BSgenome.Scerevisiae.UCSC.sacCer2,\nBSgenome.Celegans.UCSC.ce2, BSgenome.Hsapiens.NCBI.GRCh38,\nTxDb.Dmelanogaster.UCSC.dm3.ensGene, RUnit, BiocStyle, knitr",null,"ShortRead, rtracklayer, BSgenome, GenomicFeatures,\nRNAseqData.HNRNPC.bam.chr14, pasillaBamSubset,\nTxDb.Hsapiens.UCSC.hg19.knownGene,\nTxDb.Dmelanogaster.UCSC.dm3.ensGene,\nBSgenome.Dmelanogaster.UCSC.dm3, BSgenome.Hsapiens.UCSC.hg19,\nDESeq2, edgeR, RUnit, BiocStyle","RMariaDB, org.Mm.eg.db, org.Hs.eg.db, BSgenome,\nBSgenome.Hsapiens.UCSC.hg19 (&gt;= 1.3.17),\nBSgenome.Celegans.UCSC.ce2, BSgenome.Dmelanogaster.UCSC.dm3 (&gt;=\n1.3.17), mirbase.db, FDb.UCSC.tRNAs,\nTxDb.Hsapiens.UCSC.hg19.knownGene,\nTxDb.Dmelanogaster.UCSC.dm3.ensGene (&gt;= 2.7.1),\nTxDb.Mmusculus.UCSC.mm10.knownGene,\nTxDb.Hsapiens.UCSC.hg19.lincRNAsTranscripts,\nTxDb.Hsapiens.UCSC.hg38.knownGene,\nSNPlocs.Hsapiens.dbSNP144.GRCh38, Rsamtools, pasillaBamSubset\n(&gt;= 0.0.5), GenomicAlignments (&gt;= 1.15.7), ensembldb, RUnit,\nBiocStyle, knitr","Matrix, Biobase, AnnotationDbi, annotate, Biostrings (&gt;=\n2.25.3), SummarizedExperiment (&gt;= 0.1.5), Rsamtools (&gt;=\n1.13.53), GenomicAlignments, rtracklayer, BSgenome,\nGenomicFeatures, Gviz, VariantAnnotation, AnnotationHub,\nDESeq2, DEXSeq, edgeR, KEGGgraph, RNAseqData.HNRNPC.bam.chr14,\npasillaBamSubset, KEGG.db, hgu95av2.db, hgu95av2probe,\nBSgenome.Scerevisiae.UCSC.sacCer2, BSgenome.Hsapiens.UCSC.hg19,\nBSgenome.Mmusculus.UCSC.mm10,\nTxDb.Athaliana.BioMart.plantsmart22,\nTxDb.Dmelanogaster.UCSC.dm3.ensGene,\nTxDb.Hsapiens.UCSC.hg19.knownGene,\nTxDb.Mmusculus.UCSC.mm10.knownGene, RUnit, digest, knitr,\nBiocStyle","testthat, rgl, R.matlab, tripack","knitr, rmarkdown, BiocGenerics, testthat, covr","methods, raster","testthat (&gt;= 1.0.0), knitr, markdown","argon2","broom (&gt;= 0.4.0), chemometrics, geosphere (&gt;= 1.5-1), igraph\n(&gt;= 1.0.1), intergraph (&gt;= 2.0-2), maps (&gt;= 3.1.0), mapproj,\nnetwork (&gt;= 1.12.0), scagnostics, scales (&gt;= 0.4.0), sna (&gt;=\n2.3-2), survival, packagedocs (&gt;= 0.4.0), rmarkdown, roxygen2,\ntestthat, crosstalk","gridExtra","vsn, BSgenome.Hsapiens.UCSC.hg19, Homo.sapiens,\nTxDb.Hsapiens.UCSC.hg19.knownGene, chipseq,\nTxDb.Mmusculus.UCSC.mm9.knownGene, knitr, BiocStyle, testthat,\nEnsDb.Hsapiens.v75, tinytex","rpart(&gt;= 4.0-0), tree, testthat, knitr, cluster, scales","knitr (&gt;= 1.7), rmarkdown, rstudioapi (&gt;= 0.5), testthat,\nvdiffr, fontquiver, svglite, withr, devtools","knitr, rmarkdown, devtools","ape, ggtree, gridGraphics, knitr, prettydoc, rsvg, stringi","knitr, testthat, rmarkdown, maps, shiny, sf (&gt;= 0.3-4), dplyr",null,"MASS, stringr, hexbin, dplyr","covr, dplyr, ggplot2movies, hexbin, Hmisc, lattice, mapproj,\nmaps, maptools, multcomp, munsell, nlme, testthat (&gt;= 0.11.0),\nvdiffr, quantreg, knitr, rgeos, rpart, rmarkdown, sf (&gt;=\n0.3-4), svglite (&gt;= 1.2.0.9001)","colorspace, cowplot, ggimage, knitr, lattice, prettydoc, vcd","grDevices, knitr, RColorBrewer, gtable","network, knitr, rmarkdown","knitr, rmarkdown, testthat, gridExtra, devtools, prettydoc","covr, dplyr, gridExtra, ggplot2movies, DAAG, forcats, knitr,\nrmarkdown, testthat, vdiffr, viridis","knitr, rmarkdown, gridExtra, reshape2","testthat, knitr, rmarkdown","Hmisc, testthat, vdiffr","colorspace, cowplot, emojifont, ggimage, knitr, prettydoc,\nrmarkdown, testthat","MASS, mgcv, lubridate, testthat (&gt;= 0.8.1), knitr (&gt;= 1.6),\nrmarkdown","covr, pingr, testthat","getPass","survival, knitr, lars","testthat (&gt;= 1.0.0), knitr, markdown",null,"testthat, covr, magrittr, crayon, knitr, rmarkdown, DBI,\nRSQLite, R.utils, forcats, microbenchmark, rprintf, stringr,\nggplot2, dplyr, withr","mvtnorm, car, stabledist, MASS, timeDate, timeSeries","DBI",null,"covr, ggplot2, knitr, rmarkdown, rprojroot, testthat (&gt;=\n1.0.0)","AnnotationHub, BiocManager, clusterProfiler, DOSE, knitr,\norg.Hs.eg.db, prettydoc, testthat","testthat, knitr, rmarkdown","grid, MASS","MASS","SparseM (&gt;= 0.36), XML, RBGL, RUnit, cluster","lattice","ggplot2, egg, lattice, knitr, testthat","magick (&gt;= 1.3), pdftools (&gt;= 1.6)","lattice","hgu95av2.db, GO.db, org.Hs.eg.db, Rgraphviz, ReportingTools,\ntestthat, BiocStyle, knitr","boot, chron, grid, lattice, quantreg, svUnit, tcltk, zoo","testthat, covr",null,"xtable, BSgenome.Hsapiens.UCSC.hg19, BiocStyle","gWidgetstcltk","covr, fs, knitr, rmarkdown, testthat","h5vcData, SummarizedExperiment (&gt;= 1.9.6), GenomicRanges,\nExperimentHub, BiocParallel, BiocStyle","knitr, covr, tidyselect, rmarkdown, testthat","Biobase, hgu95av2.db, limma",null,"marray, affy, Biobase, limma",null,null,null,"DBI, annotate, RUnit","knitr, rmarkdown, survival, ggplot2, httr, viridisLite,\nshiny, MASS, gapminder",null,"knitr, testit","chron, rms, mice, tables, knitr, ff, ffbase, plotly (&gt;=\n4.5.6)","crayon, lubridate, pillar (&gt;= 1.1.0), testthat",null,"org.Hs.eg.db, GO.db, knitr, BiocStyle, testthat","testthat, XML, xtable, ztable, Hmisc, reshape, rmarkdown,\npander, chron, lubridate, tibble, tidyr (&gt;= 0.7.2), dplyr (&gt;=\n0.7.4)","markdown, testthat","knitr (&gt;= 1.8)","testthat, callr","covr, httpuv, jpeg, knitr, png, readr, rmarkdown, testthat\n(&gt;= 0.8.0), xml2","DBI, annotate, RUnit","spelling, testthat, pdftools, janeaustenr, wordcloud2, knitr,\nstopwords, rmarkdown",null,null,"ape, digest, graph, igraphdata, rgl, scales, stats4, tcltk,\ntestthat","htmltools, datasets, shiny, testthat, rmarkdown, reshape2,\nroxygen2, covr, webshot","knitr, rmarkdown,ggplot2,dplyr,scales,\ntestthat,OpenMPController,raster,spatstat",null,null,null,"testthat","testthat","RUnit, hgu95av2.db, knitr, GenomicRanges,\nSummarizedExperiment, GOstats, ggbio, GO.db, Gviz, rtracklayer,\nmetagenomeSeq, gplots, vegan, Biobase","knitr",null,null,"mvtnorm, mlbench, TH.data","XVector, GenomicRanges, Rsamtools, GenomicAlignments,\nGenomicFeatures, BSgenome.Celegans.UCSC.ce2, pasillaBamSubset,\nRUnit, BiocStyle","PMA",null,"MSnbase, OrgMassSpecR, XML, RJSONIO, Hmisc, gplots,\nRColorBrewer, gridExtra, limma, boot, DBI, MASS",null,"RUnit, foreach","dplyr, testthat",null,"httr, curl, plyr, testthat, knitr, rmarkdown, R.rsp, sp","Rgraphviz, RBGL, RUnit, RColorBrewer, KEGG.db, org.Hs.eg.db,\nhgu133plus2.db, SPIA","RBGL,ALL","RUnit, BiocGenerics, knitr",null,null,"scatterplot3d (&gt;= 0.3-22), som, mlbench, rpart, e1071","testthat, BiocStyle, knitr, rmarkdown, FNN, scRNAseq, scater","formatR, testit, digest, rgl (&gt;= 0.95.1201), codetools,\nrmarkdown, htmlwidgets (&gt;= 0.7), webshot, tikzDevice (&gt;= 0.10),\ntinytex, reticulate (&gt;= 1.4), JuliaCall (&gt;= 0.11.1), png, jpeg,\ngifski, xml2 (&gt;= 1.2.0), httr, DBI (&gt;= 0.4-1), showtext,\ntibble, styler",null,"testthat, knitr, rmarkdown, questionr","testit","KernSmooth","knitr, rmarkdown, testthat","maps, mapproj, deldir, tripack, zoo, MASS, quantreg, mgcv","KernSmooth, Matrix, Rgraphviz, ascii, data.table, ellipse,\nfields, foreach, geepack, gof (&gt;= 0.9), graph, igraph (&gt;= 0.6),\nlava.tobit (&gt;= 0.4.7), lme4, mets (&gt;= 1.1), nlme, optimx,\npolycor, quantreg, rgl, testthat (&gt;= 0.11), visNetwork, zoo","knitr, rmarkdown (&gt;= 0.2.65), testthat, covr",null,"knitr, maps, sf, shiny, testit (&gt;= 0.4), rgdal, rgeos, R6,\nRJSONIO, purrr, testthat","biglm","coin","affy, AnnotationDbi, BiasedUrn, Biobase, ellipse, GO.db,\ngplots, illuminaio, locfit, MASS, org.Hs.eg.db, splines,\nstatmod (&gt;= 1.2.2), vsn","R.utils, R.rsp, markdown","knitr, boot, PKPDmodels, MEMSS, testthat (&gt;= 0.8.1), ggplot2,\nmlmRev, optimx (&gt;= 2013.8.6), gamm4, pbkrtest, HSAUR2,\nnumDeriv, car, dfoptim","covr, pillar, pkgdown, testthat","akima, gam",null,"testthat, knitr, covr",null,"spelling, jsonlite, methods, rmarkdown, rsvg, webp, png,\npdftools, ggplot2, raster, rgdal, gapminder, IRdisplay,\ntesseract (&gt;= 2.0)","testthat, knitr","knitr, testthat (&gt;= 0.8)","knitr, testthat (&gt;= 0.8), RNetCDF (&gt;= 1.6.1)","dygraphs, leaflet, plotly, xts, rmarkdown, testthat, covr",null,"mapproj (&gt;= 1.2-0), mapdata (&gt;= 2.3.0), sp, maptools,\nrnaturalearth","rgeos (&gt;= 0.1-8), spatstat (&gt;= 1.50), PBSmapping, maps,\nRColorBrewer, raster, polyclip, spatstat.utils",null,"knitr, RCurl","tkWidgets","xcms, caTools",null,"base64enc, ggplot2, knitr, microbenchmark, R.devices, R.rsp","BiocStyle, testthat","TH.data, MASS, fields, BayesX, gbm, mlbench, RColorBrewer,\nrpart (&gt;= 4.0-3), randomForest, nnet, testthat (&gt;= 0.10.0),\nkangar00",null,"knitr (&gt;= 1.12), rmarkdown (&gt;= 0.9), mix (&gt;= 1.0), geometry\n(&gt;= 0.3-6), MASS","xtable, Iso",null,"testthat, aws.s3, httr, covr","knitr, lme4, magrittr, rmarkdown, testthat","",null,"ggplot2, multcomp",null,null,"MASS",null,null,"lattice","class, e1071, ipred, randomForest, gpls, pamr, nnet, ALL,\nhgu95av2.db, som, hu6800.db, lattice, caret (&gt;= 5.07),\ngolubEsets, ada, keggorthology, kernlab, mboost, party","ada, adabag, bartMachine, batchtools, brnn, bst, C50, care,\ncaret (&gt;= 6.0-57), class, clue, cluster, clusterSim (&gt;=\n0.44-5), clValid, cmaes, cowplot, CoxBoost, crs, Cubist,\ndeepnet, DiceKriging, DiscriMiner, e1071, earth, elasticnet,\nemoa, evtree, extraTrees, fda.usc, FDboost, flare, FNN, fpc,\nfrbs, FSelector, gbm, GenSA, glmnet, hrbrthemes, h2o (&gt;=\n3.6.0.8), GPfit, Hmisc, irace (&gt;= 2.0), kernlab, kknn, klaR,\nknitr, laGP, LiblineaR, lintr (&gt;= 1.0.0.9001), MASS, mboost,\nmco, mda, mlbench, mldr, mlrMBO, mmpf, modeltools, mRMRe, nnet,\nnodeHarvest (&gt;= 0.7-3), neuralnet, numDeriv, pamr, party,\npenalized (&gt;= 0.9-47), pls, PMCMR (&gt;= 4.1), purrr,\nrandomForest, randomForestSRC (&gt;= 2.7.0), ranger (&gt;= 0.8.0),\nrefund, rex, rFerns, rknn, rmarkdown, ROCR, rotationForest,\nrpart, RRF, rgenoud, rrlda, rsm, RSNNS, RWeka, sda, sf, smoof,\nsparseLDA, stepPlr, survAUC, svglite, SwarmSVM, testthat, tgp,\nTH.data, wavelets, xgboost (&gt;= 0.6-2)",null,"testthat","compiler, covr, ggplot2, testthat",null,"xcms, mzR, MSnbase","knitr, testthat","gWidgets, knitr, testthat",null,null,null,"testthat, pryr, gridExtra, microbenchmark, zoo, knitr (&gt;=\n1.1.0), rols, Rdisop, pRoloc, pRolocdata (&gt;= 1.7.1), msdata (&gt;=\n0.19.3), roxygen2, rgl, rpx, AnnotationHub, BiocStyle (&gt;=\n2.5.19), rmarkdown, imputeLCMD, norm, gplots, shiny, magrittr,\nSummarizedExperiment","BiocStyle, msmsTests, ggplot2, RUnit, BiocGenerics","LFQbench,\nzoo,\nknitr","BiocStyle, knitr, rmarkdown, MSstatsBioData","BiocStyle, knitr, MSstats","knitr, rmarkdown, BiocStyle, animation","lme4 (&gt;= 0.999375-16), nlme, robustbase, coin, MASS, car,\nforeign, xtable, lmtest, coxme (&gt;= 2.2-1), SimComp, ISwR","BiocStyle, testthat, knitr, rmarkdown, R.rsp, HDF5Array,\nRaggedExperiment, UpSetR, survival, survminer","snow","ggplot2, testthat",null,"knitr,\nrmarkdown",null,"knitr, testthat","msdata (&gt;= 0.15.1), RUnit, mzID, BiocStyle (&gt;= 2.5.19),\nknitr, XML",null,null,"knitr, rmarkdown, testthat","testthat (&gt;= 0.8.1), knitr, rmarkdown, inline (&gt;= 0.3.14)","fastICA, doMPI, bigmemory (&gt;= 4.2), synchronicity(&gt;= 1.3.2),\ncorpcor, xtable, devtools, knitr, bibtex, RUnit, mail, Biobase",null,null,null,null,"testthat, devEMF, knitr,tibble,ggplot2, rmarkdown","testthat, digest, knitr, rmarkdown, jsonlite, jose","knitr, testthat","crayon, rex, testthat","Homo.sapiens, Rattus.norvegicus, BSgenome.Hsapiens.UCSC.hg19,\nAnnotationHub, FDb.UCSC.tRNAs, rtracklayer, biomaRt, RUnit,\nRMariaDB","DBI, annotate, RUnit","lattice, knitr, rmarkdown","DBI, annotate, RUnit","testthat (&gt;= 0.7), devtools, httr, knitr, rmarkdown",null,"grid, lattice, ggplot2 (&gt;= 0.9.2), sylly, sylly.en,\nfutile.logger, survival, microbenchmark, zoo, nlme, descr,\nMASS, knitr, rmarkdown, tables, reshape, memisc, Epi,\nrandomForest, tseries, gtable, rms, forecast, data.table","testthat, Rmpi, BatchJobs (&gt;= 1.5)","akima, eaf, emoa, GGally, ggplot2, gridExtra, grid, irace (&gt;=\n2.1), lhs, plyr, reshape2, testthat","covr, testthat","TH.data (&gt;= 1.0-3), mlbench, colorspace, MASS, vcd, ipred,\nvarImp","XML, pmml, rJava, sandwich, strucchange, vcd, AER, mlbench,\nTH.data (&gt;= 1.0-3), coin (&gt;= 1.1-0), RWeka (&gt;= 0.4-19),\ndatasets, parallel, psychotools (&gt;= 0.3-0), psychotree, party\n(&gt;= 1.3-0)",null,"gage, org.Mm.eg.db, RUnit, BiocGenerics","testthat (&gt;= 0.8), ggplot2, BSgenome.Hsapiens.NCBI.GRCh38,\nTxDb.Hsapiens.UCSC.hg38.knownGene, AnnotationHub, knitr,\nrmarkdown, BiocStyle, EnsDb.Hsapiens.v86 (&gt;= 2.0.0)",null,"matrixStats, lattice, ggplot2",null,"knitr (&gt;= 1.19), lubridate (&gt;= 1.7.4), testthat (&gt;= 2.0.0),\nwithr (&gt;= 2.1.2)","testthat","Rcpp, testthat, covr","covr, testthat, disposables (&gt;= 1.0.3)","Biobase, Rgraphviz, RCurl, BiocManager","covr, devtools, htmlwidgets, jsonlite, knitr, leaflet,\ntestthat, rticles","bitops, covr, Rcpp, testthat","devtools (&gt;= 0.8), roxygen2, RUnit, testthat, knitr,\nrmarkdown, yaml, Biobase",null,"Rcpp","MASS, maps, ggthemes, GGally, testthat, knitr, devtools,\nshiny (&gt;= 1.1.0), curl, rmarkdown, Rserve, RSclient, Cairo,\nbroom, webshot, listviewer, dendextend, sf, maptools, rgeos,\npng, IRdisplay, processx, plotlyGeoAssets, forcats","MASS, parallel, Rmpi, testthat, RUnit","abind, testthat, tcltk, foreach, doParallel, itertools,\niterators, covr",null,null,null,null,"spdep, maptools, foreign, mvtnorm","testthat","testthat (&gt;= 0.11.0), knitr (&gt;= 1.11), rmarkdown (&gt;= 0.8.1)",null,"testthat","callr, covr, crayon, debugme, parallel, testthat, withr",null,"Rcpp, testthat, withr","testthat, rmarkdown, pRolocdata (&gt;= 1.9.4), roxygen2,\nsynapter, xtable, rgl, BiocStyle (&gt;= 2.5.19), hpar (&gt;= 1.15.3),\ndplyr, GO.db, AnnotationDbi, Rtsne (&gt;= 0.13), nipals, reshape","pRoloc (&gt;= 1.13.8), testthat","pRolocdata, knitr, BiocStyle (&gt;= 2.5.19), rmarkdown","testthat, future, knitr, rmarkdown","knitr, BiocStyle, rmarkdown, testthat",null,"gplots","testthat, covr","lattice, RUnit, xtable","cba","knitr, rmarkdown","testthat (&gt;= 0.8.0)","callr, covr, curl, pingr, processx (&gt;= 3.1.0), R6, rlang,\ntestthat, tibble","GPArotation, lavaan, sem, lme4,Rcsdp, graph, Rgraphviz","covr, crayon, dplyr (&gt;= 0.7.8), knitr, rmarkdown, testthat,\ntibble, tidyselect","knitr, pepDat","ggplot2, scales, knitr, rmarkdown","rpx","testthat","affy, MSnbase, ggplot2, lattice, yaqcaffy, MAQCsubsetAFX,\nRforProteomics, AnnotationDbi, mzR, hgu133plus2cdf, BiocStyle",null,"DBI,RMySQL,RSQLite,timeSeries,XML,downloader,jsonlite(&gt;= 1.1)","tripack, akima, MASS, survival, rgl, logspline, nor1mix,\nFormula, zoo","memisc, testthat, roxygen2, dplyr, tidyr, janitor, forcats,\nknitr","knitr","knitr, microbenchmark, pryr, testthat, ggplot2, scales",null,null,"RColorBrewer, MASS","survival, testthat","testthat","testthat, roxygen2","rgdal (&gt;= 0.9-1), rgeos (&gt;= 0.3-8), ncdf4, igraph, tcltk,\nparallel, rasterVis, MASS, sf, testthat",null,"Rgraphviz, XML, RUnit, BiocGenerics","Rgraphviz, RCurl, graph, RUnit, BiocGenerics, nem, RBGL,\nigraph",null,"covr, MASS, testthat, vdiffr, sf, spData","covr, knitr, mockery, rmarkdown, testthat",null,"RUnit, inline, rbenchmark, knitr, rmarkdown, pinp, pkgKitten\n(&gt;= 0.1.2)","RUnit, knitr, rmarkdown, pinp","RUnit, Matrix, pkgKitten, reticulate, rmarkdown, knitr, pinp,\nslam","RUnit, inline, knitr, rmarkdown, pinp","inline, RUnit, pkgKitten, microbenchmark","Rcpp, RUnit, knitr, rmarkdown","zoo, testthat","Rcompression, XML","RUnit",null,"RUnit","synchronicity","pixmap, testthat",null,null,"curl, testthat, knitr, rmarkdown, stringi, covr, spelling","covr, knitr, rmarkdown, rprojroot (&gt;= 1.1), testthat","covr, ddalpha, dimRed (&gt;= 0.2.2), fastICA, ggplot2, igraph,\nkernlab, knitr, NMF, pls, RANN, rmarkdown, rpart, rsample,\nRSpectra, testthat","RUnit",null,"covr, testthat","covr, testthat","curl, covr, git2r (&gt;= 0.23.0), knitr, mockery, pkgbuild (&gt;=\n1.0.1), pingr, rmarkdown, testthat, withr",null,"covr, devtools, fortunes, knitr, miniUI, rprojroot,\nrstudioapi, shiny, styler (&gt;= 1.0.2), testthat (&gt;= 2.0.0)","gapminder, jsonlite, purrr, rprojroot, testthat, wesanderson,\nxml2",null,"covr, lattice, testthat (&gt;= 0.8.0)","testthat","testthat, knitr, rmarkdown, dplyr, ggplot2, Hmisc, stringr,\nrvest, roxygen2, covr","AnnotationDbi, rpx, DT, knitr, rmarkdown, BiocStyle, mzR,\nxcms, msdata, isobar, MALDIquant (&gt;= 1.12), MALDIquantForeign,\nreadBrukerFlexData, rTANDEM, synapter, synapterdata, IPPD,\nRdisop, OrgMassSpecR, SummarizedExperiment, BRAIN, rols, hpar,\nGO.db, org.Hs.eg.db, e1071, biomaRt, RColorBrewer, ggplot2,\nreshape2, xtable, lattice, mzID, pRoloc, pRolocdata, MSGFplus,\nMSGFgui, MSnID, msmsTests, msmsEDA, DEP, corrplot, beanplot,\nHeatplus, gplots, VennDiagram, protViz, genefilter, plotly,\ngridExtra, dplyr, lubridate",null,"MASS, rmarkdown, deldir, orientlib, lattice, misc3d,\nrstudioapi, magick","PBSmapping, maptools, sp, loa, RColorBrewer, leaflet","RUnit, BiocGenerics, XML","bit64, BiocStyle, knitr, rmarkdown, testthat, devtools,\nmicrobenchmark, dplyr, ggplot2","BiocStyle, knitr, rmarkdown","covr, testthat","datasets, bit64, testthat, knitr, magrittr, clipr, csvy,\nfeather, fst, hexView, jsonlite, readODS (&gt;= 1.6.4), readr,\nrmatio, xml2 (&gt;= 1.2.0), yaml",null,null,null,"crayon, magrittr, methods, pillar, testthat (&gt;= 2.0.0), covr","testthat, stringdist, pipeR","shiny (&gt;= 0.11), tufte, testthat, digest, dygraphs, tibble,\ncallr (&gt;= 2.0.0)","shinyAce","knitr",null,null,"boot, tcltk, plotly (&gt;= 4.5.6)",null,"testthat, curl","RUnit, testthat","grid, MASS, lattice, boot, cluster, Matrix, robust,\nfit.models, MPV, xtable, ggplot2, GGally, RColorBrewer,\nreshape2, sfsmisc, catdata, doParallel, foreach, skewt",null,"GO.db, knitr (&gt;= 1.1.0), BiocStyle (&gt;= 2.5.19), testthat,\nlubridate, DT, rmarkdown,","tools",null,"covr, devtools, knitr, rmarkdown, testthat (&gt;= 0.8.0)","earth (&gt;= 4.4.6)","testthat, mockr, knitr, withr, rmarkdown","MSnbase, Biostrings, BiocStyle, testthat, knitr","GenomicAlignments, ShortRead (&gt;= 1.19.10), GenomicFeatures,\nTxDb.Dmelanogaster.UCSC.dm3.ensGene, KEGG.db,\nTxDb.Hsapiens.UCSC.hg18.knownGene, RNAseqData.HNRNPC.bam.chr14,\nBSgenome.Hsapiens.UCSC.hg19, RUnit, BiocStyle","knitr, testthat, rmarkdown (&gt;= 1.1), plumber (&gt;= 0.3.2),\nreticulate, shiny, sourcetools, xtable","DBItest, knitr, rmarkdown, testthat","testthat, knitr, rmarkdown","biomaRt","BSgenome (&gt;= 1.33.4), humanStemCell, microRNA (&gt;= 1.1.1),\ngenefilter, limma, org.Hs.eg.db, hgu133plus2.db,\nGenomicFeatures, BSgenome.Hsapiens.UCSC.hg19,\nTxDb.Hsapiens.UCSC.hg19.knownGene, RUnit",null,"irlba, testthat","XML (&gt;= 3.1.0)","digest (&gt;= 0.6.10)",null,"testthat, knitr, png, stringi (&gt;= 0.3.1), rmarkdown, covr","htmltools, testthat, covr, grid",null,"IRanges, GenomicRanges, SummarizedExperiment, Matrix,\nDelayedArray, ShortRead, graph, data.table, RUnit, BiocStyle",null,"AER, car, geepack, lattice, lmtest, MASS, multiwayvcov,\nparallel, pcse, plm, pscl, scatterplot3d, stats4, strucchange,\nsurvival","dichromat, bit64, covr, hms, testthat (&gt;= 2.0)","BiocStyle, biomaRt, beachmat, cowplot, cluster, destiny,\nknitr, monocle, mvoutlier, rmarkdown, Rtsne, umap, testthat,\nmagrittr, pheatmap, irlba, tximport, tximportData, tximeta,\nBiobase, limma",null,"knitr, prettydoc, roxygen2","testthat, BiocStyle, knitr, beachmat, HDF5Array, irlba,\norg.Mm.eg.db, DESeq2, monocle, pracma, Biobase, aroma.light","covr, mockery, testthat, knitr, rmarkdown, withr",null,"testthat, XML, xml2",null,"testthat, knitr","biclust, testthat, DendSer, GA","tools, later, rstudioapi, knitr (&gt;= 1.9), rmarkdown","callr, covr, mockery, testthat","datasets, tcltk, cluster, lattice, MASS, Matrix, nlme, lokern",null,"datasets, Cairo (&gt;= 1.5-5), testthat, knitr (&gt;= 1.6),\nmarkdown, rmarkdown, ggplot2, magrittr",null,null,"knitr (&gt;= 1.7), rmarkdown, shinyAce, testthat (&gt;= 0.9.1), V8\n(&gt;= 0.6)",null,"knitr, rmarkdown, prettydoc, curl, jsonlite","curl","affy, annotate, genefilter, KernSmooth, scrime (&gt;= 1.2.5)","pracma",null,"testthat, BiocStyle, knitr, rmarkdown, scRNAseq, magrittr,\nRtsne","Rmpi,rlecuyer,nws",null,null,"testthat","RColorBrewer, rgdal (&gt;= 0.8-7), rgeos (&gt;= 0.3-13), gstat,\nmaptools, deldir",null,"foreign, maptools, raster, rgdal, sf, sp, spDataLarge (&gt;=\n0.3.0), spdep","testthat,\ncovr","setRNG","glmnet, lars, mboost (&gt; 2.3-0), gamboostLSS (&gt;= 1.2-0), QUIC,\nTH.data, hdi, testthat, knitr, rmarkdown, igraph, huge",null,"MASS, tweedie","quanteda, testthat, covr","testthat, knitr, rmarkdown, pacman, covr, spelling","testthat,\ncovr","testthat",null,"covr, htmltools, htmlwidgets, knitr, rmarkdown, testthat","stats4, car, dynlm, e1071, foreach, lmtest, mvtnorm, tseries","data.tree, dplyr, here, knitr, prettycode, rmarkdown,\nrstudioapi, testthat","BiocStyle, knitr, rmarkdown, testthat","annotate, AnnotationDbi, hgu95av2.db, GenomicFeatures,\nTxDb.Hsapiens.UCSC.hg19.knownGene, BiocStyle, knitr, rmarkdown,\ndigest, jsonlite, rhdf5, HDF5Array (&gt;= 1.7.5), airway, RUnit,\ntestthat","pamr, bladderbatch, BiocStyle, zebrafishRNASeq, testthat","covr, knitr, testthat","htmltools, testthat, xml2 (&gt;= 1.0.0), covr, fontquiver (&gt;=\n0.2.0), knitr, rmarkdown","covr, knitr, testthat","distr, MASS, survival, distrEx, Matrix, splines","synapterdata (&gt;= 1.13.2), xtable, testthat (&gt;= 0.8), BRAIN,\nBiocStyle",null,"spelling, unix (&gt;= 1.3), testthat","curl, jsonlite","tkrplot, lattice, MASS, rgl, tcltk, tcltk2, png, ggplot2,\nlogspline, maptools, R2wd, manipulate","covr, devtools, knitr, rmarkdown, usethis, xml2","dplyr, gdata, plyr","maps","bench (&gt;= 1.0.1), covr (&gt;= 3.2.1), dplyr (&gt;= 0.7.8),\nhtmltools (&gt;= 0.3.6), import (&gt;= 1.1.0), knitr (&gt;= 1.21), mockr\n(&gt;= 0.1), nycflights13 (&gt;= 1.0.0), rmarkdown (&gt;= 1.11),\ntestthat (&gt;= 2.0.1), withr (&gt;= 2.1.2)","knitr, BiocStyle, biobroom","network, data.tree, ape, graph, methods, testthat, covr,\nseriation, netrankr, influenceR, NetSwan","covr, gapminder, knitr, rmarkdown, testthat","covr, dplyr, testthat","readr, tidyr, XML, tm, quanteda, knitr, rmarkdown, ggplot2,\nreshape2, wordcloud, topicmodels, NLP, scales, gutenbergr,\ntestthat, mallet, stm, data.table","ggtree, knitr, prettydoc, testthat, treeio","feather (&gt;= 0.3.1), knitr (&gt;= 1.17), rmarkdown (&gt;= 1.7.4)",null,"date, RUnit","testit, rstudioapi","lattice","GenomicFeatures, shiny, BiocManager, rmarkdown, testthat","covr, knitr, rmarkdown, stopwords (&gt;= 0.9.0), testthat","ggplot2, ggtree, knitr, prettydoc, testthat, tidyr","knitr, rmarkdown, testthat","fpc","testthat",null,"maps, sp, maptools, testthat","RUnit",null,"testthat, covr",null,null,"knitr, rmarkdown, testthat, tximportData,\nTxDb.Hsapiens.UCSC.hg19.knownGene, readr (&gt;= 0.2.2), limma,\nedgeR, DESeq2 (&gt;= 1.11.6), rhdf5, jsonlite, matrixStats","testthat",null,"xml2, testthat, knitr","covr,\ntestthat,\nUniProt.ws","udunits2, NISTunits, measurements, xml2, tibble, pillar (&gt;=\n1.3.0), knitr, testthat, ggforce, rmarkdown, magrittr","knitr","testthat, knitr","covr, knitr, magick, rmarkdown, roxygen2, spelling (&gt;= 1.2),\nstyler (&gt;= 1.0.2), testthat (&gt;= 2.0.0), withr","knitr, rmarkdown, testthat",null,"testthat, knitr, rmarkdown","RUnit, AnnotationHub, BSgenome.Hsapiens.UCSC.hg19,\nTxDb.Hsapiens.UCSC.hg19.knownGene,\nSNPlocs.Hsapiens.dbSNP.20101109, SIFT.Hsapiens.dbSNP132,\nSIFT.Hsapiens.dbSNP137, PolyPhen.Hsapiens.dbSNP131, snpStats,\nggplot2, BiocStyle","testthat","xtable",null,null,"testthat, beeswarm, lattice, ggplot2, beanplot, vioplot,\nggbeeswarm,","hexbin (&gt;= 1.27.0), scales, MASS, knitr, dichromat,\ncolorspace, rasterVis, httr, mapproj, vdiffr, svglite (&gt;=\n1.2.0), testthat, covr, rmarkdown, rgdal","hexbin (&gt;= 1.27.0), ggplot2 (&gt;= 1.0.1), testthat, covr","knitr, rmarkdown, webshot, igraph, rpart, shiny,\nshinyWidgets, colourpicker, sparkline, ggraph, flashClust","affydata, hgu95av2cdf, BiocStyle, knitr, dplyr, testthat","fftw","httpuv, knitr, rmarkdown, shiny","markdown","covr, mockery, testthat","testthat,\ncovr,\nlattice,\nDBI,\nRSQLite,\nmethods,\nknitr,\nrmarkdown","rstudioapi, testit","BiocStyle, caTools, knitr (&gt;= 1.1.0), faahKO, msdata, ncdf4,\nrgl, microbenchmark, testthat, pander, magrittr, MALDIquant","testit, parallel, rstudioapi, tinytex, mime, markdown, knitr,\nhtmltools, base64enc, rmarkdown",null,null,"bitops, RCurl","testthat, curl, covr, knitr, rmarkdown, magrittr, httr",null,"ps, testthat",null,"knitr, spdep, splm, sphet, plm, zoo, survival","timeSeries, timeDate, tseries, chron, fts, tis, RUnit","Biostrings, drosophila2probe, RUnit","testthat, Rgraphviz, graph","RUnit","MAQCsubsetAFX, affydata, xtable, tcltk2, tcltk","testthat, covr","covr, testthat, withr",null,"coda, chron, DAAG, fts, ggplot2, mondate, scales,\nstrucchange, timeDate, timeSeries, tis, tseries, xts","methods","MASS, survival",null,"MASS, Matrix",null,null,null,null,null,"KernSmooth","lattice","MASS","KernSmooth, MASS, latticeExtra","lattice, nlme, nnet, survival","expm, MASS","codetools","splines, parallel, survival, MASS","Hmisc, MASS","MASS","methods","survival","MASS","Matrix, methods","MASS, Matrix, SuppDists, methods, stats4",null,null,null,"codetools, methods, xml2, curl","methods, xml2, commonmark"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"AnnotationHubData",null,null,null,null,null,null,null,"png",null,null,null,null,null,null,null,null,"leaps","biganalytics, bigtabulate","bigmemory, synchronicity",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rmpi",null,null,null,null,null,null,null,null,null,null,null,null,null,null,"FastRWeb",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"RWeka, ape, cba, cclust, flexclust, flexmix, kernlab, mclust,\nmovMF, modeltools",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"",null,null,null,null,null,null,"ggdendro, dendroextras, Hmisc, data.table, rpart",null,"robustbase",null,null,null,null,null,null,null,null,null,null,null,"compiler, RUnit","compiler",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"RMariaDB, shiny",null,null,null,null,"ExperimentHubData",null,null,null,null,null,null,"stats, flashClust",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"compiler, doMC, RUnit, doParallel",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"sp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rgraphviz",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"knitr","shiny (&gt;= 1.1)",null,null,null,null,null,null,null,null,null,null,null,null,null,null,"rstudio","rstudioapi",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"memisc",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"chron, fts, timeSeries, timeDate, tis, tseries, xts, zoo",null,null,null,null,null,null,null,null,"gpclib, RArcInfo",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"parallel",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"gmp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"cleaver",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"mgcv, rpart, nor1mix, polycor, sm, tikzDevice",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"knitr, shiny",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"quantmod",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rgraphviz, Rmpi, XML",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"chron",null,"MatrixModels, graph, SparseM, sfsmisc",null,null,null,null,"snow, nws, Rmpi",null,null,null,null,null,null,null,null,null],["LGPL (&gt;= 2)","MIT + file LICENSE","GPL","MIT + file LICENSE","GPL-3","GPL (&gt;= 2)","LGPL (&gt;= 2.0)","GPL (&gt;= 2)","LGPL (&gt;= 2)","GPL (&gt;= 2)","Apache License 2.0","GPL (&gt;= 3)","Artistic-2.0","GPL","GPL","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL (&gt;= 2)","CC BY-SA 4.0","MIT + file LICENSE","GPL-3","GPL-2","Artistic-2.0","GPL-2 | GPL-3","GPL (&gt;= 2)","BSD_2_clause + file LICENSE","GPL-3","GPL-2","Artistic-2.0","BSL-1.0","GPL (&gt;= 2)","GPL-3","GPL","LGPL-3 | Apache License 2.0","LGPL-3 | Apache License 2.0","MIT + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2)","Artistic-2.0","LGPL","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL-3","GPL-2 | GPL-3","GPL-3","Artistic-2.0","Artistic-2.0","Artistic-2.0","MIT + file LICENSE","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL-2","GPL-2","GPL (&gt;= 2)","GPL-3","GPL-3","GPL (&gt;= 2)","GPL-3","GPL-2","GPL-2","MIT + file LICENSE","Artistic-2.0","Artistic-2.0","GPL-3","GPL-2","MIT + file LICENSE","GPL (&gt;= 2)","GPL (&gt;= 2)","Artistic-2.0","GPL (&gt;= 2)","Artistic-2.0","GPL-3","MIT + file LICENSE","BSD_3_clause + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2)","GPL (&gt;= 3)","MIT + file LICENSE","GPL-3","MIT + file LICENSE","GPL-2","Artistic-2.0","GPL (&gt;= 2)","GPL (==2)","GPL-2","BSD_3_clause + file LICENSE","MIT + file LICENSE","GPL-2","BSD_2_clause + file LICENSE","MIT + file LICENSE","GPL-3","GPL","GPL-3","GPL-2","GPL-3","GPL-3","MIT + file LICENSE","MIT + file LICENSE","GPL-3","Artistic-2.0","MIT + file LICENSE","GPL (&gt;= 2.0)","GPL-3 | file LICENSE","MIT + file LICENSE","Artistic-2.0","GPL-2","MIT + file LICENSE","MPL-2.0 | file LICENSE","GPL (&gt;= 2)","LGPL (&gt;= 2)","MIT + file LICENSE","GPL-2","MIT + file LICENSE","Artistic-2.0","MIT + file LICENSE","GPL-2 | GPL-3","GPL-2 | GPL-3","GPL (&gt;= 2)","Artistic-2.0","MIT + file LICENSE","LGPL (&gt;= 3)","GPL (&gt;= 2)","MIT + file LICENSE","GPL-2","GPL (&gt;= 2)","GPL-3 | file LICENSE","GPL (&gt;= 2)","LGPL-3","Artistic-2.0","GPL-2","GPL-2","Artistic-2.0","GPL-2","GPL-2","MIT + file LICENSE","GPL-3 | file LICENSE","GPL-3 | file LICENSE","GPL (&gt;= 2)","GPL-2","LGPL","GPL (&gt;=2)","GPL-3","GPL (&gt;= 2)","Artistic-2.0","GPL-3","Artistic-2.0","Artistic-2.0","LGPL","GPL-3","MIT + file LICENSE","GPL-3","GPL-3","Artistic-2.0","GPL (&gt;= 2)","LGPL","GPL-2","GPL (&gt;= 2)","GPL (&gt;= 2)","MIT + file LICENSE","FreeBSD | GPL-2 | file LICENSE","GPL-2","GPL-3","GPL (&gt;= 3)","GPL-2 | file LICENSE","GPL-3","GPL (&gt;= 2)","MIT + file LICENCE","GPL-3","MIT + file LICENCE","GPL (&gt;= 2)","GPL-2","MIT + file LICENSE","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-3","Apache License (== 2.0)","GPL","GPL-2 | GPL-3","GPL","GPL-3","LGPL-3","LGPL-3","LGPL (&gt;= 2.1)","CC0","GPL (&gt;= 2) | file LICENSE","GPL (&gt;= 2)","LGPL","GPL-2","GPL-3 | file LICENSE","GPL-3","GPL (&gt;= 3)","Artistic-2.0","Artistic-2.0","GPL-2","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL (&gt;= 3) + file LICENSE","GPL-2","GPL (&gt;= 3)","MIT + file LICENSE","BSD 2-clause License + file LICENSE","GPL (&gt;= 2.0)","GPL (&gt;= 2)","Artistic-2.0","GPL-2 | GPL-3","MIT + file LICENSE","GPL (&gt;= 2)","Artistic-2.0","GPL-3","MIT + file LICENSE","GPL-2","GPL-2 | file LICENSE","Artistic-2.0","GPL-2","GPL-3","GPL-3 | file LICENSE","GPL-2 | file LICENSE","GPL-3 | file LICENSE","GPL-3","GPL-3","Artistic-2.0","GPL-2 | file LICENSE","MIT + file LICENSE","GPL-2","GPL-2","MIT + file LICENSE","LGPL (&gt;= 2.1)","MIT + file LICENSE","GPL (&gt;= 2)","Artistic-2.0","LGPL","MIT + file LICENSE","Artistic-2.0","GPL-3","GPL-2","Artistic-2.0","Artistic-2.0","GPL","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL","Artistic-2.0","GPL (&gt;= 2)","GPL-2","GPL-2","Artistic-2.0","GPL (&gt;= 2)","MIT + file LICENSE","Artistic-2.0","GPL-2 | GPL-3","GPL (&gt;= 2)","GPL-3","GPL-2","Artistic-2.0","LGPL","LGPL","Artistic-2.0","MIT + file LICENSE","GPL (&gt;= 3)","GPL","GPL (&gt;= 2)","GPL-3","Artistic-2.0","Artistic-2.0","GPL (&gt;= 3)","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 2) | file LICENSE","MIT + file LICENSE","Artistic-2.0","GPL-2 | LGPL-2.1 | MPL-1.1","LGPL-2.1","MIT + file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","LGPL","GPL-3","GPL-2","GPL (&gt;= 2)","GPL-2","GPL-3","Artistic-2.0","Artistic-2.0","GPL-2","GPL (version 2 or later)","GPL (&gt;= 2)","Artistic-2.0","GPL-3","GPL (&gt;= 2)","LGPL-2","GPL-2","Apache License (== 2.0)","MIT + file LICENSE","GPL-2 | GPL-3","MIT + file LICENSE","GPL (&gt;= 2)","Artistic-2.0","Artistic-2.0","GPL-2","GPL (&gt;= 2)","GPL-2","GPL-3","GPL","MIT + file LICENSE | Unlimited","GPL-3","LGPL-3","MIT + file LICENSE","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-3","GPL-3","GPL (&gt;= 2)","GPL-3","GPL (&gt;= 2)","GPL-2","GPL (&gt;=2)","LGPL (&gt;= 2.1)","GPL (&gt;= 2)","GPL-3","GPL (&gt;= 2)","LGPL-2","GPL (&gt;= 2)","GPL-2","MIT + file LICENSE","MIT + file LICENSE","GPL (&gt;= 3)","GPL (&gt;= 3)","GPL (&gt;= 2) | file LICENSE","Lucent Public License","GPL-2","GPL (&gt;= 2)","Artistic-2.0","GPL-2","LGPL","LGPL (&gt;= 2)","GPL (&gt;= 2)","Artistic-2.0","Artistic-2.0","GPL-2","GPL (&gt;= 2)","GPL (&gt;= 2)","MIT + file LICENSE","GPL-3","MIT + file LICENSE","GPL-2","","GPL (&gt;= 2)","BSD_2_clause + file LICENSE","GPL","GPL-3","GPL-3","GPL-2","GPL (&gt;= 2)","GPL-2","LGPL","BSD_2_clause + file LICENSE","GPL-2 | GPL-3","GPL (&gt;= 2)","GPL-3","GPL-2","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-3","GPL-2","GPL-2","Artistic-2.0","Artistic-2.0","LGPL (&gt;= 3)","Artistic-2.0","Artistic-2.0","GPL-2","GPL-2","Artistic-2.0","LGPL","MIT + file LICENSE","GPL-2","GPL-2","What license is it under?","GPL (&gt;= 2)","Artistic-2.0","GPL-2","GPL (&gt;= 3)","GPL-3","LGPL-3","GPL (&gt;= 2)","GPL (&gt;= 2)","file LICENSE","LGPL-2","GPL-2","GPL-3","MIT + file LICENSE","MIT + file LICENSE","BSD_3_clause + file LICENCE","Artistic-2.0","Artistic-2.0","BSD_2_clause + file LICENSE","Artistic-2.0","GPL-2","GPL-2","AGPL-3 | file LICENSE","BSD_2_clause + file LICENSE","BSD_2_clause + file LICENSE","GPL-2","GPL-2","GPL-2 | GPL-3","MIT + file LICENSE","GPL (&gt;=3.0)","GPL-3","GPL (&gt;= 2)","GPL (&gt;= 3)","GPL-2","GPL-3","MIT + file LICENSE","GPL-3","MIT + file LICENSE","GPL-2","MIT + file LICENSE","GPL-3","GPL (&gt;= 2)","GPL-2 | GPL-3 | file LICENSE","MIT + file LICENSE","MIT + file LICENSE","GPL-2","MIT + file LICENSE","GPL-2 | GPL-3","GPL (&gt;= 2)","GPL-2","GPL-2","GPL","MIT + file LICENSE","GPL-3","LGPL (&gt;= 2)","MIT + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","GPL-2","GPL-2","GPL-2","MIT + file LICENSE","Artistic-2.0","Artistic-2.0","GPL (&gt;= 2)","GPL-2","GPL-3","GPL-2","GPL-3","GPL-2","BSD_3_clause + file LICENSE","GPL (&gt;= 2)","GPL-3 | file LICENSE","Artistic-2.0","GPL (&gt;= 3)","CC0","GPL-3","GPL-2","GPL (&gt;= 2)","GPL-3","GPL (&gt;= 2)","GPL (&gt;= 2)","LGPL","MIT + file LICENSE","Artistic-2.0","Artistic-2.0","GPL (&gt;= 2)","GPL-3","GPL (&gt;= 3)","MIT + file LICENSE","GPL (&gt;= 3)","GPL-3","Artistic-2.0","GPL (&gt;= 2)","LGPL (&gt;= 2.1)","MIT + file LICENSE","MIT + file LICENSE","Apache License 2.0","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 2) | file LICENSE","GPL-2","GPL (&gt;= 2)","BSD","GPL-2","GPL (&gt;= 2)","GPL-2","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 3)","GPL (&gt;= 3)","GPL (&gt;= 2) | file LICENSE","GPL-3","GPL-2","GPL (&gt;= 2)","GPL-2","MIT + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2)","GPL v2","MIT + file LICENSE","CC0","MIT + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","Artistic-2.0","GPL (&gt;= 3)","GPL","GPL","EPL","Artistic-2.0","Artistic-2.0","MIT + file LICENSE","GPL-2","GPL-2","GPL-2","BSD_3_clause + file LICENSE","GPL-3","MIT + file LICENSE","GPL-3","MIT","MIT + file LICENSE","LGPL (&gt;= 2.1)","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-3","GPL-2","GPL-3","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-2","LGPL (&gt;= 2.1)","GPL-2","GPL (&gt;= 2)","GPL-3","GPL-3","GPL-2","Artistic-2.0 | file LICENSE","GPL-2","LGPL (&gt;= 2)","MIT + file LICENSE","Artistic-1.0 | file LICENSE","Artistic-2.0 + file LICENSE","GPL-2","file LICENSE","GPL-2","LGPL (&gt;= 2.1)","Artistic-2.0","GPL-3","GPL-3","GPL-2","Artistic-2.0","GPL (&gt;= 2)","GPL-2 | GPL-3","MIT + file LICENSE","GPL-3","GPL-2","MIT + file LICENSE","GPL-3","MIT + file LICENSE","GPL","BSD_3_clause + file LICENCE","GPL (&gt;= 2)","GPL-3","GPL-3","GPL","GPL-2","GPL (&gt;= 2)","GPL (&gt;= 3)","GPL-3 | file LICENSE","GPL (&gt;= 2) | file LICENSE","GPL (&gt;= 2)","AGPL-3","GPL-3 | file LICENSE","Apache License (&gt;= 2.0)","Apache License (&gt;= 2.0)","LGPL (&gt;= 2)","GPL-2","GPL (&gt;= 2)","GPL-3","GPL","BSD_3_clause + file LICENSE","GPL (&gt;= 3)","MIT + file LICENSE","GPL (&gt;= 2)","GPL (&gt;= 2)","CC0","AGPL","GPL (&gt;= 2)","GPL-2","LGPL-3","GPL-2 | GPL-3","MIT + file LICENSE","GPL-3","GPL-3","GPL-3","file LICENSE","GPL-2 | file LICENSE","GPL-2 | GPL-3","GPL-3","GPL-2","Artistic-2.0","Artistic-2.0","GPL-2","GPL (&gt;= 2)","GPL-2","LGPL-3","GPL-2","GPL-2","MIT + file LICENSE","GPL-2","Artistic-2.0","MIT + file LICENSE","GPL-3","MIT + file LICENSE","MIT + file LICENSE","Artistic-2.0","MIT + file LICENSE","MIT + file LICENSE","GPL-3","MIT + file LICENSE","Artistic-2.0","GPL-3 | file LICENSE","GPL-2 | GPL-3","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 2)","AGPL-3","MIT + file LICENSE","Artistic-2.0","MIT + file LICENSE","GPL","GPL (&gt;= 2)","GPL","GPL-3","GPL-2","GPL-3","MIT + file LICENSE","Artistic-2.0","Artistic-2.0","GPL (&gt;=2)","MIT + file LICENSE","GPL-2","GPL-3","Artistic License 2.0","GPL-2","MIT + file LICENSE","MIT + file LICENSE","GPL-3","Apache License (== 2.0) | file LICENSE","MIT + file LICENSE","MIT + file LICENSE","Artistic-2.0","GPL-2","GPL","MPL-1.1","BSD","GPL (&gt;= 2)","MIT + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","Artistic-2.0","BSD_3_clause + file LICENSE","GPL-2","GPL-3","MIT + file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 2) + file LICENSE","MIT + file LICENSE","GPL-3","GPL-3","BSD_2_clause + file LICENSE","GPL (&gt;= 2)","BSD","MIT + file LICENSE","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 2)","Artistic-2.0","GPL-2","BSD_3_clause + file LICENSE","Artistic-2.0","GPL (&gt;= 2)","CC0","Artistic-2.0 + file LICENSE","GPL-2 | GPL-3","Part of R 3.6.0","Unlimited","GPL-2 | GPL-3","GPL (&gt;= 2)","GPL","Part of R 3.6.0","Part of R 3.6.0","GPL (&gt;= 2)","Part of R 3.6.0","Part of R 3.6.0","Part of R 3.6.0","Unlimited","GPL (&gt;= 2)","GPL-2 | GPL-3","GPL (&gt;= 2) | file LICENCE","Part of R 3.6.0","GPL (&gt;= 2)","GPL (&gt;= 2) | file LICENCE","GPL-2 | GPL-3","Part of R 3.6.0","GPL-2 | GPL-3","GPL-2 | GPL-3","Part of R 3.6.0","Part of R 3.6.0","Part of R 3.6.0","LGPL (&gt;= 2)","Part of R 3.6.0","Part of R 3.6.0","Part of R 3.6.0"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"yes",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"yes",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"no",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"unix",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["no","yes","no",null,"no","yes","yes","no","yes","yes","yes","no","no","yes","no","no","no","no","no","yes","no","yes","no","yes","no","yes","no","yes","yes","yes","yes","no","no","no","yes","yes","yes","yes","no","no","yes","yes","yes","no","no","no","no","no","yes","yes","no","no","no","no","no","no","no","yes","yes","yes","yes","yes","no","no","no","no","no",null,"no","no",null,"yes","yes","no","no","no","yes","yes","no","yes","no","yes","no","yes","no","no","no","no","yes","no","no","yes","yes","yes","no",null,"yes","no","no","no","yes","no","no","no","no","no","yes","no","yes","no","no","no","no","no","no","yes","no","no","no","yes","no","yes","no","no","no","no","no","no","yes","no","no",null,"yes","yes","yes","yes","no","no","no","no","no","no","yes","no","no","no","yes","yes","yes","no","no","no","yes","no","no","no","no","no","yes","no","no","yes","no","no","no","yes","yes","yes","yes","no","yes","yes","yes","yes","yes","yes",null,null,"yes","no","no","yes","no","no","no","no","no","yes","no","no","no","no","yes","no","yes","no","yes","no","yes","yes","no","no","no","no","yes","no","yes","yes","no","yes","no","yes","no","no","no","no","no","yes","no","yes",null,"no","no","no","no","yes","yes","no","no","no","no","no","no","no","yes","yes","no","no","yes","no","no","no","no","yes","yes","no","no","yes","no","no","no","no","no","no","no","yes","no","no","yes","no","no","no","no","yes","no","no","no","no","no","yes","no","yes","no","no","no","no","yes","no","yes","no","no","yes","no",null,"yes","no","yes","yes","yes","no","yes","no","no","no","no","yes","yes","yes","yes","yes","no","no","no","no","yes","yes","no","no","no","yes","no","no","yes","no","no","no","no","no","yes","no","no","yes","no","no","yes","yes","yes","no","yes","yes","yes","yes","yes","no","yes","no","yes","no","no","yes","yes","yes","no","yes","no","yes","no","yes","yes","yes",null,"yes","yes","yes","no","yes",null,"no","yes","yes","no","yes","yes","yes",null,"no","yes","yes","yes","no","no","no","no","no","no","no","no","yes","no",null,"no","no","no","no","no","yes","no","yes",null,null,"no","yes","no","yes","no","yes","yes",null,"yes","no","no","yes","yes","yes","yes","no","no","no","no","no","yes","yes","no","yes","yes","yes","yes",null,"no","no","no","yes","no","no","yes","no","no","no","no","yes","no","yes","no","no","no","yes","yes","yes","no","yes","no","no","yes","yes","no","yes","yes","no","yes",null,"no","yes","no","no","no","no","yes","yes","no","yes","yes","no","yes","no","no","no","yes","no","yes","no","yes","no","no","no","no","no","yes","yes","yes","yes","yes","yes","yes","no","no","no","no","no","yes","yes","yes","yes","yes","yes","yes","yes","no","no","yes","no","no","no","no","yes","yes","no","yes","no","no","no","no",null,"no","no","yes","yes","no","no","no","yes","yes","no","yes","yes","yes","no","no","yes","yes","yes","yes","no","no",null,"no","no","yes","yes","no","yes","no","yes","no","no","no","yes","yes","no","no","no","yes","no","yes","no","yes","yes","no","yes","no","no","no","no","yes","no","yes","yes","no","yes","yes","no","no","yes","no","no","no","yes","yes","yes","no","no","no","no","no","no","no","no","no","yes","no","no","yes","yes","no","no","yes","yes","yes","yes","yes","no",null,"no","no","no","yes","no","yes",null,"yes","yes","no","no","no","yes","no","yes","no","yes","no","no","no","no","yes","yes","no","yes","no","no","yes","no","yes","yes","yes","no","no","no","yes","no","no","yes","no","yes","no","yes","no","yes","no","yes","yes","no","yes","no","no","no","no","yes",null,null,"yes","no","yes","no","yes","yes","yes","yes","no",null,null,null,"no","no","no","no","yes","yes","no","no","no",null,"no","yes","no","no","no","yes","yes",null,"no","no","no","yes","yes","no","yes","no","no","yes","yes","yes",null,"no","yes","yes","no",null,null,"yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes",null,"yes","yes","yes","yes"],["3.6.0","3.6.0","3.6.0","3.5.0","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.0","3.6.0","3.5.1","3.5.0","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.5.0","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.0","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.0","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.4.0","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.5.0","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.5.1","3.4.2","3.5.1","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.5.0","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.5.0","3.5.1","3.5.1","3.4.2","3.6.0","3.5.1","3.5.0","3.5.1","3.5.0","3.6.0","3.5.1","3.6.0","3.6.0","3.5.0","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.5.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.0","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.4.2","3.5.0","3.6.0","3.6.0","3.5.0","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.5.0","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.5.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.6.0","3.5.0","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.5.0","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.5.0","3.5.1","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.5.0","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.5.0","3.5.1","3.5.1","3.4.0","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.5.0","3.4.0","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.5.0","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.5.0","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.5.0","3.5.1","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.5.0","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.4.0","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.5.0","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.5.0","3.5.1","3.5.1","3.6.0","3.5.0","3.4.0","3.5.0","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.4.0","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.4.0","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.5.0","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.4.2","3.6.0","3.6.0","3.5.0","3.6.0","3.6.0","3.5.1","3.4.0","3.6.0","3.5.0","3.6.0","3.6.0","3.5.1","3.4.2","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.5.0","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.0","3.5.0","3.5.1","3.6.0","3.5.0","3.5.0","3.6.0","3.6.0","3.6.0","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.6.0","3.6.0","3.5.1","3.6.0","3.4.0","3.5.1","3.6.0","3.5.1","3.5.1","3.5.1","3.4.0","3.5.0","3.5.1","3.6.0","3.6.0","3.5.1","3.5.1","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0","3.6.0"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>LibPath<\/th>\n      <th>Version<\/th>\n      <th>Priority<\/th>\n      <th>Depends<\/th>\n      <th>Imports<\/th>\n      <th>LinkingTo<\/th>\n      <th>Suggests<\/th>\n      <th>Enhances<\/th>\n      <th>License<\/th>\n      <th>License_is_FOSS<\/th>\n      <th>License_restricts_use<\/th>\n      <th>OS_type<\/th>\n      <th>MD5sum<\/th>\n      <th>NeedsCompilation<\/th>\n      <th>Built<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>It is always useful to finish a Rmarkdown report with a section
providing all the session information details with the
<code>sessionInfo()</code> function, such at the end of this material. This
allows readers to review the version of R itself and all the
packages that were used to produce the report.</li>
</ul>
<blockquote>
<p><strong>Challenge</strong></p>
<p>Prepare an Rmarkdown report summarising the portal ecology data. The
report should include a <em>Material and methods</em> section where the
data is read in (ideally from the online file) and briefly
described, a <em>Data preparation</em> section where rows with missing
values are filtered out, and a <em>Visualisation</em> section where one or
two plots are rendered. Finish your report with a <em>Session
information</em> section.</p>
</blockquote>
<p>The <a href="https://github.com/rstudio/cheatsheets/raw/master/rmarkdown-2.0.pdf">R Markdown Cheat
Sheet</a>
and <a href="https://www.rstudio.com/wp-content/uploads/2015/03/rmarkdown-reference.pdf">Reference
Guide</a>
will help you with the markdown synatax, R code chunk options, and
RStudio utlisation.</p>
</div>
<div id="docker-containers" class="section level2">
<h2><span class="header-section-number">7.3</span> Docker containers</h2>
<p>There are other tools for reproducible research, that aim to
disseminate more than code and data. Docker containers for example
enable to share the complete image of an operating system, including
all system dependencies and software/data to repeat a complete
analysis. These are useful tools, even though their aren’t necessarily
ideal, and beyond the aim of this course.</p>

</div>
</div>
<p style="text-align: center;">
<a href="sec-join.html"><button class="btn btn-default">Previous</button></a>
<a href="sec-bioinfo.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2019-02-06
</p>
</div>
</div>



</body>
</html>
