<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta property="og:title" content="Chapter 7 Reproducible research | Introduction to bioinformatics" />
<meta property="og:type" content="book" />


<meta property="og:description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2019-03-12" />


<meta name="description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain.">

<title>Chapter 7 Reproducible research | Introduction to bioinformatics</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/jquery-1.12.4/jquery.min.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.5/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Introduction to bioinformatics<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html">Preamble</a>
<a href="sec-dataorg.html"><span class="toc-section-number">1</span> Data organisation with Spreadsheets</a>
<a href="sec-startr.html"><span class="toc-section-number">2</span> Starting with R</a>
<a href="sec-startdata.html"><span class="toc-section-number">3</span> Starting with data</a>
<a href="sec-dplyr.html"><span class="toc-section-number">4</span> Manipulating and analyzing data with dplyr</a>
<a href="sec-vis.html"><span class="toc-section-number">5</span> Data visualization</a>
<a href="sec-join.html"><span class="toc-section-number">6</span> Joining tables</a>
<a id="active-page" href="sec-rr.html"><span class="toc-section-number">7</span> Reproducible research</a><ul class="toc-sections">
<li class="toc"><a href="#knitr-and-rmarkdown"> <code>knitr</code> and <code>rmarkdown</code></a></li>
<li class="toc"><a href="#additional-features"> Additional features</a></li>
<li class="toc"><a href="#docker-containers"> Docker containers</a></li>
<li class="toc"><a href="#additional-exercises-6"> Additional exercises</a></li>
</ul>
<a href="sec-bioinfo.html"><span class="toc-section-number">8</span> Bioinformatics</a>
<a href="sec-prog.html"><span class="toc-section-number">9</span> Additional programming concepts</a>
<a href="sec-ccl.html"><span class="toc-section-number">10</span> Conclusions</a>
<a href="session-information.html">Session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="sec:rr" class="section level1">
<h1><span class="header-section-number">Chapter 7</span> Reproducible research</h1>
<p><strong>Learning Objectives</strong></p>
<ul>
<li>Understand the concept of reproducible research and reproducible documents.</li>
<li>Undertand the process by which a source document in compiled into a final report.</li>
<li>Generate a reproducible report in html or pdf from an Rmarkdown document using RStudio.</li>
</ul>
<p>For a general introduction on the topic in French, see <span class="citation">Pouzat, Davison, and Hinsen (<label for="tufte-mn-23" class="margin-toggle">&#8853;</label><input type="checkbox" id="tufte-mn-23" class="margin-toggle">2015<span class="marginnote">Pouzat, Christophe, Andrew Davison, and Konrad Hinsen. 2015. “La Recherche Reproductible : Une Communication Scientifique Explicite.” <em>Statistique et Société</em> 3 (1). <a href="http://www.publications-sfds.fr/index.php/stat_soc/article/view/448" class="uri">http://www.publications-sfds.fr/index.php/stat_soc/article/view/448</a>.</span>)</span>. If you want to explore the topic of reproducible research in French, the <a href="https://www.fun-mooc.fr/courses/course-v1:inria+41016+session01bis/about">Recherche reproductible : principes méthodologiques pour une science transparente</a> MOOC is of interest.</p>
<p>Reproducible research refers to research that can be reproduced under various conditions and by different people. It applies to every area of research, both experimental and computational, but is often (but not always) easier to implement for computational work. The different levels of reproducibility can formalised<label for="tufte-sn-12" class="margin-toggle sidenote-number">12</label><input type="checkbox" id="tufte-sn-12" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">12</span> The nomenclature is taken from <a href="https://lgatto.github.io/rr-what-should-be-our-goals/">this blog post</a> that provides links and highlights that confusion that exists around the terms and concepts of reproducibility.</span> as follows:</p>
<ul>
<li><p><strong>Repeat</strong> my experiment, i.e. obtain the same tables/graphs/results using the same setup (data, software, …) in the same lab or on the same computer. That’s basically re-running one of my analysis some time after I original developed it.</p></li>
<li><p><strong>Reproduce</strong> an experiment (not ones own), i.e. obtain the same tables/graphs/results in a different lab or on a different computer, using the same setup (the data would be downloaded from a public repository and the same software, but possibly using a different version, or a different operation system).</p></li>
<li><p><strong>Replicate</strong> an experiment, i.e. obtain the same (similar enough) tables/graphs/results in a different set up. The data could still be downloaded from the public repository, or possibly re-generated/re-simulated, and the analysis would be re-implemented based on the original description.</p></li>
<li><p>Finally, <strong>re-use</strong> the information/knowledge from one experiment to run a different experiment with the aim to <strong>confirm</strong> results from scratch.</p></li>
</ul>
<p>The table below summerised these concepts focusing on data and code in computational projects.</p>
<table>
<thead>
<tr class="header">
<th></th>
<th align="center">Same data</th>
<th align="center">Different data</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Same code</strong></td>
<td align="center">Repeat</td>
<td align="center">Reproduce</td>
</tr>
<tr class="even">
<td><strong>Different code</strong></td>
<td align="center">Reproduce</td>
<td align="center">Replicate</td>
</tr>
</tbody>
</table>
<p>There are many reasons to work reproducibly, and <span class="citation">Markowetz (<label for="tufte-mn-24" class="margin-toggle">&#8853;</label><input type="checkbox" id="tufte-mn-24" class="margin-toggle">2015<span class="marginnote">Markowetz, Florian. 2015. “Five Selfish Reasons to Work Reproducibly.” <em>Genome Biology</em> 16 (1): 274. doi:<a href="https://doi.org/10.1186/s13059-015-0850-7">10.1186/s13059-015-0850-7</a>.</span>)</span> nicely summarises 5 good reasons. Importantly, he stressed out that the first beneficiary of reproducible work are the student/research that apply these principles:</p>
<ol style="list-style-type: decimal">
<li>Reproducibility helps to avoid disaster.</li>
<li>Reproducibility makes it easier to write papers<label for="tufte-sn-13" class="margin-toggle sidenote-number">13</label><input type="checkbox" id="tufte-sn-13" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">13</span> And course reports! The exam of this course will consist in a reproducible report using the tools described below.</span>.</li>
<li>Reproducibility helps reviewers see it your way<label for="tufte-sn-14" class="margin-toggle sidenote-number">14</label><input type="checkbox" id="tufte-sn-14" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">14</span> Not only reviewers, also professors that read exams. See previous footnote.</span>.</li>
<li>Reproducibility enables continuity of your work.</li>
<li>Reproducibility helps to build your reputation.</li>
</ol>
<div id="knitr-and-rmarkdown" class="section level2">
<h2><span class="header-section-number">7.1</span> <code>knitr</code> and <code>rmarkdown</code></h2>
<p>Reproducible research is an essential part of any data analysis. With the tools that are available, one can argue that it has become more difficult not to produce reproducible reports than to producing then.</p>
<p>Reproducible documents have been a part of R since the very beginning. See for example <span class="citation">Gentleman and Temple Lang (<label for="tufte-mn-25" class="margin-toggle">&#8853;</label><input type="checkbox" id="tufte-mn-25" class="margin-toggle">2004<span class="marginnote">Gentleman, Robert, and Duncan Temple Lang. 2004. “Statistical Analyses and Reproducible Research.” <em>Bioconductor Project Working Papers. Working Paper 2</em>. <a href="https://biostats.bepress.com/bioconductor/paper2" class="uri">https://biostats.bepress.com/bioconductor/paper2</a>.</span>)</span>, to see how such <em>compendia</em> play a central role within the <a href="https://bioconductor.org/">Bioconductor</a> project (more about Bioconductor in it’s dedicated chapter). Originally, these were written in LaTeX, interleaved with R code chunks, forming so called Sweave documents (with extension <code>.Rnw</code>).</p>
<p>More recently, it has become to use the <a href="https://daringfireball.net/projects/markdown/">markdown</a> syntax markup language, rather than LaTeX. Once interleaved with R code chunks, these documents become Rmarkdown files (<code>.Rmd</code>). The can be converted into markdown using <code>knitr::knit</code>, that executes the code chunk and incorporates their output in the resulting markdown documents, which itself is converted to one of many output formats, typically pdf of html using <a href="http://pandoc.org/">pandoc</a>. In R, this final conversion is done using <code>rmarkdown::render</code> (that relies on pandoc). The <a href="http://rmarkdown.rstudio.com/">rmarkdown</a> package is developed and maintained by RStudio and benefits from excellent documentation, support and integrates into the RStudio editor.</p>
<div class="figure">
<img src="figs/rmd.png" />

</div>
<p>RStudio now also support <em>R Notebook</em> documents<label for="tufte-sn-15" class="margin-toggle sidenote-number">15</label><input type="checkbox" id="tufte-sn-15" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">15</span> See also <a href="https://jupyter.org/">Jupyter notebooks</a> initially developed for Python, but that can run R code as well.</span> that execute individual code chunks that are displayed directly in the source document.</p>
<div class="figure">
<img src="figs/rnb.png" />

</div>
</div>
<div id="additional-features" class="section level2">
<h2><span class="header-section-number">7.2</span> Additional features</h2>
<ul>
<li><p>Among the most options that can set for code chunks is <code>cache</code>. Setting <code>cache=TRUE</code> will avoid that specific code chunk to be cached and not recomputed every time the documented in <em>knitted</em>, unless the code chunk was modified. This is an important feature when long computations are necessary.</p></li>
<li><p>The <code>DT::datatable</code> function allows to create dynamic tables directly from R, as show below.</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ip &lt;-<span class="st"> </span><span class="kw">installed.packages</span>()
DT<span class="op">::</span><span class="kw">datatable</span>(ip)</code></pre></div>
<div id="htmlwidget-cf0fd377cded17393790" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cf0fd377cded17393790">{"x":{"filter":"none","data":[["affy","affyio","ALL","annotate","AnnotationDbi","AnnotationHub","askpass","assertthat","backports","base64enc","BH","Biobase","BiocGenerics","BiocManager","BiocParallel","BiocVersion","biocViews","biomaRt","bit","bit64","bitops","blob","bookdown","brew","broom","callr","caret","caTools","cellranger","cli","clipr","clisymbols","coda","colorspace","commonmark","crayon","crosstalk","curatedTCGAData","curl","data.table","DBI","dbplyr","DelayedArray","dendextend","DEoptimR","desc","devtools","digest","diptest","doParallel","dplyr","DT","e1071","ellipsis","evaluate","ExperimentHub","fansi","flexmix","FNN","forcats","foreach","formatR","fpc","fs","futile.logger","futile.options","gbm","gdata","genefilter","generics","GenomeInfoDb","GenomeInfoDbData","GenomicRanges","ggplot2","ggvis","gh","git2r","glue","gower","graph","gridExtra","gtable","gtools","haven","hexbin","highr","hms","htmltools","htmlwidgets","httpuv","httr","hwriter","igraph","impute","ini","interactiveDisplayBase","ipred","IRanges","iterators","jsonlite","kernlab","knitr","labeling","lambda.r","LaplacesDemon","later","lava","lazyeval","limma","lpSolve","lubridate","magrittr","MALDIquant","markdown","matrixStats","mclust","memoise","mime","mixtools","mlbench","MLInterfaces","ModelMetrics","modelr","modeltools","msmbstyle","MSnbase","MultiAssayExperiment","munsell","mvtnorm","mzID","mzR","ncdf4","numDeriv","openssl","pcaMethods","pillar","pkgbuild","pkgconfig","pkgload","plogr","pls","plyr","prabclus","preprocessCore","prettyunits","processx","prodlim","progress","pRoloc","pRolocdata","promises","ProtGenerics","proxy","ps","purrr","R.methodsS3","R.oo","R.utils","R6","randomForest","RBGL","rcmdcheck","RColorBrewer","Rcpp","RcppArmadillo","RcppRoll","RCurl","rda","readr","readxl","recipes","rematch","remotes","reprex","reshape2","RforProteomics","rhdf5","Rhdf5lib","rlang","rmarkdown","robustbase","roxygen2","rprojroot","RSQLite","rstudioapi","RUnit","rvest","rWSBIM1207","S4Vectors","sampling","scales","segmented","selectr","servr","sessioninfo","sfsmisc","shiny","snow","sourcetools","SQUAREM","stringi","stringr","SummarizedExperiment","sys","threejs","tibble","tidyr","tidyselect","tidyverse","timeDate","tinytex","trimcluster","tufte","usethis","utf8","viridis","viridisLite","vsn","whisker","withr","xaringan","xfun","XML","xml2","xopen","xtable","XVector","yaml","zlibbioc","base","boot","class","cluster","codetools","compiler","datasets","foreign","graphics","grDevices","grid","KernSmooth","lattice","MASS","Matrix","methods","mgcv","nlme","nnet","parallel","rpart","spatial","splines","stats","stats4","survival","tcltk","tools","utils"],["affy","affyio","ALL","annotate","AnnotationDbi","AnnotationHub","askpass","assertthat","backports","base64enc","BH","Biobase","BiocGenerics","BiocManager","BiocParallel","BiocVersion","biocViews","biomaRt","bit","bit64","bitops","blob","bookdown","brew","broom","callr","caret","caTools","cellranger","cli","clipr","clisymbols","coda","colorspace","commonmark","crayon","crosstalk","curatedTCGAData","curl","data.table","DBI","dbplyr","DelayedArray","dendextend","DEoptimR","desc","devtools","digest","diptest","doParallel","dplyr","DT","e1071","ellipsis","evaluate","ExperimentHub","fansi","flexmix","FNN","forcats","foreach","formatR","fpc","fs","futile.logger","futile.options","gbm","gdata","genefilter","generics","GenomeInfoDb","GenomeInfoDbData","GenomicRanges","ggplot2","ggvis","gh","git2r","glue","gower","graph","gridExtra","gtable","gtools","haven","hexbin","highr","hms","htmltools","htmlwidgets","httpuv","httr","hwriter","igraph","impute","ini","interactiveDisplayBase","ipred","IRanges","iterators","jsonlite","kernlab","knitr","labeling","lambda.r","LaplacesDemon","later","lava","lazyeval","limma","lpSolve","lubridate","magrittr","MALDIquant","markdown","matrixStats","mclust","memoise","mime","mixtools","mlbench","MLInterfaces","ModelMetrics","modelr","modeltools","msmbstyle","MSnbase","MultiAssayExperiment","munsell","mvtnorm","mzID","mzR","ncdf4","numDeriv","openssl","pcaMethods","pillar","pkgbuild","pkgconfig","pkgload","plogr","pls","plyr","prabclus","preprocessCore","prettyunits","processx","prodlim","progress","pRoloc","pRolocdata","promises","ProtGenerics","proxy","ps","purrr","R.methodsS3","R.oo","R.utils","R6","randomForest","RBGL","rcmdcheck","RColorBrewer","Rcpp","RcppArmadillo","RcppRoll","RCurl","rda","readr","readxl","recipes","rematch","remotes","reprex","reshape2","RforProteomics","rhdf5","Rhdf5lib","rlang","rmarkdown","robustbase","roxygen2","rprojroot","RSQLite","rstudioapi","RUnit","rvest","rWSBIM1207","S4Vectors","sampling","scales","segmented","selectr","servr","sessioninfo","sfsmisc","shiny","snow","sourcetools","SQUAREM","stringi","stringr","SummarizedExperiment","sys","threejs","tibble","tidyr","tidyselect","tidyverse","timeDate","tinytex","trimcluster","tufte","usethis","utf8","viridis","viridisLite","vsn","whisker","withr","xaringan","xfun","XML","xml2","xopen","xtable","XVector","yaml","zlibbioc","base","boot","class","cluster","codetools","compiler","datasets","foreign","graphics","grDevices","grid","KernSmooth","lattice","MASS","Matrix","methods","mgcv","nlme","nnet","parallel","rpart","spatial","splines","stats","stats4","survival","tcltk","tools","utils"],["/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/home/lgatto/R/x86_64-pc-linux-gnu-library/3.5","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library","/opt/Rpatched/lib/R/library"],["1.60.0","1.52.0","1.24.0","1.60.0","1.44.0","2.14.4","1.1","0.2.0","1.1.3","0.1-3","1.69.0-1","2.42.0","0.28.0","1.30.4","1.16.6","3.8.0","1.50.10","2.38.0","1.1-14","0.9-7","1.0-6","1.1.1","0.9","1.0-6","0.5.1","3.1.1","6.0-81","1.17.1.2","1.1.0","1.0.1","0.5.0","1.2.0","0.19-2","1.4-0","1.7","1.3.4","1.0.0","1.4.3","3.3","1.12.0","1.0.0","1.3.0","0.8.0","1.9.0","1.0-8","1.2.0","2.0.1","0.6.18","0.75-7","1.0.14","0.8.0.1","0.5","1.7-0.1","0.1.0","0.13","1.8.0","0.4.0","2.3-15","1.1.3","0.4.0","1.4.4","1.6","2.1-11.1","1.2.6","1.4.3","1.0.1","2.1.5","2.18.0","1.64.0","0.0.2","1.18.2","1.2.0","1.34.0","3.1.0","0.4.4","1.0.1","0.24.0","1.3.0","0.2.0","1.60.0","2.3","0.2.0","3.8.1","2.1.0","1.27.2","0.7","0.4.2","0.3.6","1.3","1.4.5.1","1.4.0","1.3.2","1.2.4","1.56.0","0.3.1","1.20.0","0.9-8","2.16.0","1.0.10","1.6","0.9-27","1.21","0.3","1.2.3","16.1.1","0.8.0","1.6.5","0.2.1","3.38.3","5.6.13","1.7.4","1.5","1.18","0.9","0.54.0","5.4.2","1.1.0","0.6","1.1.0","2.1-1","1.62.0","1.2.2","0.1.4","0.2-22","0.0.4","2.9.3","1.8.3","0.5.0","1.0-10","1.20.1","2.16.2","1.16","2016.8-1","1.2.2","1.74.0","1.3.1","1.0.2","2.0.2","1.0.2","0.2.0","2.7-0","1.8.4","2.2-7","1.44.0","1.0.2","3.2.1","2018.04.18","1.2.0","1.22.1","1.20.0","1.0.1","1.14.0","0.4-23","1.3.0","0.3.1","1.7.1","1.22.0","2.8.0","2.4.0","4.6-14","1.58.1","1.3.2","1.1-2","1.0.0","0.9.200.7.0","0.3.0","1.95-4.12","1.0.2-2.1","1.3.1","1.3.0","0.1.4","1.0.1","2.0.2","0.2.1","1.4.3","1.20.0","2.26.2","1.4.2","0.3.1","1.11","0.93-3","6.1.1","1.3-2","2.1.1","0.9.0","0.4.32","0.3.2","0.1.3","0.20.1","2.8","1.0.0","0.5-3.0","0.4-1","0.13","1.1.1","1.1-3","1.2.0","0.4-3","0.1.7","2017.10-1","1.3.1","1.4.0","1.12.0","3.0","0.3.1","2.0.1","0.8.3","0.2.5","1.2.1","3043.102","0.10","0.1-2.1","0.4","1.4.0","1.1.4","0.5.1","0.3.0","3.50.0","0.3-2","2.1.2","0.9","0.5","3.98-1.19","1.2.0","1.0.0","1.8-3","0.22.0","2.2.0","1.28.0","3.5.2","1.3-20","7.3-15","2.0.7-1","0.2-16","3.5.2","3.5.2","0.8-71","3.5.2","3.5.2","3.5.2","2.23-15","0.20-38","7.3-51.1","1.2-15","3.5.2","1.8-27","3.1-137","7.3-12","3.5.2","4.1-13","7.3-11","3.5.2","3.5.2","3.5.2","2.43-3","3.5.2","3.5.2","3.5.2"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"base","recommended","recommended","recommended","recommended","base","base","recommended","base","base","base","recommended","recommended","recommended","recommended","base","recommended","recommended","recommended","base","recommended","recommended","base","base","base","recommended","base","base","base"],["R (&gt;= 2.8.0), BiocGenerics (&gt;= 0.1.12), Biobase (&gt;= 2.5.5)","R (&gt;= 2.6.0)","R (&gt;= 2.10), Biobase (&gt;= 2.5.5)","R (&gt;= 2.10), AnnotationDbi (&gt;= 1.27.5), XML","R (&gt;= 2.7.0), methods, utils, stats4, BiocGenerics (&gt;=\n0.23.1), Biobase (&gt;= 1.17.0), IRanges","BiocGenerics (&gt;= 0.15.10)",null,null,"R (&gt;= 3.0.0)","R (&gt;= 2.9.0)",null,"R (&gt;= 2.10), BiocGenerics (&gt;= 0.27.1), utils","methods, utils, graphics, stats, parallel","R (&gt;= 3.5.0)","methods","R (&gt;= 3.5.0), R (&lt; 3.6.0)","R (&gt;= 3.5.0)","methods","R (&gt;= 2.9.2)","R (&gt;= 3.0.1), bit (&gt;= 1.1-12), utils, methods, stats",null,null,null,null,"R (&gt;= 3.1)",null,"R (&gt;= 2.10), lattice (&gt;= 0.20), ggplot2","R (&gt;= 2.2.0)","R (&gt;= 3.0.0)","R (&gt;= 2.10)",null,null,"R (&gt;= 2.14.0)","R (&gt;= 3.0.0), methods",null,null,null,"R (&gt;= 3.5.0), MultiAssayExperiment","R (&gt;= 3.0.0)","R (&gt;= 3.1.0)","R (&gt;= 3.0.0), methods","R (&gt;= 3.1)","R (&gt;= 3.4), methods, stats4, matrixStats, BiocGenerics (&gt;=\n0.27.1), S4Vectors (&gt;= 0.19.15), IRanges (&gt;= 2.11.17),\nBiocParallel","R (&gt;= 3.0.0)",null,"R (&gt;= 3.1.0)","R (&gt;= 3.0.2)","R (&gt;= 3.1.0)",null,"R (&gt;= 2.14.0), foreach(&gt;= 1.2.0), iterators(&gt;= 1.0.0),\nparallel, utils","R (&gt;= 3.1.2)",null,null,"R (&gt;= 3.1)","R (&gt;= 3.0.2)","methods, BiocGenerics (&gt;= 0.15.10), AnnotationHub (&gt;= 2.13.4)","R (&gt;= 3.1.0)","R (&gt;= 2.15.0), lattice","R (&gt;= 3.0.0)","R (&gt;= 3.1)","R (&gt;= 2.5.0)","R (&gt;= 3.0.2)","R (&gt;= 2.0)","R (&gt;= 3.1)","R (&gt;= 3.0.0)","R (&gt;= 2.8.0)","R (&gt;= 2.9.0)","R (&gt;= 2.3.0)",null,"R (&gt;= 3.1)","R (&gt;= 3.1), methods, BiocGenerics (&gt;= 0.13.8), S4Vectors (&gt;=\n0.17.25), IRanges (&gt;= 2.13.12)","R (&gt;= 3.3)","R (&gt;= 2.10), methods, stats4, BiocGenerics (&gt;= 0.25.3),\nS4Vectors (&gt;= 0.19.11), IRanges (&gt;= 2.15.12), GenomeInfoDb (&gt;=\n1.15.2)","R (&gt;= 3.1)","R (&gt;= 3.0)",null,"R (&gt;= 3.1)","R (&gt;= 3.1)",null,"R (&gt;= 2.10), methods, BiocGenerics (&gt;= 0.13.11)",null,"R (&gt;= 2.14)","methods, stats, utils","R (&gt;= 3.1)","R (&gt;= 2.0.1), methods","R (&gt;= 3.0.2)",null,"R (&gt;= 2.14.1)",null,"R (&gt;= 2.15.1), methods","R (&gt;= 3.1)","R (&gt;= 2.6.0)","methods","R (&gt;= 2.10)",null,"R (&gt;= 2.10), methods, BiocGenerics","R (&gt;= 2.10)","R (&gt;= 3.1.0), methods, utils, stats, BiocGenerics (&gt;= 0.25.3),\nS4Vectors (&gt;= 0.19.11)","R (&gt;= 2.5.0), utils","methods","R (&gt;= 2.10)","R (&gt;= 3.1.0)",null,"R (&gt;= 3.0.0)","R (&gt;= 3.0.0)",null,"R (&gt;= 3.0)","R (&gt;= 3.1.0)","R (&gt;= 2.3.0)",null,"methods, R (&gt;= 3.0.0)",null,"R (&gt;= 3.2.0), methods","R (&gt;= 2.11.1)","R (&gt;= 2.12.0)","R (&gt;= 3.0)",null,null,"R (&gt;= 3.2.0)","R (&gt;= 2.10)","R (&gt;= 3.5), methods, BiocGenerics (&gt;= 0.13.11), Biobase,\nannotate, cluster","R (&gt;= 3.2.2)","R (&gt;= 3.1)","stats, stats4","bookdown","R (&gt;= 3.1),\nmethods,\nBiocGenerics (&gt;= 0.7.1),\nBiobase (&gt;= 2.15.2),\nmzR (&gt;= 2.15.1),\nS4Vectors,\nProtGenerics (&gt;= 1.9.1)","R (&gt;= 3.5.0), SummarizedExperiment (&gt;= 1.3.81)",null,"R(&gt;= 3.5.0)","methods","Rcpp (&gt;= 0.10.1), methods, utils",null,"R (&gt;= 2.11.1)",null,"Biobase, methods",null,"R (&gt;= 3.1)",null,null,null,"R (&gt;= 2.10)","R (&gt;= 3.1.0)","R (&gt;= 2.10), MASS, mclust",null,null,null,"R (&gt;= 2.9.0)",null,"R (&gt;= 2.15), MSnbase (&gt;= 1.19.20), MLInterfaces (&gt;= 1.37.1),\nmethods, Rcpp (&gt;= 0.10.3), BiocParallel","R (&gt;= 2.15), MSnbase",null,"methods","R (&gt;= 3.4.0)","R (&gt;= 3.1)","R (&gt;= 3.1)","R (&gt;= 2.13.0)","R (&gt;= 2.13.0), R.methodsS3 (&gt;= 1.7.1)","R (&gt;= 2.14.0), R.oo (&gt;= 1.22.0)","R (&gt;= 3.0)","R (&gt;= 3.2.2), stats","graph, methods",null,"R (&gt;= 2.0.0)","R (&gt;= 3.0.0)","R (&gt;= 3.3.0)","R (&gt;= 2.15.1)","R (&gt;= 3.0.0), methods, bitops","R (&gt;= 2.10)","R (&gt;= 3.1)",null,"R (&gt;= 3.1), dplyr",null,"R (&gt;= 3.0.0)","R (&gt;= 3.1)","R (&gt;= 3.1)","MSnbase (&gt;= 2.5.3)","R (&gt;= 3.5.0), methods",null,"R (&gt;= 3.1.0)","R (&gt;= 3.0)","R (&gt;= 3.1.0)","R (&gt;= 3.1)","R (&gt;= 3.0.0)","R (&gt;= 3.1.0)",null,"R (&gt;= 2.5.0), utils (&gt;= 2.5.0), methods (&gt;= 2.5.0), graphics\n(&gt;= 2.5.0)","R (&gt;= 3.0.1), xml2",null,"R (&gt;= 3.3.0), methods, utils, stats, stats4, BiocGenerics (&gt;=\n0.23.3)",null,"R (&gt;= 3.1)",null,"R (&gt;= 3.0)","R (&gt;= 3.0.0)",null,"R (&gt;= 3.2.0)","R (&gt;= 3.0.2), methods","R (&gt;= 2.13.1), utils","R (&gt;= 3.0.2)","R (&gt;= 3.0)","R (&gt;= 2.14)","R (&gt;= 3.1)","R (&gt;= 3.2), methods, GenomicRanges (&gt;= 1.33.6), Biobase,\nDelayedArray (&gt;= 0.3.20)",null,"R (&gt;= 3.0.0), igraph (&gt;= 1.0.0)","R (&gt;= 3.1.0)","R (&gt;= 3.1)","R (&gt;= 3.1)",null,"R (&gt;= 2.15.1), graphics, utils, stats, methods",null,"R (&gt;= 1.9.0)",null,"R (&gt;= 3.1)","R (&gt;= 2.10)","R (&gt;= 2.10), viridisLite (&gt;= 0.3.0)","R (&gt;= 2.10)","R (&gt;= 3.4.0), Biobase",null,"R (&gt;= 3.0.2)",null,null,"R (&gt;= 2.13.0), methods, utils","R (&gt;= 3.1.0)","R (&gt;= 3.1)","R (&gt;= 2.10.0)","R (&gt;= 2.8.0), methods, BiocGenerics (&gt;= 0.19.2), S4Vectors (&gt;=\n0.19.15), IRanges (&gt;= 2.15.12)",null,null,null,"R (&gt;= 3.0.0), graphics, stats","R (&gt;= 3.0.0), stats, utils","R (&gt;= 3.2.0)","R (&gt;= 2.1)",null,null,"R (&gt;= 3.0.0)",null,null,null,"R (&gt;= 2.5.0), stats","R (&gt;= 3.0.0)","R (&gt;= 3.1.0), grDevices, graphics, stats, utils","R (&gt;= 3.2.0)",null,"R (&gt;= 2.14.0), nlme (&gt;= 3.1-64)","R (&gt;= 3.4.0)","R (&gt;= 2.14.0), stats, utils",null,"R (&gt;= 2.15.0), graphics, stats, grDevices","R (&gt;= 3.0.0), graphics, stats, utils",null,null,null,"R (&gt;= 2.13.0)",null,null,null],["affyio (&gt;= 1.13.3), BiocManager, graphics, grDevices, methods,\npreprocessCore, stats, utils, zlibbioc","zlibbioc, methods",null,"Biobase, DBI, xtable, graphics, utils, stats, methods,\nBiocGenerics (&gt;= 0.13.8), RCurl","DBI, RSQLite, S4Vectors (&gt;= 0.9.25)","utils, methods, grDevices, RSQLite, BiocManager, curl,\nAnnotationDbi (&gt;= 1.31.19), S4Vectors, interactiveDisplayBase,\nhttr, yaml","sys (&gt;= 2.1)","tools","utils",null,null,"methods","methods, utils, graphics, stats, parallel","utils","stats, utils, futile.logger, parallel, snow",null,"Biobase, graph (&gt;= 1.9.26), methods, RBGL (&gt;= 1.13.5), tools,\nutils, XML, RCurl, RUnit","utils, XML, RCurl, AnnotationDbi, progress, stringr, httr",null,null,null,"methods, prettyunits","htmltools (&gt;= 0.3.6), knitr (&gt;= 1.16), rmarkdown (&gt;= 1.9),\nxfun (&gt;= 0.2), tinytex, yaml (&gt;= 2.1.14)",null,"backports, dplyr, generics (&gt;= 0.0.2), methods, nlme, purrr,\nreshape2, stringr, tibble, tidyr","processx (&gt;= 3.2.1), R6, utils","foreach, methods, plyr, ModelMetrics (&gt;= 1.1.0), nlme,\nreshape2, stats, stats4, utils, grDevices, recipes (&gt;= 0.1.4),\nwithr (&gt;= 2.0.0)","bitops","rematch, tibble","assertthat, crayon (&gt;= 1.3.4), methods, utils","utils",null,"lattice","graphics, grDevices, stats",null,"grDevices, methods, utils","htmltools (&gt;= 0.3.5), jsonlite, lazyeval, R6, shiny (&gt;= 0.11),\nggplot2","AnnotationHub, ExperimentHub, S4Vectors, utils",null,"methods",null,"assertthat (&gt;= 0.2.0), DBI (&gt;= 1.0.0), dplyr (&gt;= 0.7.5), glue\n(&gt;= 1.2.0), methods, purrr (&gt;= 0.2.5), R6 (&gt;= 2.2.2), rlang (&gt;=\n0.2.0), tibble (&gt;= 1.4.2), tidyselect (&gt;= 0.2.4), utils","stats","utils, stats, datasets, magrittr (&gt;= 1.0.1), ggplot2, fpc,\nwhisker, viridis","stats","assertthat, utils, R6, crayon, rprojroot","callr, cli, digest, git2r (&gt;= 0.23.0), httr (&gt;= 0.4),\njsonlite, memoise (&gt;= 1.0.0), pkgbuild (&gt;= 1.0.2), pkgload (&gt;=\n1.0.1), rcmdcheck (&gt;= 1.3.0), remotes (&gt;= 2.0.0), rstudioapi\n(&gt;= 0.6.0.9000), sessioninfo, stats, tools, usethis (&gt;= 1.4.0),\nutils, withr",null,null,null,"assertthat (&gt;= 0.2.0), glue (&gt;= 1.1.1), magrittr (&gt;= 1.5),\nmethods, pkgconfig (&gt;= 2.0.1), R6 (&gt;= 2.2.2), Rcpp (&gt;= 1.0.0),\nrlang (&gt;= 0.3.0), tibble (&gt;= 2.0.0), tidyselect (&gt;= 0.2.5),\nutils","htmltools (&gt;= 0.3.6), htmlwidgets (&gt;= 1.3), magrittr,\ncrosstalk, promises","graphics, grDevices, class, stats, methods, utils",null,"methods","utils, S4Vectors, BiocManager, curl",null,"graphics, grid, grDevices, methods, modeltools (&gt;= 0.2-16),\nnnet, stats, stats4, utils",null,"ellipsis, magrittr, rlang, tibble","codetools, utils, iterators",null,"MASS, cluster, mclust, flexmix, prabclus, class, diptest,\nmvtnorm, robustbase, kernlab, trimcluster, grDevices, graphics,\nmethods, stats, utils","methods, Rcpp","utils, lambda.r (&gt;= 1.1.0), futile.options",null,"gridExtra, lattice, parallel, survival","gtools, stats, methods, utils","S4Vectors (&gt;= 0.9.42), AnnotationDbi, annotate, Biobase,\ngraphics, methods, stats, survival","methods","stats, stats4, utils, RCurl, GenomeInfoDbData",null,"utils, stats, XVector (&gt;= 0.19.8)","digest, grid, gtable (&gt;= 0.1.1), lazyeval, MASS, mgcv, plyr\n(&gt;= 1.7.1), reshape2, rlang (&gt;= 0.2.1), scales (&gt;= 0.5.0),\nstats, tibble, viridisLite, withr (&gt;= 2.0.0)","assertthat, jsonlite (&gt;= 0.9.11), shiny (&gt;= 0.11.1), magrittr,\ndplyr (&gt;= 0.4.0), lazyeval, htmltools (&gt;= 0.2.4), methods","ini, jsonlite, httr","graphics, utils","methods",null,"stats, stats4, utils","gtable, grid, grDevices, graphics, utils","grid",null,"forcats (&gt;= 0.2.0), hms, Rcpp (&gt;= 0.11.4), readr (&gt;= 0.1.0),\ntibble","lattice, grid, graphics, grDevices, stats, utils",null,"methods, pkgconfig, rlang","utils, digest, Rcpp","grDevices, htmltools (&gt;= 0.3), jsonlite (&gt;= 0.9.16), yaml","Rcpp (&gt;= 0.11.0), utils, promises, later (&gt;= 0.7.3)","curl (&gt;= 0.9.1), jsonlite, mime, openssl (&gt;= 0.8), R6",null,"graphics, grDevices, magrittr, Matrix, pkgconfig (&gt;= 2.0.0),\nstats, utils",null,null,"shiny","rpart (&gt;= 3.1-8), MASS, survival, nnet, class, prodlim","stats4",null,null,"methods, stats, grDevices, graphics","evaluate (&gt;= 0.10), highr, markdown, stringr (&gt;= 0.6), yaml\n(&gt;= 2.1.19), methods, xfun, tools",null,"formatR","parallel, grDevices, graphics, stats, utils","Rcpp (&gt;= 0.12.9), rlang","grDevices, graphics, methods, numDeriv, stats, survival,\nSQUAREM, utils",null,"grDevices, graphics, stats, utils, methods",null,"stringr, Rcpp (&gt;= 0.12.13),",null,"parallel","utils, mime (&gt;= 0.3)",null,"stats, utils, graphics, grDevices","digest (&gt;= 0.6.3)","tools","MASS, segmented, stats, survival",null,"gdata, pls, sfsmisc, MASS, rpart, rda, genefilter, fpc, ggvis,\nshiny, gbm, RColorBrewer, hwriter, threejs (&gt;= 0.2.2), mlbench,\nstats4, tools, grDevices, graphics, stats","Rcpp, data.table","broom, dplyr, magrittr, purrr (&gt;= 0.2.2), rlang (&gt;= 0.2.0),\ntibble, tidyr (&gt;= 0.8.0)","methods","tufte, htmltools, xfun, knitr (&gt;= 1.18), rmarkdown, stringr,\nmagrittr","BiocParallel,\nIRanges (&gt;= 2.13.28),\nplyr,\npreprocessCore,\nvsn,\ngrid,\nstats4,\naffy,\nimpute,\npcaMethods,\nMALDIquant (&gt;= 1.16),\nmzID (&gt;= 1.5.2),\ndigest,\nlattice,\nggplot2,\nXML,\nscales,\nMASS,\nRcpp,\nrhdf5","methods, GenomicRanges (&gt;= 1.25.93), BiocGenerics, S4Vectors\n(&gt;= 0.17.5), IRanges, Biobase, stats, tidyr, utils","colorspace, methods","stats, methods","XML, plyr, parallel, doParallel, foreach, iterators,\nProtGenerics","Biobase, BiocGenerics (&gt;= 0.13.6), ProtGenerics (&gt;= 1.9.1),\nncdf4",null,null,"askpass","BiocGenerics, Rcpp (&gt;= 0.11.3), MASS","cli (&gt;= 1.0.0), crayon (&gt;= 1.3.4), fansi (&gt;= 0.4.0), methods,\nrlang (&gt;= 0.3.0.1), utf8 (&gt;= 1.1.3)","callr (&gt;= 2.0.0), cli, crayon, desc, prettyunits, R6,\nrprojroot, withr (&gt;= 2.1.2)","utils","desc, methods, pkgbuild, rlang, rprojroot, rstudioapi, utils,\nwithr",null,"grDevices, graphics, methods, stats","Rcpp (&gt;= 0.11.0)",null,"stats","magrittr, assertthat, methods","ps, R6, utils","Rcpp (&gt;= 0.11.5), stats, graphics, survival, KernSmooth, lava","hms, prettyunits, R6, crayon","Biobase, mclust (&gt;= 4.3), caret, e1071, sampling, class,\nkernlab, lattice, nnet, randomForest, proxy, FNN, hexbin,\nBiocGenerics, stats, dendextend, RColorBrewer, scales, MASS,\nknitr, mvtnorm, LaplacesDemon, coda, mixtools, gtools, plyr,\nggplot2, biomaRt, utils, grDevices, graphics","Biobase, utils","R6, Rcpp, later, rlang, stats, magrittr",null,"stats, utils","utils","magrittr (&gt;= 1.5), rlang (&gt;= 0.3.1)","utils","methods, utils","methods, utils, tools, R.methodsS3 (&gt;= 1.7.1)",null,null,"methods","callr (&gt;= 2.0.0), cli, crayon, desc (&gt;= 1.2.0), digest,\npkgbuild, prettyunits, R6, rprojroot, sessioninfo (&gt;= 1.1.1),\nutils, withr, xopen",null,"methods, utils","Rcpp (&gt;= 0.11.0), stats, utils, methods","Rcpp",null,null,"Rcpp (&gt;= 0.12.0.5), tibble, hms (&gt;= 0.4.1), R6, clipr, crayon,\nmethods","cellranger, Rcpp (&gt;= 0.12.18), tibble (&gt;= 1.3.1), utils","generics, gower, ipred, lubridate, magrittr, Matrix, purrr (&gt;=\n0.2.3), RcppRoll, rlang (&gt;= 0.3.0.1), stats, tibble, tidyr,\ntidyselect (&gt;= 0.1.1), timeDate, utils, withr",null,"methods, utils","callr (&gt;= 2.0.0), clipr (&gt;= 0.4.0), fs, rlang, rmarkdown,\ntools, utils, whisker, withr","plyr (&gt;= 1.8.1), Rcpp, stringr","R.utils, shiny, biocViews, BiocManager","Rhdf5lib (&gt;= 1.3.2)",null,null,"tools, utils, knitr (&gt;= 1.18), yaml (&gt;= 2.1.19), htmltools (&gt;=\n0.3.5), evaluate (&gt;= 0.8), base64enc, jsonlite, mime, tinytex\n(&gt;= 0.3), methods, stringr (&gt;= 1.2.0)","stats, graphics, utils, methods, DEoptimR","brew, commonmark, desc (&gt;= 1.2.0), digest, methods, pkgload\n(&gt;= 1.0.2), purrr, R6 (&gt;= 2.1.2), Rcpp (&gt;= 0.11.0), stringi,\nstringr (&gt;= 1.0.0), utils, xml2","backports","bit64, blob (&gt;= 1.1.1), DBI (&gt;= 1.0.0), memoise, methods,\npkgconfig, Rcpp (&gt;= 0.12.7)",null,null,"httr (&gt;= 0.5), selectr, magrittr","utils, tibble, dplyr",null,"MASS, lpSolve","labeling, munsell (&gt;= 0.5), R6, RColorBrewer, Rcpp,\nviridisLite",null,"methods, stringr, R6","mime (&gt;= 0.2), httpuv (&gt;= 1.4.0), xfun, jsonlite","cli, tools, utils, withr","grDevices, methods, utils, stats","utils, grDevices, httpuv (&gt;= 1.4.4), mime (&gt;= 0.3), jsonlite\n(&gt;= 0.9.16), xtable, digest, htmltools (&gt;= 0.3.5), R6 (&gt;= 2.0),\nsourcetools, later (&gt;= 0.7.2), promises (&gt;= 1.0.1), tools,\ncrayon, rlang",null,null,null,"tools, utils, stats","glue (&gt;= 1.2.0), magrittr, stringi (&gt;= 1.1.7)","utils, stats, tools, Matrix, BiocGenerics (&gt;= 0.15.3),\nS4Vectors (&gt;= 0.17.25), IRanges (&gt;= 2.13.16), GenomeInfoDb (&gt;=\n1.13.1)",null,"htmlwidgets (&gt;= 0.3.2), base64enc, crosstalk, methods, stats","cli (&gt;= 1.0.1), crayon (&gt;= 1.3.4), fansi (&gt;= 0.4.0), methods,\npillar (&gt;= 1.3.1), pkgconfig (&gt;= 2.0.2), rlang (&gt;= 0.3.1),\nutils","dplyr (&gt;= 0.7.0), glue, magrittr, purrr, Rcpp, rlang, stringi,\ntibble, tidyselect (&gt;= 0.2.5), utils","glue (&gt;= 1.3.0), purrr, rlang (&gt;= 0.2.2), Rcpp (&gt;= 0.12.0)","broom (&gt;= 0.4.2), cli (&gt;= 1.0.0), crayon (&gt;= 1.3.4), dplyr (&gt;=\n0.7.4), dbplyr (&gt;= 1.1.0), forcats (&gt;= 0.2.0), ggplot2 (&gt;=\n2.2.1), haven (&gt;= 1.1.0), hms (&gt;= 0.3), httr (&gt;= 1.3.1),\njsonlite (&gt;= 1.5), lubridate (&gt;= 1.7.1), magrittr (&gt;= 1.5),\nmodelr (&gt;= 0.1.1), purrr (&gt;= 0.2.4), readr (&gt;= 1.1.1), readxl\n(&gt;= 1.0.0), reprex (&gt;= 0.1.1), rlang (&gt;= 0.1.4), rstudioapi (&gt;=\n0.7), rvest (&gt;= 0.3.2), stringr (&gt;= 1.2.0), tibble (&gt;= 1.3.4),\ntidyr (&gt;= 0.7.2), xml2 (&gt;= 1.1.1)",null,"xfun (&gt;= 0.3)",null,"htmltools, xfun, knitr (&gt;= 1.18), rmarkdown","clipr (&gt;= 0.3.0), clisymbols, crayon, curl (&gt;= 2.7), desc, fs\n(&gt;= 1.2.0), gh, git2r (&gt;= 0.23), glue, rlang, rprojroot (&gt;=\n1.2), rstudioapi, utils, whisker",null,"stats, ggplot2 (&gt;= 1.0.1), gridExtra",null,"methods, affy, limma, lattice, ggplot2",null,"stats, graphics, grDevices","htmltools, knitr (&gt;= 1.21), servr (&gt;= 0.13), xfun (&gt;= 0.5),\nrmarkdown","tools",null,"Rcpp","processx","stats, utils","methods, utils, zlibbioc, BiocGenerics, S4Vectors, IRanges",null,null,null,null,"MASS","graphics, grDevices, stats, utils",null,null,null,"methods, utils, stats","grDevices",null,"grDevices, utils",null,"grid, grDevices, graphics, stats, utils","methods","methods, graphics, grid, stats, utils, lattice","utils, stats","methods, stats, graphics, Matrix, splines, utils","graphics, stats, utils, lattice",null,"tools, compiler",null,null,"graphics, stats","utils, grDevices, graphics","graphics, methods, stats","graphics, Matrix, methods, splines, stats, utils","utils",null,null],["preprocessCore",null,null,null,null,null,null,null,null,null,null,null,null,null,"BH",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"S4Vectors",null,null,null,null,null,null,null,"BH (&gt;= 1.58.0-1), plogr (&gt;= 0.1.10), Rcpp (&gt;= 1.0.0)",null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,"S4Vectors, IRanges",null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,"Rcpp",null,"Rcpp, BH, later",null,null,null,null,null,null,null,"S4Vectors",null,null,null,null,null,null,null,"Rcpp, BH",null,null,null,null,"Rcpp,",null,null,null,null,null,null,null,null,null,null,"Rcpp",null,null,null,"Rcpp",null,null,null,null,"Rcpp, zlibbioc, Rhdf5lib (&gt;= 1.1.4)",null,null,null,"Rcpp",null,null,null,null,null,null,"Rcpp",null,null,null,null,"Rcpp",null,"Rcpp, RcppArmadillo",null,"later, Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp","Rcpp",null,null,"Rcpp, BH","progress, Rcpp",null,null,null,null,"Rcpp",null,"Rhdf5lib",null,null,null,null,"Rcpp",null,"BH, plogr (&gt;= 0.2.0), Rcpp",null,null,null,null,null,null,"Rcpp",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp","Rcpp (&gt;= 0.12.0),",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Rcpp (&gt;= 0.12.12)",null,null,"S4Vectors, IRanges",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["tkWidgets (&gt;= 1.19.0), affydata, widgetTools",null,"rpart","hgu95av2.db, genefilter, Biostrings (&gt;= 2.25.10), IRanges,\nrae230a.db, rae230aprobe, tkWidgets, GO.db, org.Hs.eg.db,\norg.Mm.eg.db, hom.Hs.inp.db, humanCHRLOC, Rgraphviz, RUnit,","hgu95av2.db, GO.db, org.Sc.sgd.db, org.At.tair.db, KEGG.db,\nRUnit, TxDb.Hsapiens.UCSC.hg19.knownGene, hom.Hs.inp.db,\norg.Hs.eg.db, reactome.db, AnnotationForge, graph,\nEnsDb.Hsapiens.v75, BiocStyle, knitr","IRanges, GenomicRanges, GenomeInfoDb, VariantAnnotation,\nRsamtools, rtracklayer, BiocStyle, knitr, AnnotationForge,\nrBiopaxParser, RUnit, GenomicFeatures, MSnbase, mzR,\nBiostrings, SummarizedExperiment, ExperimentHub, gdsfmt","testthat","testthat",null,null,null,"tools, tkWidgets, ALL, RUnit, golubEsets","Biobase, S4Vectors, IRanges, GenomicRanges, Rsamtools,\nAnnotationDbi, oligoClasses, oligo, affyPLM, flowClust, affy,\nDESeq2, MSnbase, annotate, RUnit","BiocStyle, BiocVersion, remotes, testthat, knitr, withr","BiocGenerics, tools, foreach, BatchJobs, BBmisc, doParallel,\nRmpi, GenomicRanges, RNAseqData.HNRNPC.bam.chr14,\nTxDb.Hsapiens.UCSC.hg19.knownGene, VariantAnnotation,\nRsamtools, GenomicAlignments, ShortRead, codetools, RUnit,\nBiocStyle, knitr, batchtools, data.table",null,"BiocGenerics, knitr, commonmark","annotate, BiocStyle, knitr, rmarkdown, testthat",null,null,null,"covr, pillar (&gt;= 1.2.1), testthat","htmlwidgets, rstudioapi, miniUI, rsconnect (&gt;= 0.4.3), servr\n(&gt;= 0.3), shiny, testit (&gt;= 0.5), tufte, webshot",null,"AER, akima, AUC, bbmle, betareg, biglm, binGroup, boot, brms,\nbtergm, car, caret, coda, covr, e1071, emmeans, ergm, gam (&gt;=\n1.15), gamlss, gamlss.data, gamlss.dist, geepack, ggplot2,\nglmnet, gmm, Hmisc, irlba, joineRML, Kendall, knitr, ks,\nLahman, lavaan, lfe, lme4, lmodel2, lmtest, lsmeans, maps,\nmaptools, MASS, Matrix, mclust, mgcv, muhaz, multcomp, network,\nnnet, orcutt (&gt;= 2.2), ordinal, plm, plyr, poLCA, psych,\nquantreg, rgeos, rmarkdown, robust, rsample, rstan, rstanarm,\nsp, speedglm, statnet.common, survey, survival, testthat,\ntseries, xergm, zoo","cliapp, covr, crayon, pingr, ps, testthat, withr","BradleyTerry2, e1071, earth (&gt;= 2.2-3), fastICA, gam (&gt;=\n1.15), ipred, kernlab, knitr, klaR, MASS, ellipse, mda, mgcv,\nmlbench, MLmetrics, nnet, party (&gt;= 0.9-99992), pls, pROC,\nproxy, randomForest, RANN, spls, subselect, pamr, superpc,\nCubist, testthat (&gt;= 0.9.1), rpart, dplyr","MASS, rpart","covr, testthat (&gt;= 1.0.0), knitr, rmarkdown","covr, fansi, mockery, testthat, webshot, withr","covr, rstudioapi (&gt;= 0.5), testthat (&gt;= 2.0.0)","testthat",null,"datasets, utils, KernSmooth, MASS, kernlab, mvtnorm, vcd,\ntcltk, shiny, shinyjs, ggplot2, dplyr, scales, grid, png, jpeg,\nknitr, rmarkdown, RColorBrewer, rcartocolor, scico, viridis","curl, testthat, xml2","mockery, rstudioapi, testthat, withr",null,"BiocStyle, knitr, readr, rmarkdown, testthat","spelling, testthat (&gt;= 1.0.0), knitr, jsonlite, rmarkdown,\nmagrittr, httpuv (&gt;= 1.4.4), webutils","bit64, curl, R.utils, knitr, xts, nanotime, zoo","blob, covr, hms, knitr, magrittr, rprojroot, rmarkdown,\nRSQLite (&gt;= 1.1-2), testthat, xml2","bit64, covr, knitr, Lahman, nycflights13, RMariaDB (&gt;=\n1.0.2), rmarkdown, RMySQL (&gt;= 0.10.11), RPostgreSQL (&gt;= 0.4.1),\nRSQLite (&gt;= 2.1.0), testthat (&gt;= 2.0.0), withr (&gt;= 2.1.2)","Matrix, HDF5Array, genefilter, SummarizedExperiment, airway,\npryr, DelayedMatrixStats, knitr, BiocStyle, RUnit","knitr, rmarkdown, testthat, seriation, colorspace, plyr, ape,\nprofdpm, microbenchmark, gplots, heatmaply, d3heatmap,\ndynamicTreeCut, pvclust, corrplot, DendSer, MASS, cluster,\ncirclize (&gt;= 0.2.5), covr",null,"covr, testthat, whoami, withr","BiocInstaller, BiocManager, bitops, covr (&gt;= 3.2.0), crayon,\ncurl (&gt;= 0.9), evaluate, foghorn (&gt;= 1.0.1), gmailr (&gt; 0.7.0),\nknitr, lintr (&gt;= 0.2.1), mockery, pingr, MASS, pkgdown, Rcpp\n(&gt;= 0.10.0), rhub, rmarkdown, roxygen2 (&gt;= 6.1.0), rversions,\nspelling (&gt;= 1.1), testthat (&gt;= 1.0.2), whisker","knitr, rmarkdown",null,"caret, mlbench, rpart, RUnit","bit64 (&gt;= 0.9.7), callr (&gt;= 3.1.1), covr (&gt;= 3.0.1), DBI (&gt;=\n0.7.14), dbplyr (&gt;= 1.2.0), dtplyr (&gt;= 0.0.2), ggplot2 (&gt;=\n2.2.1), hms (&gt;= 0.4.1), knitr (&gt;= 1.19), Lahman (&gt;= 3.0-1),\nlubridate (&gt;= 1.7.4), MASS, mgcv (&gt;= 1.8.23), microbenchmark\n(&gt;= 1.4.4), nycflights13 (&gt;= 0.2.2), rmarkdown (&gt;= 1.8), RMySQL\n(&gt;= 0.10.13), RPostgreSQL (&gt;= 0.6.2), RSQLite (&gt;= 2.0),\ntestthat (&gt;= 2.0.0), withr (&gt;= 2.1.1), broom (&gt;= 0.5.1), purrr\n(&gt;= 0.3.0), readr (&gt;= 1.3.1), crayon (&gt;= 1.3.4)","jsonlite (&gt;= 0.9.16), knitr (&gt;= 1.8), rmarkdown, shiny (&gt;=\n0.12.1)","cluster, mlbench, nnet, randomForest, rpart, SparseM, xtable,\nMatrix, MASS","covr, testthat","testthat, lattice, ggplot2","knitr, BiocStyle","unitizer, knitr, rmarkdown","actuar, codetools, diptest, Ecdat, ellipse, gclus, glmnet,\nlme4 (&gt;= 1.1), MASS, mgcv (&gt;= 1.8-0), mlbench, multcomp,\nmvtnorm, SuppDists, survival","chemometrics, mvtnorm","covr, ggplot2, testthat, readr, knitr, rmarkdown, dplyr","randomForest","codetools, shiny, testit, rmarkdown, knitr","tclust, pdfCluster","testthat, covr, pillar (&gt;= 1.0.0), crayon, withr","testthat, jsonlite",null,"knitr, pdp, RUnit, splines, viridis","RUnit","class, hgu95av2.db, tkWidgets, ALL, ROC, DESeq, pasilla,\nRColorBrewer, BiocStyle, knitr","covr, pkgload, testthat, tibble","GenomicRanges, Rsamtools, GenomicAlignments, BSgenome,\nGenomicFeatures, BSgenome.Scerevisiae.UCSC.sacCer2,\nBSgenome.Celegans.UCSC.ce2, BSgenome.Hsapiens.NCBI.GRCh38,\nTxDb.Dmelanogaster.UCSC.dm3.ensGene, RUnit, BiocStyle, knitr",null,"Matrix, Biobase, AnnotationDbi, annotate, Biostrings (&gt;=\n2.25.3), SummarizedExperiment (&gt;= 0.1.5), Rsamtools (&gt;=\n1.13.53), GenomicAlignments, rtracklayer, BSgenome,\nGenomicFeatures, Gviz, VariantAnnotation, AnnotationHub,\nDESeq2, DEXSeq, edgeR, KEGGgraph, RNAseqData.HNRNPC.bam.chr14,\npasillaBamSubset, KEGG.db, hgu95av2.db, hgu95av2probe,\nBSgenome.Scerevisiae.UCSC.sacCer2, BSgenome.Hsapiens.UCSC.hg19,\nBSgenome.Mmusculus.UCSC.mm10,\nTxDb.Athaliana.BioMart.plantsmart22,\nTxDb.Dmelanogaster.UCSC.dm3.ensGene,\nTxDb.Hsapiens.UCSC.hg19.knownGene,\nTxDb.Mmusculus.UCSC.mm10.knownGene, RUnit, digest, knitr,\nBiocStyle","covr, dplyr, ggplot2movies, hexbin, Hmisc, lattice, mapproj,\nmaps, maptools, multcomp, munsell, nlme, testthat (&gt;= 0.11.0),\nvdiffr, quantreg, knitr, rgeos, rpart, rmarkdown, sf (&gt;=\n0.3-4), svglite (&gt;= 1.2.0.9001)","MASS, mgcv, lubridate, testthat (&gt;= 0.8.1), knitr (&gt;= 1.6),\nrmarkdown","covr, pingr, testthat","getPass","testthat, covr, magrittr, crayon, knitr, rmarkdown, DBI,\nRSQLite, R.utils, forcats, microbenchmark, rprintf, stringr,\nggplot2, dplyr, withr","testthat,","SparseM (&gt;= 0.36), XML, RBGL, RUnit, cluster","ggplot2, egg, lattice, knitr, testthat","testthat, covr",null,"covr, fs, knitr, rmarkdown, testthat, pillar (&gt;= 1.1.1), cli,\ncrayon","marray, affy, Biobase, limma","knitr, testit","crayon, lubridate, pillar (&gt;= 1.1.0), testthat","markdown, testthat","knitr (&gt;= 1.8)","testthat, callr","covr, httpuv, jpeg, knitr, png, readr, rmarkdown, testthat\n(&gt;= 0.8.0), xml2",null,"ape, digest, graph, igraphdata, rgl, scales, stats4, tcltk,\ntestthat",null,"testthat","knitr","mvtnorm, mlbench, TH.data","XVector, GenomicRanges, Rsamtools, GenomicAlignments,\nGenomicFeatures, BSgenome.Celegans.UCSC.ce2, pasillaBamSubset,\nRUnit, BiocStyle","RUnit, foreach","httr, curl, plyr, testthat, knitr, rmarkdown, R.rsp, sp",null,"formatR, testit, digest, rgl (&gt;= 0.95.1201), codetools,\nrmarkdown, htmlwidgets (&gt;= 0.7), webshot, tikzDevice (&gt;= 0.10),\ntinytex, reticulate (&gt;= 1.4), JuliaCall (&gt;= 0.11.1), png, jpeg,\ngifski, xml2 (&gt;= 1.2.0), httr, DBI (&gt;= 0.4-1), showtext,\ntibble, styler",null,"testit","KernSmooth","knitr, rmarkdown, testthat","KernSmooth, Matrix, Rgraphviz, data.table, ellipse, fields,\nforeach, geepack, gof (&gt;= 0.9), graph, igraph (&gt;= 0.6),\nlava.tobit (&gt;= 0.4.7), lme4, mets (&gt;= 1.1), nlme, optimx,\npolycor, quantreg, rgl, testthat (&gt;= 0.11), visNetwork, zoo","knitr, rmarkdown (&gt;= 0.2.65), testthat, covr","affy, AnnotationDbi, BiasedUrn, Biobase, ellipse, GO.db,\ngplots, illuminaio, locfit, MASS, org.Hs.eg.db, splines,\nstatmod (&gt;= 1.2.2), vsn",null,"testthat, knitr, covr","testthat, knitr","knitr, testthat (&gt;= 0.8)","knitr, RCurl","base64enc, ggplot2, knitr, microbenchmark, R.devices, R.rsp","knitr (&gt;= 1.12), rmarkdown (&gt;= 0.9), mix (&gt;= 1.0), geometry\n(&gt;= 0.3-6), MASS","testthat, aws.s3, httr, covr",null,null,"lattice","class, e1071, ipred, randomForest, gpls, pamr, nnet, ALL,\nhgu95av2.db, som, hu6800.db, lattice, caret (&gt;= 5.07),\ngolubEsets, ada, keggorthology, kernlab, mboost, party","testthat","compiler, covr, ggplot2, testthat",null,null,"testthat,\npryr,\ngridExtra,\nmicrobenchmark,\nzoo,\nknitr (&gt;= 1.1.0),\nrols,\nRdisop,\npRoloc,\npRolocdata (&gt;= 1.7.1),\nmsdata (&gt;= 0.19.3),\nroxygen2,\nrgl,\nrpx,\nAnnotationHub,\nBiocStyle (&gt;= 2.5.19),\nrmarkdown,\nimputeLCMD,\nnorm,\ngplots,\nshiny,\nmagrittr,\nSummarizedExperiment","BiocStyle, testthat, knitr, rmarkdown, R.rsp, HDF5Array,\nRaggedExperiment, UpSetR, survival, survminer","ggplot2, testthat",null,"knitr, testthat","msdata (&gt;= 0.15.1), RUnit, mzID, BiocStyle (&gt;= 2.5.19),\nknitr, XML",null,null,"testthat, digest, knitr, rmarkdown, jsonlite, jose","matrixStats, lattice, ggplot2","knitr (&gt;= 1.19), lubridate (&gt;= 1.7.4), testthat (&gt;= 2.0.0),\nwithr (&gt;= 2.1.2)","Rcpp, testthat, covr","covr, testthat, disposables (&gt;= 1.0.3)","bitops, covr, Rcpp, testthat","Rcpp","MASS, parallel, Rmpi, testthat, RUnit","abind, testthat, tcltk, foreach, doParallel, itertools,\niterators, covr","spdep, maptools, foreign, mvtnorm",null,"testthat","callr, covr, crayon, debugme, parallel, testthat, withr",null,"Rcpp, testthat, withr","testthat, rmarkdown, pRolocdata (&gt;= 1.9.4), roxygen2,\nsynapter, xtable, rgl, BiocStyle (&gt;= 2.5.19), hpar (&gt;= 1.15.3),\ndplyr, GO.db, AnnotationDbi, Rtsne (&gt;= 0.13), nipals","pRoloc (&gt;= 1.13.8), testthat","testthat, future, knitr, rmarkdown",null,"cba","callr, covr, curl, pingr, processx (&gt;= 3.1.0), R6, rlang,\ntestthat, tibble","covr, crayon, dplyr (&gt;= 0.7.8), knitr, rmarkdown, testthat,\ntibble, tidyselect",null,"tools","digest (&gt;= 0.6.10)","knitr, microbenchmark, pryr, testthat, ggplot2, scales","RColorBrewer, MASS","Rgraphviz, XML, RUnit, BiocGenerics","covr, knitr, mockery, rmarkdown, testthat",null,"RUnit, inline, rbenchmark, knitr, rmarkdown, pinp, pkgKitten\n(&gt;= 0.1.2)","RUnit, Matrix, pkgKitten, reticulate, rmarkdown, knitr, pinp,\nslam","zoo, testthat","Rcompression, XML",null,"curl, testthat, knitr, rmarkdown, stringi, covr, spelling","covr, knitr, rmarkdown, rprojroot (&gt;= 1.1), testthat","covr, ddalpha, dimRed (&gt;= 0.2.2), fastICA, ggplot2, igraph,\nkernlab, knitr, NMF, pls, RANN, rmarkdown, rpart, rsample,\nRSpectra, testthat","covr, testthat","curl, covr, git2r (&gt;= 0.23.0), knitr, mockery, pkgbuild (&gt;=\n1.0.1), pingr, rmarkdown, testthat, withr","covr, devtools, fortunes, knitr, miniUI, rprojroot,\nrstudioapi, shiny, styler (&gt;= 1.0.2), testthat (&gt;= 2.0.0)","covr, lattice, testthat (&gt;= 0.8.0)","AnnotationDbi, rpx, DT, knitr, rmarkdown, BiocStyle, mzR,\nxcms, msdata, isobar, MALDIquant (&gt;= 1.12), MALDIquantForeign,\nreadBrukerFlexData, rTANDEM, synapter, synapterdata, IPPD,\nRdisop, OrgMassSpecR, SummarizedExperiment, BRAIN, rols, hpar,\nGO.db, org.Hs.eg.db, e1071, biomaRt, RColorBrewer, ggplot2,\nreshape2, xtable, lattice, mzID, pRoloc, pRolocdata, MSGFplus,\nMSGFgui, MSnID, msmsTests, msmsEDA, DEP, corrplot, beanplot,\nHeatplus, gplots, VennDiagram, protViz, genefilter, plotly,\ngridExtra, dplyr, lubridate","bit64, BiocStyle, knitr, rmarkdown, testthat","BiocStyle, knitr, rmarkdown","crayon, magrittr, methods, pillar, testthat (&gt;= 2.0.0), covr","shiny (&gt;= 0.11), tufte, testthat, digest, dygraphs, tibble,\ncallr (&gt;= 2.0.0)","grid, MASS, lattice, boot, cluster, Matrix, robust,\nfit.models, MPV, xtable, ggplot2, GGally, RColorBrewer,\nreshape2, sfsmisc, catdata, doParallel, foreach, skewt","covr, devtools, knitr, rmarkdown, testthat (&gt;= 0.8.0)","testthat, mockr, knitr, withr, rmarkdown","DBItest, knitr, rmarkdown, testthat","testthat, knitr, rmarkdown","XML (&gt;= 3.1.0)","testthat, knitr, png, stringi (&gt;= 0.3.1), rmarkdown, covr","MSnbase, Biobase, SummarizedExperiment, pRolocdata,\ncuratedTCGAData, ALL, tidyverse, rmarkdown","IRanges, GenomicRanges, SummarizedExperiment, Matrix,\nDelayedArray, ShortRead, graph, data.table, RUnit, BiocStyle",null,"dichromat, bit64, covr, hms, testthat (&gt;= 2.0)",null,"testthat, XML, xml2","tools, later, rstudioapi, knitr (&gt;= 1.9), rmarkdown","callr, covr, mockery, testthat","datasets, tcltk, cluster, lattice, MASS, Matrix, nlme, lokern","datasets, Cairo (&gt;= 1.5-5), testthat, knitr (&gt;= 1.6),\nmarkdown, rmarkdown, ggplot2, magrittr","Rmpi,rlecuyer,nws","testthat","setRNG",null,"covr, htmltools, htmlwidgets, knitr, rmarkdown, testthat","annotate, AnnotationDbi, hgu95av2.db, GenomicFeatures,\nTxDb.Hsapiens.UCSC.hg19.knownGene, BiocStyle, knitr, rmarkdown,\ndigest, jsonlite, rhdf5, HDF5Array (&gt;= 1.7.5), airway, RUnit,\ntestthat","unix (&gt;= 1.4), spelling, testthat","maps","bench (&gt;= 1.0.1), covr (&gt;= 3.2.1), dplyr (&gt;= 0.7.8),\nhtmltools (&gt;= 0.3.6), import (&gt;= 1.1.0), knitr (&gt;= 1.21), mockr\n(&gt;= 0.1), nycflights13 (&gt;= 1.0.0), rmarkdown (&gt;= 1.11),\ntestthat (&gt;= 2.0.1), withr (&gt;= 2.1.2)","covr, gapminder, knitr, rmarkdown, testthat","covr, dplyr, testthat","feather (&gt;= 0.3.1), knitr (&gt;= 1.17), rmarkdown (&gt;= 1.7.4)","date, RUnit","testit, rstudioapi","fpc",null,"covr, knitr, magick, rmarkdown, roxygen2, spelling (&gt;= 1.2),\nstyler (&gt;= 1.0.2), testthat (&gt;= 2.0.0), withr","knitr, rmarkdown, testthat","hexbin (&gt;= 1.27.0), scales, MASS, knitr, dichromat,\ncolorspace, rasterVis, httr, mapproj, vdiffr, svglite (&gt;=\n1.2.0), testthat, covr, rmarkdown, rgdal","hexbin (&gt;= 1.27.0), ggplot2 (&gt;= 1.0.1), testthat, covr","affydata, hgu95av2cdf, BiocStyle, knitr, dplyr, testthat","markdown","testthat, covr, lattice, DBI, RSQLite, methods, knitr,\nrmarkdown","rstudioapi, testit","testit, parallel, rstudioapi, tinytex, mime, markdown, knitr,\nhtmltools, base64enc, remotes, rmarkdown","bitops, RCurl","testthat, curl, covr, knitr, rmarkdown, magrittr, httr","ps, testthat","knitr, spdep, splm, sphet, plm, zoo, survival","Biostrings, drosophila2probe, RUnit","RUnit",null,"methods","MASS, survival",null,"MASS, Matrix",null,null,null,null,null,"KernSmooth","lattice","MASS","KernSmooth, MASS, latticeExtra","lattice, nlme, nnet, survival","expm, MASS","codetools","parallel, survival, MASS","Hmisc, MASS","MASS","methods","survival","MASS","Matrix, methods","MASS, Matrix, SuppDists, methods, stats4",null,null,null,"codetools, methods, xml2, curl","methods, xml2, commonmark"],[null,null,null,null,null,"AnnotationHubData",null,null,null,"png",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"ggdendro, dendroextras, Hmisc, data.table, rpart","robustbase",null,null,null,null,"compiler",null,null,null,null,null,"ExperimentHubData",null,null,null,null,"compiler, doMC, RUnit, doParallel",null,null,null,null,null,null,null,null,null,null,null,null,"sp",null,null,null,null,null,"Rgraphviz",null,null,null,null,null,null,null,"knitr","shiny (&gt;= 1.1)",null,null,null,null,null,null,"rstudioapi",null,null,null,null,null,null,null,null,null,null,null,null,null,null,"chron, fts, timeSeries, timeDate, tis, tseries, xts, zoo",null,null,null,null,null,null,null,null,null,"parallel",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"cleaver",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"mgcv, rpart, nor1mix, polycor, sm, tikzDevice",null,null,null,null,null,null,null,null,"knitr, shiny",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"chron",null,"MatrixModels, graph, SparseM, sfsmisc",null,null,null,null,"snow, nws, Rmpi",null,null,null,null,null,null,null,null,null],["LGPL (&gt;= 2.0)","LGPL (&gt;= 2)","Artistic-2.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","MIT + file LICENSE","GPL-3","GPL-2","GPL-2 | GPL-3","BSL-1.0","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL-2 | GPL-3","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL-2","GPL-2","GPL (&gt;= 2)","GPL-3","GPL-3","GPL-2","MIT + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2)","GPL-3","MIT + file LICENSE","MIT + file LICENSE","GPL-3","MIT + file LICENSE","GPL (&gt;= 2)","BSD_3_clause + file LICENSE","BSD_2_clause + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","Artistic-2.0","MIT + file LICENSE","MPL-2.0 | file LICENSE","LGPL (&gt;= 2)","MIT + file LICENSE","Artistic-2.0","GPL-2 | GPL-3","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-2","MIT + file LICENSE","GPL-3 | file LICENSE","GPL-2","GPL-3","MIT + file LICENSE","Artistic-2.0","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-3","Apache License (== 2.0)","GPL","GPL","GPL-3","LGPL-3","LGPL-3","GPL (&gt;= 2) | file LICENSE","GPL-2","Artistic-2.0","GPL-2","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL-2 | file LICENSE","GPL-2 | file LICENSE","MIT + file LICENSE","GPL-2","MIT + file LICENSE","GPL-3","Artistic-2.0","GPL (&gt;= 2)","GPL-2","GPL-2","MIT + file LICENSE","GPL-2","GPL","GPL-3","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 2) | file LICENSE","MIT + file LICENSE","LGPL-2.1","GPL (&gt;= 2)","GPL-2","GPL-3","Artistic-2.0","GPL (&gt;= 2)","Artistic-2.0","Apache License (== 2.0)","MIT + file LICENSE","GPL-2","GPL","MIT + file LICENSE | Unlimited","LGPL-3","MIT + file LICENSE","GPL (&gt;= 2)","GPL-3","GPL-3","GPL (&gt;=2)","LGPL-2","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 3)","GPL-2","Artistic-2.0","GPL (&gt;= 2)","MIT + file LICENSE","GPL","GPL (&gt;= 2)","GPL-2","LGPL","GPL (&gt;= 2)","GPL-3","GPL-2","GPL-3","Artistic-2.0","Artistic-2.0","MIT + file LICENSE","GPL-2","GPL (&gt;= 2)","Artistic-2.0","GPL (&gt;= 3)","GPL-2","MIT + file LICENSE","GPL (&gt;= 3)","GPL-3","GPL-3","MIT + file LICENSE","GPL-3","MIT + file LICENSE","GPL-2","MIT + file LICENSE","GPL","LGPL (&gt;= 2)","MIT + file LICENSE","MIT + file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","GPL-2","GPL-2","MIT + file LICENSE","Artistic-2.0","GPL-2","BSD_3_clause + file LICENSE","GPL-3 | file LICENSE","LGPL (&gt;= 2.1)","LGPL (&gt;= 2.1)","LGPL (&gt;= 2.1)","MIT + file LICENSE","GPL (&gt;= 2)","Artistic-2.0","MIT + file LICENSE","Apache License 2.0","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL (&gt;= 2)","BSD","GPL (&gt;= 2)","GPL (&gt;= 2) | file LICENSE","GPL-3","GPL-2","MIT + file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","MIT + file LICENSE","Artistic-2.0","Artistic-2.0","Artistic-2.0","GPL-3","GPL-3","GPL (&gt;= 2)","GPL (&gt;= 2)","GPL-3","LGPL (&gt;= 2)","MIT + file LICENSE","GPL-2","GPL-3","GPL-2","Artistic-2.0","GPL (&gt;= 2)","MIT + file LICENSE","GPL","BSD_3_clause + file LICENCE","GPL","GPL-2","GPL (&gt;= 2)","GPL-3 | file LICENSE","GPL","MIT + file LICENSE","GPL (&gt;= 2)","file LICENSE","GPL-2 | file LICENSE","Artistic-2.0","MIT + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","MIT + file LICENSE","GPL-3","GPL-3 | file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","GPL","GPL-3","GPL-3","Apache License (== 2.0) | file LICENSE","MIT + file LICENSE","MIT + file LICENSE","Artistic-2.0","GPL-3","GPL (&gt;= 2)","MIT + file LICENSE","MIT + file LICENSE","BSD_2_clause + file LICENSE","GPL (&gt;= 2)","MIT + file LICENSE","GPL (&gt;= 2)","Artistic-2.0","BSD_3_clause + file LICENSE","Artistic-2.0 + file LICENSE","Part of R 3.5.2","Unlimited","GPL-2 | GPL-3","GPL (&gt;= 2)","GPL","Part of R 3.5.2","Part of R 3.5.2","GPL (&gt;= 2)","Part of R 3.5.2","Part of R 3.5.2","Part of R 3.5.2","Unlimited","GPL (&gt;= 2)","GPL-2 | GPL-3","GPL (&gt;= 2) | file LICENCE","Part of R 3.5.2","GPL (&gt;= 2)","GPL (&gt;= 2) | file LICENCE","GPL-2 | GPL-3","Part of R 3.5.2","GPL-2 | GPL-3","GPL-2 | GPL-3","Part of R 3.5.2","Part of R 3.5.2","Part of R 3.5.2","LGPL (&gt;= 2)","Part of R 3.5.2","Part of R 3.5.2","Part of R 3.5.2"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"yes",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["yes","yes","no","no","no","yes","yes","no","yes","yes","no","yes","no","no","yes","no","no","no","yes","yes","yes","no","no",null,"no","no","yes","yes","no","no","no","no","no","yes","yes","no","no","no","yes","yes","no","no","yes","no","no","no","no","yes","yes","no","yes","no","yes","yes","no","no","yes","no","yes","no","no","no","no","yes","no","no","yes","no","yes","no","no","no","yes","no","no","no","yes","yes","yes","yes","no","no","yes","yes","yes","no","no","yes","no","yes","no","no","yes","yes","no","no","yes","yes","no","yes","yes","no","no","no","no","yes","no","yes","yes","yes","yes","no","yes","yes","yes","yes","no","yes","yes",null,"no","yes","no","no","no",null,"no","no","yes","no","yes","yes","no","yes","yes","no","no","no","yes","no","no","yes","no","yes","no","yes","yes","no","yes","no","yes","no","yes","yes","yes","no","no","no","no","yes","yes","no","no","yes","yes","yes","yes","no","yes","yes","no","no","no","no","yes","no","yes","yes","yes","no","yes","yes","no","yes","no","no","no","no","yes","yes","yes","no","no","no","no","no","no","no","yes","no","yes","no","no","yes","no","yes","yes","yes","no","no","no","no","no","no","yes","no","no","yes","no","no","no","no","yes","yes","no","no","yes","yes","yes",null,"no","yes","yes","no",null,null,"yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes",null,"yes","yes","yes","yes"],["3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2","3.5.2"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>LibPath<\/th>\n      <th>Version<\/th>\n      <th>Priority<\/th>\n      <th>Depends<\/th>\n      <th>Imports<\/th>\n      <th>LinkingTo<\/th>\n      <th>Suggests<\/th>\n      <th>Enhances<\/th>\n      <th>License<\/th>\n      <th>License_is_FOSS<\/th>\n      <th>License_restricts_use<\/th>\n      <th>OS_type<\/th>\n      <th>MD5sum<\/th>\n      <th>NeedsCompilation<\/th>\n      <th>Built<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>It is always useful to finish a Rmarkdown report with a section providing all the session information details with the <code>sessionInfo()</code> function, such at the end of this material. This allows readers to review the version of R itself and all the packages that were used to produce the report.</li>
</ul>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Prepare an Rmarkdown report summarising the portal ecology data. The report should include a <em>Material and methods</em> section where the data is read in (ideally from the online file) and briefly described, a <em>Data preparation</em> section where rows with missing values are filtered out, and a <em>Visualisation</em> section where one or two plots are rendered. Finish your report with a <em>Session information</em> section.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<p>The <a href="https://github.com/rstudio/cheatsheets/raw/master/rmarkdown-2.0.pdf">R Markdown Cheat Sheet</a> and <a href="https://www.rstudio.com/wp-content/uploads/2015/03/rmarkdown-reference.pdf">Reference Guide</a> will help you with the markdown synatax, R code chunk options, and RStudio utlisation.</p>
</div>
<div id="docker-containers" class="section level2">
<h2><span class="header-section-number">7.3</span> Docker containers</h2>
<p>There are other tools for reproducible research, that aim to disseminate more than code and data. Docker containers for example enable to share the complete image of an operating system, including all system dependencies and software/data to repeat a complete analysis. These are useful tools, even though their aren’t necessarily ideal, and beyond the aim of this course.</p>
</div>
<div id="additional-exercises-6" class="section level2">
<h2><span class="header-section-number">7.4</span> Additional exercises</h2>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Repeat the <em>Beer consumption</em> analysis using the that was done in chapters <a href="sec-dplyr.html#sec:dplyr">4</a> and <a href="sec-vis.html#sec:vis">5</a>.</p>
<p>To test if your report is fully reproducible, uniquely name the <code>Rmd</code> file as <code>surname_surname_beers_report.Rmd</code>, post it on the course forum and ask your neighbour to download it, compile it into pdf and provide feedback on whether the document was reproducible and easy to follow<label for="tufte-sn-16" class="margin-toggle sidenote-number">16</label><input type="checkbox" id="tufte-sn-16" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">16</span> This is an important exercise, as it will mimic the exam situation, where you will hand in your <code>Rmd</code> reports that will need to be compiled (to pdf) before marking. In addition, it demonstrates the challenges of writing and reproducing an understandable and reproducible document.</span>.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>

</div>
</div>
<p style="text-align: center;">
<a href="sec-join.html"><button class="btn btn-default">Previous</button></a>
<a href="sec-bioinfo.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2019-03-12
</p>
</div>
</div>



</body>
</html>
