<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta property="og:title" content="Chapter 8 Bioinformatics | Introduction to bioinformatics" />
<meta property="og:type" content="book" />


<meta property="og:description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2019-01-16" />


<meta name="description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain.">

<title>Chapter 8 Bioinformatics | Introduction to bioinformatics</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/jquery-1.12.4/jquery.min.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.5/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>



</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Introduction to bioinformatics<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html">Preamble</a>
<a href="dataorg.html"><span class="toc-section-number">1</span> Data organisation with Spreadsheets</a>
<a href="starting-with-r.html"><span class="toc-section-number">2</span> Starting with R</a>
<a href="starting-with-data.html"><span class="toc-section-number">3</span> Starting with data</a>
<a href="manipulating-and-analyzing-data-with-dplyr.html"><span class="toc-section-number">4</span> Manipulating and analyzing data with dplyr</a>
<a href="data-visualization.html"><span class="toc-section-number">5</span> Data visualization</a>
<a href="joining-tables.html"><span class="toc-section-number">6</span> Joining tables</a>
<a href="reproducible-research.html"><span class="toc-section-number">7</span> Reproducible research</a>
<a id="active-page" href="sec-bioinfo.html"><span class="toc-section-number">8</span> Bioinformatics</a><ul class="toc-sections">
<li class="toc"><a href="#omics-data"> Omics data</a></li>
<li class="toc"><a href="#sec:bioconductor"> The Bioconductor project</a></li>
<li class="toc"><a href="#objects"> Objects</a></li>
<li class="toc"><a href="#documentation"> Documentation</a></li>
<li class="toc"><a href="#exercises"> Exercises</a></li>
</ul>
<a href="additional-programming-concepts.html"><span class="toc-section-number">9</span> Additional programming concepts</a>
<a href="conclusions.html"><span class="toc-section-number">10</span> Conclusions</a>
<a href="session-information.html">Session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="sec:bioinfo" class="section level1">
<h1><span class="header-section-number">Chapter 8</span> Bioinformatics</h1>
<p>As already alluded to ealier, <a href="https://en.wikipedia.org/wiki/Bioinformatics">Wikipedia
defines</a> bioinformatics
as</p>
<blockquote>
<p>Bioinformatics is an interdisciplinary field that develops methods
and software tools for understanding biological data.</p>
</blockquote>
<p>Bioinformatics is as varied as biology itself, and ranges from data
analysis, to software development, computational or statistical
methodological development, more theoretical work, as well as any
combination of these.</p>
<div id="omics-data" class="section level2">
<h2><span class="header-section-number">8.1</span> Omics data</h2>
<p>So far, we have explored broad data science techniques in R. A
widespread and successful area of bioinfomatics, and one that you, as
a biology or biomedical science student are likely to be confronted
with, is the analysis and interpretation of omics data.</p>
<p>
<span class="marginnote shownote">
<!--
<div class="figure">--><span id="fig:infoflow"></span>
<img src="figs/Centraldogma_nodetails.png" alt="Information flow in biological systems (Source [Wikipedia](https://en.wikipedia.org/wiki/Central_dogma_of_molecular_biology))." width="368"  />
<!--
<p class="caption marginnote">-->Figure 8.1: Information flow in biological systems (Source <a href="https://en.wikipedia.org/wiki/Central_dogma_of_molecular_biology">Wikipedia</a>).<!--</p>-->
<!--</div>--></span>
</p>
<p>It is useful to define these omics data along the flow of information
in biology (Figure <a href="sec-bioinfo.html#fig:infoflow">8.1</a>), and define the different
application domains. The technologies that focus on DNA, and the
genome in particular (either whole or parts thereof) are termed
<strong>genomics</strong>, and are currently based on sequencing, in particular
high throughput sequencing (HTS). The domain focusing on the study of
any DNA modification (such as for example methylation) is termed
<strong>epigenetics</strong>. The study of RNA, and more specifically the
quantitation of RNA levels in biological samples is termed
<strong>transcriptomics</strong>, as it assays the transcription of DNA into RNA
molecules. Without further specification, transcriptomics refers to
the quantitation of messanger RNA, although one could also focus on
non-coding RNAs such as micro RNAs. HTS is currently the technology of
choice for any transcriptomics study, while a decade ago, prior to the
development of RNA sequencing (called <strong>RNAseq</strong>), microarrays were
widely used. <strong>Proteomics</strong> focuses on the identification and
quantitation of proteins, and can also expand into the study of
protein interactions, post-translational modifications or sub-cellular
localisation of proteins. Further downstreams of proteins, small
molecules or lipids can also be assayed under the umbrella terms of
<strong>metabolomics</strong> and <strong>lipidomics</strong>. The technology of choice for
protein, lipids or smaller metabolites is mass spectrometry.</p>
<p>Important concepts</p>
<ul>
<li>high throughput</li>
<li>raw data</li>
<li>processed data</li>
<li>metadata, including experimental design (see also stats course)</li>
</ul>
<p>Reminder about reproducible research, and need for raw data and
complete processing script to reproduce the processed data and
results.</p>
</div>
<div id="sec:bioconductor" class="section level2">
<h2><span class="header-section-number">8.2</span> The Bioconductor project</h2>
<p>The <a href="http://www.bioconductor.org">Bioconductor</a> …</p>
</div>
<div id="objects" class="section level2">
<h2><span class="header-section-number">8.3</span> Objects</h2>
<p>Data in bioinformatics is often more complex than the basic data types
we have seen so far. Give a short introduction and example to some
relevant object.</p>
</div>
<div id="documentation" class="section level2">
<h2><span class="header-section-number">8.4</span> Documentation</h2>
<p>Documenting ones tasks in a bioinformatics project is absolutely
essential. It is akin a lab notebook for wet lab.</p>
</div>
<div id="exercises" class="section level2">
<h2><span class="header-section-number">8.5</span> Exercises</h2>
<p>More exercises, framed in the context of bioinformatics.</p>

</div>
</div>
<p style="text-align: center;">
<a href="reproducible-research.html"><button class="btn btn-default">Previous</button></a>
<a href="additional-programming-concepts.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2019-01-16
</p>
</div>
</div>



</body>
</html>
